{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "executionInfo": {
     "elapsed": 3605,
     "status": "ok",
     "timestamp": 1679075791770,
     "user": {
      "displayName": "19PT24 - S KIRUPA",
      "userId": "12629547787028680648"
     },
     "user_tz": -330
    },
    "id": "RvaNWdrKgXjD",
    "outputId": "7fe66ab5-ecad-49ad-f8f1-6398703962bc"
   },
   "outputs": [],
   "source": [
    "#We extract the pubmed document in BioCJSON format\n",
    "import urllib3\n",
    "import json\n",
    "import csv\n",
    "import requests\n",
    "import pandas as pd\n",
    "import io\n",
    "import os\n",
    "#import nltk\n",
    "#nltk.download('punkt')\n",
    "#from nltk.tokenize import word_tokenize\n",
    "from itertools import combinations\n",
    "\n",
    "http = urllib3.PoolManager()\n",
    "\n",
    "pmcids = []\n",
    "\n",
    "\n",
    "# Read the PMC IDs from the input file\n",
    "with open(\"pmcid.txt\", \"r\") as file:\n",
    "    for line in file:\n",
    "        pmcids.append(line.strip())\n",
    "# Initialize an empty list to store the results\n",
    "results = []\n",
    "\n",
    "for pmcid in pmcids:\n",
    "    r = http.request('GET', f'https://www.ncbi.nlm.nih.gov/research/pubtator-api/publications/export/biocjson?pmcids={pmcid}')\n",
    "    data = json.loads(r.data.decode('utf-8'))\n",
    "\n",
    "    # Add the results to the list\n",
    "    results.append(data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 143
    },
    "executionInfo": {
     "elapsed": 410,
     "status": "ok",
     "timestamp": 1679075793852,
     "user": {
      "displayName": "19PT24 - S KIRUPA",
      "userId": "12629547787028680648"
     },
     "user_tz": -330
    },
    "id": "-3bUHWus1nUq",
    "outputId": "3f1c4784-c8ea-473b-d8bb-aefd779ea62b"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'\\nEvery PMC id has passages.\\nEvery passage has many {infons, offset, text, sentences, annotations, relations}.\\nHere, text is the actual text we have to annotate. \\nEvery annotations has {id, infons, text, locations}. \\nHere infons has {identifier, type} (optional: ncbi-homologene if type is gene).\\nAlso locations has {offset, length}.\\n\\nPassages:\\n    a)infons - data realted article id, author, etc..\\n    b)offset - location index\\n    c)text - whole medical data (sentence) in which medical terms (gene name or disease name) are to be annotated.\\n    d)sentences - not required here\\n    e)annotations:\\n        1)id - key index\\n        2)infons:\\n              a)identifier\\n              b)type - \"Gene\" or \"Disease\" etc.,\\n        3)text - gene name or disease name etc., (Eg: \"K-Ras\")\\n        4)locations:\\n            a)offset - location index\\n            b)length - length of text. (Eg: len(\"tumours\") = 7) \\n    f)relations\\n'"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "output = json.dumps(data, indent=4)\n",
    "#print(output)\n",
    "\n",
    "'''\n",
    "Every PMC id has passages.\n",
    "Every passage has many {infons, offset, text, sentences, annotations, relations}.\n",
    "Here, text is the actual text we have to annotate. \n",
    "Every annotations has {id, infons, text, locations}. \n",
    "Here infons has {identifier, type} (optional: ncbi-homologene if type is gene).\n",
    "Also locations has {offset, length}.\n",
    "\n",
    "Passages:\n",
    "    a)infons - data realted article id, author, etc..\n",
    "    b)offset - location index\n",
    "    c)text - whole medical data (sentence) in which medical terms (gene name or disease name) are to be annotated.\n",
    "    d)sentences - not required here\n",
    "    e)annotations:\n",
    "        1)id - key index\n",
    "        2)infons:\n",
    "              a)identifier\n",
    "              b)type - \"Gene\" or \"Disease\" etc.,\n",
    "        3)text - gene name or disease name etc., (Eg: \"K-Ras\")\n",
    "        4)locations:\n",
    "            a)offset - location index\n",
    "            b)length - length of text. (Eg: len(\"tumours\") = 7) \n",
    "    f)relations\n",
    "'''\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "id": "3EkWKX8wA1b5"
   },
   "outputs": [],
   "source": [
    "df1 = pd.DataFrame(columns=['index','sentence'])\n",
    "df2 = pd.DataFrame(columns=['index','sentence','Disease','Mutation'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "ZSMLCcRaAlL_"
   },
   "source": [
    "**Annotations for Gene and Mutation**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "id": "kbIqNKyCgkuF"
   },
   "outputs": [],
   "source": [
    "sentence_index = 0\n",
    "sentence_entities = {}\n",
    "for j in range(len(results)):\n",
    "  for i in results[j]['passages']:\n",
    "    if i['infons']['section_type'] != 'TABLE':\n",
    "      #filter the table segment\n",
    "      text = i['text']\n",
    "      # print(\"TEXT: \", text)\n",
    "      offset = i['offset']\n",
    "      # print(\"OFFSET: \", offset)    \n",
    "      annotations = i['annotations']\n",
    "      \n",
    "      annotations = sorted(annotations, key = lambda x: x['locations'][0]['offset'])\n",
    "      # print(\"ANNOTATIONS: \", annotations)\n",
    "      #Filter to only include gene-disease annotations\n",
    "      annotations = [annotation for annotation in annotations if ((annotation['infons']['type']=='Disease') or (annotation['infons']['type']=='Mutation'))]\n",
    "      #List all possible combinations of annotations\n",
    "      annots_combinations = list(combinations(annotations, 2))\n",
    "      # print(\"ANNOT COMBIMNATIONS: \", annots_combinations)\n",
    "      #Filter combinations to only include gene-disease combinations\n",
    "      annots_combinations = [annots for annots in annots_combinations if annots[0]['infons']['type'] != annots[1]['infons']['type']]\n",
    "      # print(\"ANNOT COMBINATIONS - GENE-DISEASE: \", annots_combinations)\n",
    "\n",
    "      #processing sentences\n",
    "      sentences = text.split('. ')\n",
    "      # print(sentences)\n",
    "      sentence_offset = {}\n",
    "      sentence_len = {}\n",
    "      prev_sent_offset = offset\n",
    "\n",
    "      for sentence in sentences:\n",
    "\n",
    "        sentence_offset[sentence] = prev_sent_offset\n",
    "        current_sentence_offset = prev_sent_offset\n",
    "        sentence_len[sentence] = len(sentence)\n",
    "        current_sentence_len = len(sentence)\n",
    "        prev_sent_offset += len(sentence) + 2\n",
    "        \n",
    "\n",
    "        #Point to note: Duplicate sentences for as many combinations as present. \n",
    "        for annots in annots_combinations:\n",
    "          difference = 0\n",
    "          current_sentence = sentence\n",
    "          #sort the tuple\n",
    "          annots = sorted(annots, key = lambda x: x['locations'][0]['offset'])\n",
    "\n",
    "          entity_1 = annots[0]\n",
    "          entity_2 = annots[1]\n",
    "          \n",
    "          entity_1_offset = entity_1['locations'][0]['offset']\n",
    "          entity_2_offset = entity_2['locations'][0]['offset']\n",
    "\n",
    "          entity_1_dist = entity_1_offset - current_sentence_offset\n",
    "          entity_2_dist = entity_2_offset - current_sentence_offset\n",
    "\n",
    "          if (0 <= entity_1_dist <= ((current_sentence_len - len(entity_1['text'])) + 1)) and (0 <= entity_2_dist <= ((current_sentence_len - len(entity_2['text'])) + 1)):\n",
    "            #the pair of annotations fall within the sentence\n",
    "            sentence_entities[sentence_index] = (entity_1['text'], entity_2['text'])\n",
    "\n",
    "            entity_1_type = entity_1['infons']['type']\n",
    "            entity_1_length = entity_1['locations'][0]['length']\n",
    "            temp = '@'+ entity_1_type +'$'\n",
    "            entity_1_final_off = entity_1_dist \n",
    "            current_sentence = current_sentence[:entity_1_final_off] + \"@\" + entity_1_type + \"$\" + current_sentence[(entity_1_final_off + entity_1_length):]\n",
    "            difference += (entity_1_length - len(temp))\n",
    "            entity_2_type = entity_2['infons']['type']\n",
    "            entity_2_length = entity_2['locations'][0]['length']\n",
    "            temp = '@'+ entity_2_type +'$'\n",
    "            entity_2_final_off = entity_2_dist - (difference)\n",
    "            current_sentence = current_sentence[:entity_2_final_off] + \"@\" + entity_2_type + \"$\" + current_sentence[(entity_2_final_off + entity_2_length):]\n",
    "            difference += (entity_2_length - len(temp))\n",
    "            # tsv_writer_1.writerow([sentence_index, current_sentence])\n",
    "            row1 = [sentence_index,current_sentence]\n",
    "            df1.loc[len(df1)] = row1\n",
    "            if (entity_1['infons']['type'] == 'Disease'):\n",
    "              # print('Writing...', [sentence_index, sentence, entity_1['text'], entity_2['text']])\n",
    "              # tsv_writer_2.writerow([sentence_index, sentence, entity_1['text'], entity_2['text']])\n",
    "              row2 = [sentence_index, sentence, entity_1['text'], entity_2['text']]\n",
    "              df2.loc[len(df2)] = row2\n",
    "            else:\n",
    "              # print('Writing')\n",
    "              # tsv_writer_2.writerow([sentence_index, sentence, entity_2['text'], entity_1['text']])\n",
    "              row2 = [sentence_index, sentence, entity_2['text'], entity_1['text']]\n",
    "              df2.loc[len(df2)] = row2\n",
    "            sentence_index += 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "id": "rax5aqQmrZp-"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/skirupa/Desktop/sem8/RSL-Lab/RSL Final/GDM-BioBert/project-sample\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "print(os.getcwd())\n",
    "os.chdir('../biobert-pytorch/relation-extraction')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "id": "Oe0kay_JgprA"
   },
   "outputs": [],
   "source": [
    "df2.to_csv('pub_original_sentences_DM.tsv', sep=\"\\t\", index=False)\n",
    "df1.to_csv('inputDM/test.tsv', sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "32"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(df1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "ZZSYZQAAi5db"
   },
   "source": [
    "**BioBert Training**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/skirupa/Desktop/sem8/RSL-Lab/RSL Final/GDM-BioBert/biobert-pytorch/relation-extraction\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "print(os.getcwd())\n",
    "os.chdir('..')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Download all datasets including NER/RE/QA\n",
    "#!bash ./download.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/skirupa/Desktop/sem8/RSL-Lab/RSL Final/GDM-BioBert/biobert-pytorch\n",
      "/home/skirupa/Desktop/sem8/RSL-Lab/RSL Final/GDM-BioBert/biobert-pytorch/relation-extraction\n",
      "*****  euadr  Preprocessing Start *****\n",
      "*****  euadr  Preprocessing Done *****\n",
      "*****  GAD  Preprocessing Start *****\n",
      "*****  GAD  Preprocessing Done *****\n"
     ]
    }
   ],
   "source": [
    "print(os.getcwd())\n",
    "os.chdir('relation-extraction')\n",
    "print(os.getcwd())\n",
    "#To preprocess the datasets downloaded\n",
    "!bash ./preprocess.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "executionInfo": {
     "elapsed": 8981,
     "status": "ok",
     "timestamp": 1679070353775,
     "user": {
      "displayName": "19PT24 - S KIRUPA",
      "userId": "12629547787028680648"
     },
     "user_tz": -330
    },
    "id": "jPCXvCJLkLsy",
    "outputId": "b2ffa5ef-ecb7-4bd9-dbe3-640d4f36ddcc"
   },
   "outputs": [],
   "source": [
    "#!pip install scikit-learn\n",
    "#!pip install pandas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "executionInfo": {
     "elapsed": 6,
     "status": "ok",
     "timestamp": 1679070381883,
     "user": {
      "displayName": "19PT24 - S KIRUPA",
      "userId": "12629547787028680648"
     },
     "user_tz": -330
    },
    "id": "uv7yHEmWkPk8",
    "outputId": "678d9197-a2d7-41d0-fa94-9dc9a2532ad6"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "env: SAVE_DIR=./outputDM\n",
      "env: DATA=\"GAD\"\n",
      "env: SPLIT=\"1\"\n",
      "env: DATA_DIR=./inputDM\n",
      "env: ENTITY=${DATA}-${SPLIT}\n",
      "env: MAX_LENGTH=128\n",
      "env: BATCH_SIZE=32\n",
      "env: NUM_EPOCHS=3\n",
      "env: SAVE_STEPS=1000\n",
      "env: SEED=1\n"
     ]
    }
   ],
   "source": [
    "%env SAVE_DIR=./outputDM\n",
    "%env DATA=\"GAD\"\n",
    "%env SPLIT=\"1\"\n",
    "%env DATA_DIR=./inputDM\n",
    "%env ENTITY=${DATA}-${SPLIT}\n",
    "\n",
    "%env MAX_LENGTH=128\n",
    "%env BATCH_SIZE=32\n",
    "%env NUM_EPOCHS=3\n",
    "%env SAVE_STEPS=1000\n",
    "%env SEED=1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "executionInfo": {
     "elapsed": 7095,
     "status": "ok",
     "timestamp": 1679070394106,
     "user": {
      "displayName": "19PT24 - S KIRUPA",
      "userId": "12629547787028680648"
     },
     "user_tz": -330
    },
    "id": "Nap7OsDNomOM",
    "outputId": "4ba15580-e24e-4619-f13d-fb35ad41f6ed"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "05/01/2023 20:48:46 - WARNING - __main__ -   Process rank: -1, device: cpu, n_gpu: 0, distributed training: False, 16-bits training: False\n",
      "05/01/2023 20:48:55 - INFO - __main__ -   Training/evaluation parameters TrainingArguments(output_dir='./outputDM', overwrite_output_dir=True, do_train=False, do_eval=False, do_predict=True, evaluate_during_training=False, evaluation_strategy=<EvaluationStrategy.NO: 'no'>, prediction_loss_only=False, per_device_train_batch_size=8, per_device_eval_batch_size=8, per_gpu_train_batch_size=None, per_gpu_eval_batch_size=None, gradient_accumulation_steps=1, eval_accumulation_steps=None, learning_rate=5e-05, weight_decay=0.0, adam_beta1=0.9, adam_beta2=0.999, adam_epsilon=1e-08, max_grad_norm=1.0, num_train_epochs=3.0, max_steps=-1, warmup_steps=0, logging_dir='runs/May01_20-48-46_skirupa', logging_first_step=False, logging_steps=500, save_steps=500, save_total_limit=None, no_cuda=False, seed=42, fp16=False, fp16_opt_level='O1', local_rank=-1, tpu_num_cores=None, tpu_metrics_debug=False, debug=False, dataloader_drop_last=False, eval_steps=500, dataloader_num_workers=0, past_index=-1, run_name='./outputDM', disable_tqdm=False, remove_unused_columns=True, label_names=None, load_best_model_at_end=False, metric_for_best_model=None, greater_is_better=None)\n",
      "Some weights of the model checkpoint at dmis-lab/biobert-base-cased-v1.1 were not used when initializing BertForSequenceClassification: ['cls.predictions.bias', 'cls.predictions.transform.dense.weight', 'cls.predictions.transform.dense.bias', 'cls.predictions.transform.LayerNorm.weight', 'cls.predictions.transform.LayerNorm.bias', 'cls.predictions.decoder.weight', 'cls.predictions.decoder.bias', 'cls.seq_relationship.weight', 'cls.seq_relationship.bias']\n",
      "- This IS expected if you are initializing BertForSequenceClassification from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPretraining model).\n",
      "- This IS NOT expected if you are initializing BertForSequenceClassification from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).\n",
      "Some weights of BertForSequenceClassification were not initialized from the model checkpoint at dmis-lab/biobert-base-cased-v1.1 and are newly initialized: ['classifier.weight', 'classifier.bias']\n",
      "You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.\n",
      "05/01/2023 20:49:07 - INFO - root -   *** Test ***\n",
      "100%|█████████████████████████████████████████████| 2/2 [00:00<00:00,  2.08it/s]/home/skirupa/miniconda3/envs/env1/lib/python3.6/site-packages/transformers/trainer.py:1118: FutureWarning: This method is deprecated, use `Trainer.is_world_process_zero()` instead.\n",
      "  warnings.warn(\"This method is deprecated, use `Trainer.is_world_process_zero()` instead.\", FutureWarning)\n",
      "05/01/2023 20:49:11 - INFO - __main__ -   ***** Test results sst-2 *****\n",
      "100%|█████████████████████████████████████████████| 2/2 [00:00<00:00,  2.02it/s]\n"
     ]
    }
   ],
   "source": [
    "!python run_re.py --task_name SST-2 --config_name bert-base-cased --model_name_or_path dmis-lab/biobert-base-cased-v1.1 \\\n",
    "        --do_predict --data_dir ${DATA_DIR} \\\n",
    "        --output_dir ${SAVE_DIR} \\\n",
    "        --overwrite_output_dir\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 328
    },
    "executionInfo": {
     "elapsed": 13,
     "status": "error",
     "timestamp": 1679064615235,
     "user": {
      "displayName": "19PT24 - S KIRUPA",
      "userId": "12629547787028680648"
     },
     "user_tz": -330
    },
    "id": "QgHoYa6Noo9w",
    "outputId": "6af921bc-1802-4a23-bb95-187243510d51"
   },
   "outputs": [],
   "source": [
    "original_sentences = pd.read_csv('pub_original_sentences_DM.tsv', sep=\"\\t\")\n",
    "predictions = pd.read_csv('outputDM/test_results.txt', sep='\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "id": "QX9DHYpyBbeC"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>sentence</th>\n",
       "      <th>Disease</th>\n",
       "      <th>Mutation</th>\n",
       "      <th>prediction</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>Four additional K-Ras mutations (Leu19Phe (1 o...</td>\n",
       "      <td>tumours</td>\n",
       "      <td>Leu19Phe</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>Four additional K-Ras mutations (Leu19Phe (1 o...</td>\n",
       "      <td>tumours</td>\n",
       "      <td>Lys117Asn</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>Four additional K-Ras mutations (Leu19Phe (1 o...</td>\n",
       "      <td>tumours</td>\n",
       "      <td>Ala146Thr</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>Four additional K-Ras mutations (Leu19Phe (1 o...</td>\n",
       "      <td>tumours</td>\n",
       "      <td>Arg164Gln</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>The most common B-Raf mutation is a T to A tra...</td>\n",
       "      <td>tumours</td>\n",
       "      <td>T to A</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>5</td>\n",
       "      <td>The most common B-Raf mutation is a T to A tra...</td>\n",
       "      <td>tumours</td>\n",
       "      <td>valine to glutamic acid</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>6</td>\n",
       "      <td>The most common B-Raf mutation is a T to A tra...</td>\n",
       "      <td>tumours</td>\n",
       "      <td>V600E</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>7</td>\n",
       "      <td>Meanwhile, 1 mul chromosome 12p BAC RP11-707G1...</td>\n",
       "      <td>tumour</td>\n",
       "      <td>C for 5</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>8</td>\n",
       "      <td>Meanwhile, 1 mul chromosome 12p BAC RP11-707G1...</td>\n",
       "      <td>tumour</td>\n",
       "      <td>C for 5</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>9</td>\n",
       "      <td>Mutations at codon 19 (G57T, Leu19Phe) and in ...</td>\n",
       "      <td>tumours</td>\n",
       "      <td>G57T</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>10</td>\n",
       "      <td>Mutations at codon 19 (G57T, Leu19Phe) and in ...</td>\n",
       "      <td>tumours</td>\n",
       "      <td>Leu19Phe</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>11</td>\n",
       "      <td>Mutations at codon 19 (G57T, Leu19Phe) and in ...</td>\n",
       "      <td>tumours</td>\n",
       "      <td>V600E</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>12</td>\n",
       "      <td>Mutation analysis in Costello syndrome: functi...</td>\n",
       "      <td>Costello syndrome</td>\n",
       "      <td>p.Lys117Arg</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>13</td>\n",
       "      <td>Following a pilot study in Italy and Spain, an...</td>\n",
       "      <td>breast cancer</td>\n",
       "      <td>rs299290</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>14</td>\n",
       "      <td>Effect of HMMR rs299290 variation on breast ca...</td>\n",
       "      <td>breast cancer</td>\n",
       "      <td>rs299290</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>15</td>\n",
       "      <td>However, while the rs299290 variant represents...</td>\n",
       "      <td>SIFT</td>\n",
       "      <td>rs299290</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>16</td>\n",
       "      <td>However, while the rs299290 variant represents...</td>\n",
       "      <td>SIFT</td>\n",
       "      <td>V368A</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>17</td>\n",
       "      <td>However, while the rs299290 variant represents...</td>\n",
       "      <td>SIFT</td>\n",
       "      <td>rs299284</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>18</td>\n",
       "      <td>However, while the rs299290 variant represents...</td>\n",
       "      <td>SIFT</td>\n",
       "      <td>R92C</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>19</td>\n",
       "      <td>In data provided by several CIMBA groups (Text...</td>\n",
       "      <td>tumors</td>\n",
       "      <td>rs299290</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>20</td>\n",
       "      <td>That is, despite the expected differences in t...</td>\n",
       "      <td>tumor</td>\n",
       "      <td>rs299290</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>21</td>\n",
       "      <td>Importantly, both proteasome inhibition and BR...</td>\n",
       "      <td>breast cancer</td>\n",
       "      <td>S7</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>22</td>\n",
       "      <td>This model is consistent with reduced expressi...</td>\n",
       "      <td>growth arrest</td>\n",
       "      <td>S11A</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>23</td>\n",
       "      <td>Accordingly, BRCA1 depletion increases the clo...</td>\n",
       "      <td>epithelia</td>\n",
       "      <td>S308A</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>24</td>\n",
       "      <td>This work was funded by the Spanish Ministries...</td>\n",
       "      <td>Contra el Cancer</td>\n",
       "      <td>C1287/A8874</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>25</td>\n",
       "      <td>This work was funded by the Spanish Ministries...</td>\n",
       "      <td>Childhood Cancer</td>\n",
       "      <td>C1287/A8874</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>26</td>\n",
       "      <td>This work was funded by the Spanish Ministries...</td>\n",
       "      <td>cancer</td>\n",
       "      <td>C1287/A8874</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>27</td>\n",
       "      <td>This work was funded by the Spanish Ministries...</td>\n",
       "      <td>Hereditary Tumors</td>\n",
       "      <td>C1287/A8874</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>28</td>\n",
       "      <td>This work was funded by the Spanish Ministries...</td>\n",
       "      <td>Breast Cancer</td>\n",
       "      <td>C1287/A8874</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>29</td>\n",
       "      <td>This work was funded by the Spanish Ministries...</td>\n",
       "      <td>Breast Cancer</td>\n",
       "      <td>C1287/A8874</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>30</td>\n",
       "      <td>This work was funded by the Spanish Ministries...</td>\n",
       "      <td>Breast Cancer</td>\n",
       "      <td>C1287/A8874</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>31</td>\n",
       "      <td>Inconsistent association between the STK15 F31...</td>\n",
       "      <td>breast cancer</td>\n",
       "      <td>F31I</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    index                                           sentence  \\\n",
       "0       0  Four additional K-Ras mutations (Leu19Phe (1 o...   \n",
       "1       1  Four additional K-Ras mutations (Leu19Phe (1 o...   \n",
       "2       2  Four additional K-Ras mutations (Leu19Phe (1 o...   \n",
       "3       3  Four additional K-Ras mutations (Leu19Phe (1 o...   \n",
       "4       4  The most common B-Raf mutation is a T to A tra...   \n",
       "5       5  The most common B-Raf mutation is a T to A tra...   \n",
       "6       6  The most common B-Raf mutation is a T to A tra...   \n",
       "7       7  Meanwhile, 1 mul chromosome 12p BAC RP11-707G1...   \n",
       "8       8  Meanwhile, 1 mul chromosome 12p BAC RP11-707G1...   \n",
       "9       9  Mutations at codon 19 (G57T, Leu19Phe) and in ...   \n",
       "10     10  Mutations at codon 19 (G57T, Leu19Phe) and in ...   \n",
       "11     11  Mutations at codon 19 (G57T, Leu19Phe) and in ...   \n",
       "12     12  Mutation analysis in Costello syndrome: functi...   \n",
       "13     13  Following a pilot study in Italy and Spain, an...   \n",
       "14     14  Effect of HMMR rs299290 variation on breast ca...   \n",
       "15     15  However, while the rs299290 variant represents...   \n",
       "16     16  However, while the rs299290 variant represents...   \n",
       "17     17  However, while the rs299290 variant represents...   \n",
       "18     18  However, while the rs299290 variant represents...   \n",
       "19     19  In data provided by several CIMBA groups (Text...   \n",
       "20     20  That is, despite the expected differences in t...   \n",
       "21     21  Importantly, both proteasome inhibition and BR...   \n",
       "22     22  This model is consistent with reduced expressi...   \n",
       "23     23  Accordingly, BRCA1 depletion increases the clo...   \n",
       "24     24  This work was funded by the Spanish Ministries...   \n",
       "25     25  This work was funded by the Spanish Ministries...   \n",
       "26     26  This work was funded by the Spanish Ministries...   \n",
       "27     27  This work was funded by the Spanish Ministries...   \n",
       "28     28  This work was funded by the Spanish Ministries...   \n",
       "29     29  This work was funded by the Spanish Ministries...   \n",
       "30     30  This work was funded by the Spanish Ministries...   \n",
       "31     31  Inconsistent association between the STK15 F31...   \n",
       "\n",
       "              Disease                 Mutation  prediction  \n",
       "0             tumours                 Leu19Phe         1.0  \n",
       "1             tumours                Lys117Asn         1.0  \n",
       "2             tumours                Ala146Thr         1.0  \n",
       "3             tumours                Arg164Gln         1.0  \n",
       "4             tumours                   T to A         0.0  \n",
       "5             tumours  valine to glutamic acid         1.0  \n",
       "6             tumours                    V600E         0.0  \n",
       "7              tumour                  C for 5         1.0  \n",
       "8              tumour                  C for 5         1.0  \n",
       "9             tumours                     G57T         1.0  \n",
       "10            tumours                 Leu19Phe         1.0  \n",
       "11            tumours                    V600E         1.0  \n",
       "12  Costello syndrome              p.Lys117Arg         1.0  \n",
       "13      breast cancer                 rs299290         NaN  \n",
       "14      breast cancer                 rs299290         NaN  \n",
       "15               SIFT                 rs299290         NaN  \n",
       "16               SIFT                    V368A         NaN  \n",
       "17               SIFT                 rs299284         NaN  \n",
       "18               SIFT                     R92C         NaN  \n",
       "19             tumors                 rs299290         NaN  \n",
       "20              tumor                 rs299290         NaN  \n",
       "21      breast cancer                       S7         NaN  \n",
       "22      growth arrest                     S11A         NaN  \n",
       "23          epithelia                    S308A         NaN  \n",
       "24   Contra el Cancer              C1287/A8874         NaN  \n",
       "25   Childhood Cancer              C1287/A8874         NaN  \n",
       "26             cancer              C1287/A8874         NaN  \n",
       "27  Hereditary Tumors              C1287/A8874         NaN  \n",
       "28      Breast Cancer              C1287/A8874         NaN  \n",
       "29      Breast Cancer              C1287/A8874         NaN  \n",
       "30      Breast Cancer              C1287/A8874         NaN  \n",
       "31      breast cancer                     F31I         NaN  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Merge the pub original sentences and test results \n",
    "\n",
    "final_re_output = pd.merge(original_sentences, predictions, on ='index', how='left')\n",
    "final_re_output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Store the result in a csv\n",
    "#os.chdir('outputDM')\n",
    "#final_re_output.to_csv('final_DM_output.tsv', sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(0, 'Four additional K-Ras mutations (Leu19Phe (1 out of 106 tumours), Lys117Asn (1 out of 106), Ala146Thr (7 out of 106) and Arg164Gln (1 out of 106)) were identified', 'tumours', 'Leu19Phe', 1.0)\n",
      "(1, 'Four additional K-Ras mutations (Leu19Phe (1 out of 106 tumours), Lys117Asn (1 out of 106), Ala146Thr (7 out of 106) and Arg164Gln (1 out of 106)) were identified', 'tumours', 'Lys117Asn', 1.0)\n",
      "(2, 'Four additional K-Ras mutations (Leu19Phe (1 out of 106 tumours), Lys117Asn (1 out of 106), Ala146Thr (7 out of 106) and Arg164Gln (1 out of 106)) were identified', 'tumours', 'Ala146Thr', 1.0)\n",
      "(3, 'Four additional K-Ras mutations (Leu19Phe (1 out of 106 tumours), Lys117Asn (1 out of 106), Ala146Thr (7 out of 106) and Arg164Gln (1 out of 106)) were identified', 'tumours', 'Arg164Gln', 1.0)\n",
      "(4, 'The most common B-Raf mutation is a T to A transversion, resulting in a valine to glutamic acid substitution (V600E), present in approximately 90% of B-Raf mutant tumours (Davies et al, 2002), which results in a 500-fold increase in in vitro kinase activity and the induction of cell proliferation and transformation (Wan et al, 2004)', 'tumours', 'T to A', 0.0)\n",
      "(5, 'The most common B-Raf mutation is a T to A transversion, resulting in a valine to glutamic acid substitution (V600E), present in approximately 90% of B-Raf mutant tumours (Davies et al, 2002), which results in a 500-fold increase in in vitro kinase activity and the induction of cell proliferation and transformation (Wan et al, 2004)', 'tumours', 'valine to glutamic acid', 1.0)\n",
      "(6, 'The most common B-Raf mutation is a T to A transversion, resulting in a valine to glutamic acid substitution (V600E), present in approximately 90% of B-Raf mutant tumours (Davies et al, 2002), which results in a 500-fold increase in in vitro kinase activity and the induction of cell proliferation and transformation (Wan et al, 2004)', 'tumours', 'V600E', 0.0)\n",
      "(7, 'Meanwhile, 1 mul chromosome 12p BAC RP11-707G18 (Red) (BlueGnome), 1 mul chromosome 12q BAC RP11-89H19 (Green) (BlueGnome) and 9 mul LSI-WCP hybridisation buffer (Vysis) was mixed together for each slide and pipetted onto a 22 x 22 mm coverslip which was placed onto a region of the slide containing tumour tissue and placed onto a heat block at 75 C for 5 min to co-denature the probes and tumour tissue', 'tumour', 'C for 5', 1.0)\n",
      "(8, 'Meanwhile, 1 mul chromosome 12p BAC RP11-707G18 (Red) (BlueGnome), 1 mul chromosome 12q BAC RP11-89H19 (Green) (BlueGnome) and 9 mul LSI-WCP hybridisation buffer (Vysis) was mixed together for each slide and pipetted onto a 22 x 22 mm coverslip which was placed onto a region of the slide containing tumour tissue and placed onto a heat block at 75 C for 5 min to co-denature the probes and tumour tissue', 'tumour', 'C for 5', 1.0)\n",
      "(9, 'Mutations at codon 19 (G57T, Leu19Phe) and in B-Raf (V600E) were found in single tumours (Table 2).', 'tumours', 'G57T', 1.0)\n",
      "(10, 'Mutations at codon 19 (G57T, Leu19Phe) and in B-Raf (V600E) were found in single tumours (Table 2).', 'tumours', 'Leu19Phe', 1.0)\n",
      "(11, 'Mutations at codon 19 (G57T, Leu19Phe) and in B-Raf (V600E) were found in single tumours (Table 2).', 'tumours', 'V600E', 1.0)\n",
      "(12, 'Mutation analysis in Costello syndrome: functional and structural characterization of the HRAS p.Lys117Arg mutation', 'Costello syndrome', 'p.Lys117Arg', 1.0)\n",
      "(13, 'Following a pilot study in Italy and Spain, analysis of carriers (n = 11,609) collected through 24 study groups participating in the Consortium of Investigators of Modifiers of BRCA1/2 (CIMBA) detected significant modification of breast cancer risk by HMMR rs299290 variant among BRCA1, but not BRCA2, mutation carriers: BRCA1 mutation carriers n = 7,584, Cox proportional-hazards regression model, hazard ratio (HR) = 1.08 (95% confidence interval (CI) 1.02-1.13), ptrend = 0.004 (p 2df = 0.014), in the same direction as originally detected in Ashkenazi Jewish populations; BRCA2 mutation carriers n = 3,965, HR = 1.03 (95% CI 0.96-1.10), ptrend = 0.42 (p 2df = 0.67)', 'breast cancer', 'rs299290', None)\n",
      "(14, 'Effect of HMMR rs299290 variation on breast cancer risk among BRCA1 and BRCA2 mutation carriers.', 'breast cancer', 'rs299290', None)\n",
      "(15, \"However, while the rs299290 variant represents a missense change predicted to be benign (V368A; concordant predictions for PolyPhen-2 and SIFT were obtained), it is in linkage disequilibrium (according to HapMap Caucasians data: D' = 1 and r2 = 0.48) with rs299284 (R92C in Entrez accession number NP_036616), which is predicted to be damaging\", 'SIFT', 'rs299290', None)\n",
      "(16, \"However, while the rs299290 variant represents a missense change predicted to be benign (V368A; concordant predictions for PolyPhen-2 and SIFT were obtained), it is in linkage disequilibrium (according to HapMap Caucasians data: D' = 1 and r2 = 0.48) with rs299284 (R92C in Entrez accession number NP_036616), which is predicted to be damaging\", 'SIFT', 'V368A', None)\n",
      "(17, \"However, while the rs299290 variant represents a missense change predicted to be benign (V368A; concordant predictions for PolyPhen-2 and SIFT were obtained), it is in linkage disequilibrium (according to HapMap Caucasians data: D' = 1 and r2 = 0.48) with rs299284 (R92C in Entrez accession number NP_036616), which is predicted to be damaging\", 'SIFT', 'rs299284', None)\n",
      "(18, \"However, while the rs299290 variant represents a missense change predicted to be benign (V368A; concordant predictions for PolyPhen-2 and SIFT were obtained), it is in linkage disequilibrium (according to HapMap Caucasians data: D' = 1 and r2 = 0.48) with rs299284 (R92C in Entrez accession number NP_036616), which is predicted to be damaging\", 'SIFT', 'R92C', None)\n",
      "(19, 'In data provided by several CIMBA groups (Text S1), no ER-positive tumors were observed among rare rs299290 homozygotes in BRCA1 mutation carriers (p interaction = 0.006), whereas this bias was not observed in BRCA2 mutation carriers (p interaction = 0.95) (Table S4)', 'tumors', 'rs299290', None)\n",
      "(20, 'That is, despite the expected differences in the frequency of tumor types between the two sets of carriers, heterogeneity was observed in the distribution of rs299290 genotypes in BRCA1, but not BRCA2, mutation carriers', 'tumor', 'rs299290', None)\n",
      "(21, 'Importantly, both proteasome inhibition and BRCA1 depletion increased the abundance of RHAMM (Figure S7), which is also consistent with observed RHAMM over-expression in breast cancer cell lines derived from BRCA1 mutation carriers', 'breast cancer', 'S7', None)\n",
      "(22, 'This model is consistent with reduced expression of AURKA, TPX2, and HMMR, but to a lesser extent BRCA1, with polarization and growth arrest of nonmalignant mammary epithelial cells, as measured by gene expression profiling (Figure S11A)', 'growth arrest', 'S11A', None)\n",
      "(23, 'Accordingly, BRCA1 depletion increases the clonogenic potential of mammary epithelia, while a BRCA1 S308A mutant alters embryonic stem cell differentiation', 'epithelia', 'S308A', None)\n",
      "(24, 'This work was funded by the Spanish Ministries of Health, and Science ane Innovation (CB07/02/2005; FIS 08/1120, 08/1359, 08/1635, and 09/02483; RTICCC RD06/0020/1060 and RD06/0020/0028; Transversal Action Against Cancer; the Spanish Biomedical Research Centre Networks for Epidemiology and Public Health, and Rare Diseases; and the \"Ramon y Cajal\" Young Investigator Program), the Spanish National Society of Medical Oncology (2010), the Spanish Association Against Cancer (AECC 2010), the AGAUR Catalan Government Agency (2009SGR1489 and 2009SGR293; and the Beatriu Pinos Postdoctoral Program), the Ramon Areces Foundation (XV), the \"Roses Contra el Cancer\" Foundation, the Michael Cuccione Foundation for Childhood Cancer Research, Cancer Research-UK (C490/A10119, C1287/A8874, C1287/A10118, C5047/A8385, and C8197/A10123), the National Institute for Health Research (UK), the Association for International Cancer Research (AICR-07-0454), the Ligue National Contre le Cancer (France), the Association \"Le cancer du sein, parlons-en!\", the Dutch Cancer Society (NKI 1998-1854, 2004-3088, and 2007-3756), the Fondazione Italiana per la Ricerca sul Cancro (\"Hereditary Tumors\"), the Associazione Italiana per la Ricerca sul Cancro (4017), the Italian Ministero della Salute (RFPS-2006-3-340203 and \"Progetto Tumori Femminili\"), the Italian Ministero dell\\'Universita\\' e Ricerca (RBLAO3-BETH), the Fondazione IRCCS Istituto Nazionale Tumori (INT \"5x1000\"), the Fondazione Cassa di Risparmio di Pisa (Istituto Toscano Tumori), the National Breast Cancer Foundation (Australia), the Australian National Health and Medical Research Council (145684, 288704, and 454508), the Queensland Cancer Fund, the Cancer Councils of New South Wales, Victoria, Tasmania, and South Australia, the Cancer Foundation of Western Australia, the German Cancer Aid (107054), the Center for Molecular Medicine Cologne (TV93), the National Cancer Institute (USA; CA128978 and CA122340), National Institutes of Health (RFA-CA-06-503, BCFR U01 CA69398, CA69417, CA69446, CA69467, CA69631, and CA69638), the Research Triangle Institute Informatics Support Center (RFP N02PC45022-46), the Specialized Program of Research Excellence (SPORE P50 CA83638 and CA113916), the Department of Defense Breast Cancer Research Program (05/0612), the Eileen Stein Jacoby Fund, the Breast Cancer Research Foundation, the Marianne and Robert MacDonald Foundation, the Komen Foundation, the Helsinki University Central Hospital Research Fund, the Academy of Finland (110663), the Finnish Cancer Society, the Sigrid Juselius Foundation, and the EU FP7 (223175, HEALTH-F2-2009-223175)', 'Contra el Cancer', 'C1287/A8874', None)\n",
      "(25, 'This work was funded by the Spanish Ministries of Health, and Science ane Innovation (CB07/02/2005; FIS 08/1120, 08/1359, 08/1635, and 09/02483; RTICCC RD06/0020/1060 and RD06/0020/0028; Transversal Action Against Cancer; the Spanish Biomedical Research Centre Networks for Epidemiology and Public Health, and Rare Diseases; and the \"Ramon y Cajal\" Young Investigator Program), the Spanish National Society of Medical Oncology (2010), the Spanish Association Against Cancer (AECC 2010), the AGAUR Catalan Government Agency (2009SGR1489 and 2009SGR293; and the Beatriu Pinos Postdoctoral Program), the Ramon Areces Foundation (XV), the \"Roses Contra el Cancer\" Foundation, the Michael Cuccione Foundation for Childhood Cancer Research, Cancer Research-UK (C490/A10119, C1287/A8874, C1287/A10118, C5047/A8385, and C8197/A10123), the National Institute for Health Research (UK), the Association for International Cancer Research (AICR-07-0454), the Ligue National Contre le Cancer (France), the Association \"Le cancer du sein, parlons-en!\", the Dutch Cancer Society (NKI 1998-1854, 2004-3088, and 2007-3756), the Fondazione Italiana per la Ricerca sul Cancro (\"Hereditary Tumors\"), the Associazione Italiana per la Ricerca sul Cancro (4017), the Italian Ministero della Salute (RFPS-2006-3-340203 and \"Progetto Tumori Femminili\"), the Italian Ministero dell\\'Universita\\' e Ricerca (RBLAO3-BETH), the Fondazione IRCCS Istituto Nazionale Tumori (INT \"5x1000\"), the Fondazione Cassa di Risparmio di Pisa (Istituto Toscano Tumori), the National Breast Cancer Foundation (Australia), the Australian National Health and Medical Research Council (145684, 288704, and 454508), the Queensland Cancer Fund, the Cancer Councils of New South Wales, Victoria, Tasmania, and South Australia, the Cancer Foundation of Western Australia, the German Cancer Aid (107054), the Center for Molecular Medicine Cologne (TV93), the National Cancer Institute (USA; CA128978 and CA122340), National Institutes of Health (RFA-CA-06-503, BCFR U01 CA69398, CA69417, CA69446, CA69467, CA69631, and CA69638), the Research Triangle Institute Informatics Support Center (RFP N02PC45022-46), the Specialized Program of Research Excellence (SPORE P50 CA83638 and CA113916), the Department of Defense Breast Cancer Research Program (05/0612), the Eileen Stein Jacoby Fund, the Breast Cancer Research Foundation, the Marianne and Robert MacDonald Foundation, the Komen Foundation, the Helsinki University Central Hospital Research Fund, the Academy of Finland (110663), the Finnish Cancer Society, the Sigrid Juselius Foundation, and the EU FP7 (223175, HEALTH-F2-2009-223175)', 'Childhood Cancer', 'C1287/A8874', None)\n",
      "(26, 'This work was funded by the Spanish Ministries of Health, and Science ane Innovation (CB07/02/2005; FIS 08/1120, 08/1359, 08/1635, and 09/02483; RTICCC RD06/0020/1060 and RD06/0020/0028; Transversal Action Against Cancer; the Spanish Biomedical Research Centre Networks for Epidemiology and Public Health, and Rare Diseases; and the \"Ramon y Cajal\" Young Investigator Program), the Spanish National Society of Medical Oncology (2010), the Spanish Association Against Cancer (AECC 2010), the AGAUR Catalan Government Agency (2009SGR1489 and 2009SGR293; and the Beatriu Pinos Postdoctoral Program), the Ramon Areces Foundation (XV), the \"Roses Contra el Cancer\" Foundation, the Michael Cuccione Foundation for Childhood Cancer Research, Cancer Research-UK (C490/A10119, C1287/A8874, C1287/A10118, C5047/A8385, and C8197/A10123), the National Institute for Health Research (UK), the Association for International Cancer Research (AICR-07-0454), the Ligue National Contre le Cancer (France), the Association \"Le cancer du sein, parlons-en!\", the Dutch Cancer Society (NKI 1998-1854, 2004-3088, and 2007-3756), the Fondazione Italiana per la Ricerca sul Cancro (\"Hereditary Tumors\"), the Associazione Italiana per la Ricerca sul Cancro (4017), the Italian Ministero della Salute (RFPS-2006-3-340203 and \"Progetto Tumori Femminili\"), the Italian Ministero dell\\'Universita\\' e Ricerca (RBLAO3-BETH), the Fondazione IRCCS Istituto Nazionale Tumori (INT \"5x1000\"), the Fondazione Cassa di Risparmio di Pisa (Istituto Toscano Tumori), the National Breast Cancer Foundation (Australia), the Australian National Health and Medical Research Council (145684, 288704, and 454508), the Queensland Cancer Fund, the Cancer Councils of New South Wales, Victoria, Tasmania, and South Australia, the Cancer Foundation of Western Australia, the German Cancer Aid (107054), the Center for Molecular Medicine Cologne (TV93), the National Cancer Institute (USA; CA128978 and CA122340), National Institutes of Health (RFA-CA-06-503, BCFR U01 CA69398, CA69417, CA69446, CA69467, CA69631, and CA69638), the Research Triangle Institute Informatics Support Center (RFP N02PC45022-46), the Specialized Program of Research Excellence (SPORE P50 CA83638 and CA113916), the Department of Defense Breast Cancer Research Program (05/0612), the Eileen Stein Jacoby Fund, the Breast Cancer Research Foundation, the Marianne and Robert MacDonald Foundation, the Komen Foundation, the Helsinki University Central Hospital Research Fund, the Academy of Finland (110663), the Finnish Cancer Society, the Sigrid Juselius Foundation, and the EU FP7 (223175, HEALTH-F2-2009-223175)', 'cancer', 'C1287/A8874', None)\n",
      "(27, 'This work was funded by the Spanish Ministries of Health, and Science ane Innovation (CB07/02/2005; FIS 08/1120, 08/1359, 08/1635, and 09/02483; RTICCC RD06/0020/1060 and RD06/0020/0028; Transversal Action Against Cancer; the Spanish Biomedical Research Centre Networks for Epidemiology and Public Health, and Rare Diseases; and the \"Ramon y Cajal\" Young Investigator Program), the Spanish National Society of Medical Oncology (2010), the Spanish Association Against Cancer (AECC 2010), the AGAUR Catalan Government Agency (2009SGR1489 and 2009SGR293; and the Beatriu Pinos Postdoctoral Program), the Ramon Areces Foundation (XV), the \"Roses Contra el Cancer\" Foundation, the Michael Cuccione Foundation for Childhood Cancer Research, Cancer Research-UK (C490/A10119, C1287/A8874, C1287/A10118, C5047/A8385, and C8197/A10123), the National Institute for Health Research (UK), the Association for International Cancer Research (AICR-07-0454), the Ligue National Contre le Cancer (France), the Association \"Le cancer du sein, parlons-en!\", the Dutch Cancer Society (NKI 1998-1854, 2004-3088, and 2007-3756), the Fondazione Italiana per la Ricerca sul Cancro (\"Hereditary Tumors\"), the Associazione Italiana per la Ricerca sul Cancro (4017), the Italian Ministero della Salute (RFPS-2006-3-340203 and \"Progetto Tumori Femminili\"), the Italian Ministero dell\\'Universita\\' e Ricerca (RBLAO3-BETH), the Fondazione IRCCS Istituto Nazionale Tumori (INT \"5x1000\"), the Fondazione Cassa di Risparmio di Pisa (Istituto Toscano Tumori), the National Breast Cancer Foundation (Australia), the Australian National Health and Medical Research Council (145684, 288704, and 454508), the Queensland Cancer Fund, the Cancer Councils of New South Wales, Victoria, Tasmania, and South Australia, the Cancer Foundation of Western Australia, the German Cancer Aid (107054), the Center for Molecular Medicine Cologne (TV93), the National Cancer Institute (USA; CA128978 and CA122340), National Institutes of Health (RFA-CA-06-503, BCFR U01 CA69398, CA69417, CA69446, CA69467, CA69631, and CA69638), the Research Triangle Institute Informatics Support Center (RFP N02PC45022-46), the Specialized Program of Research Excellence (SPORE P50 CA83638 and CA113916), the Department of Defense Breast Cancer Research Program (05/0612), the Eileen Stein Jacoby Fund, the Breast Cancer Research Foundation, the Marianne and Robert MacDonald Foundation, the Komen Foundation, the Helsinki University Central Hospital Research Fund, the Academy of Finland (110663), the Finnish Cancer Society, the Sigrid Juselius Foundation, and the EU FP7 (223175, HEALTH-F2-2009-223175)', 'Hereditary Tumors', 'C1287/A8874', None)\n",
      "(28, 'This work was funded by the Spanish Ministries of Health, and Science ane Innovation (CB07/02/2005; FIS 08/1120, 08/1359, 08/1635, and 09/02483; RTICCC RD06/0020/1060 and RD06/0020/0028; Transversal Action Against Cancer; the Spanish Biomedical Research Centre Networks for Epidemiology and Public Health, and Rare Diseases; and the \"Ramon y Cajal\" Young Investigator Program), the Spanish National Society of Medical Oncology (2010), the Spanish Association Against Cancer (AECC 2010), the AGAUR Catalan Government Agency (2009SGR1489 and 2009SGR293; and the Beatriu Pinos Postdoctoral Program), the Ramon Areces Foundation (XV), the \"Roses Contra el Cancer\" Foundation, the Michael Cuccione Foundation for Childhood Cancer Research, Cancer Research-UK (C490/A10119, C1287/A8874, C1287/A10118, C5047/A8385, and C8197/A10123), the National Institute for Health Research (UK), the Association for International Cancer Research (AICR-07-0454), the Ligue National Contre le Cancer (France), the Association \"Le cancer du sein, parlons-en!\", the Dutch Cancer Society (NKI 1998-1854, 2004-3088, and 2007-3756), the Fondazione Italiana per la Ricerca sul Cancro (\"Hereditary Tumors\"), the Associazione Italiana per la Ricerca sul Cancro (4017), the Italian Ministero della Salute (RFPS-2006-3-340203 and \"Progetto Tumori Femminili\"), the Italian Ministero dell\\'Universita\\' e Ricerca (RBLAO3-BETH), the Fondazione IRCCS Istituto Nazionale Tumori (INT \"5x1000\"), the Fondazione Cassa di Risparmio di Pisa (Istituto Toscano Tumori), the National Breast Cancer Foundation (Australia), the Australian National Health and Medical Research Council (145684, 288704, and 454508), the Queensland Cancer Fund, the Cancer Councils of New South Wales, Victoria, Tasmania, and South Australia, the Cancer Foundation of Western Australia, the German Cancer Aid (107054), the Center for Molecular Medicine Cologne (TV93), the National Cancer Institute (USA; CA128978 and CA122340), National Institutes of Health (RFA-CA-06-503, BCFR U01 CA69398, CA69417, CA69446, CA69467, CA69631, and CA69638), the Research Triangle Institute Informatics Support Center (RFP N02PC45022-46), the Specialized Program of Research Excellence (SPORE P50 CA83638 and CA113916), the Department of Defense Breast Cancer Research Program (05/0612), the Eileen Stein Jacoby Fund, the Breast Cancer Research Foundation, the Marianne and Robert MacDonald Foundation, the Komen Foundation, the Helsinki University Central Hospital Research Fund, the Academy of Finland (110663), the Finnish Cancer Society, the Sigrid Juselius Foundation, and the EU FP7 (223175, HEALTH-F2-2009-223175)', 'Breast Cancer', 'C1287/A8874', None)\n",
      "(29, 'This work was funded by the Spanish Ministries of Health, and Science ane Innovation (CB07/02/2005; FIS 08/1120, 08/1359, 08/1635, and 09/02483; RTICCC RD06/0020/1060 and RD06/0020/0028; Transversal Action Against Cancer; the Spanish Biomedical Research Centre Networks for Epidemiology and Public Health, and Rare Diseases; and the \"Ramon y Cajal\" Young Investigator Program), the Spanish National Society of Medical Oncology (2010), the Spanish Association Against Cancer (AECC 2010), the AGAUR Catalan Government Agency (2009SGR1489 and 2009SGR293; and the Beatriu Pinos Postdoctoral Program), the Ramon Areces Foundation (XV), the \"Roses Contra el Cancer\" Foundation, the Michael Cuccione Foundation for Childhood Cancer Research, Cancer Research-UK (C490/A10119, C1287/A8874, C1287/A10118, C5047/A8385, and C8197/A10123), the National Institute for Health Research (UK), the Association for International Cancer Research (AICR-07-0454), the Ligue National Contre le Cancer (France), the Association \"Le cancer du sein, parlons-en!\", the Dutch Cancer Society (NKI 1998-1854, 2004-3088, and 2007-3756), the Fondazione Italiana per la Ricerca sul Cancro (\"Hereditary Tumors\"), the Associazione Italiana per la Ricerca sul Cancro (4017), the Italian Ministero della Salute (RFPS-2006-3-340203 and \"Progetto Tumori Femminili\"), the Italian Ministero dell\\'Universita\\' e Ricerca (RBLAO3-BETH), the Fondazione IRCCS Istituto Nazionale Tumori (INT \"5x1000\"), the Fondazione Cassa di Risparmio di Pisa (Istituto Toscano Tumori), the National Breast Cancer Foundation (Australia), the Australian National Health and Medical Research Council (145684, 288704, and 454508), the Queensland Cancer Fund, the Cancer Councils of New South Wales, Victoria, Tasmania, and South Australia, the Cancer Foundation of Western Australia, the German Cancer Aid (107054), the Center for Molecular Medicine Cologne (TV93), the National Cancer Institute (USA; CA128978 and CA122340), National Institutes of Health (RFA-CA-06-503, BCFR U01 CA69398, CA69417, CA69446, CA69467, CA69631, and CA69638), the Research Triangle Institute Informatics Support Center (RFP N02PC45022-46), the Specialized Program of Research Excellence (SPORE P50 CA83638 and CA113916), the Department of Defense Breast Cancer Research Program (05/0612), the Eileen Stein Jacoby Fund, the Breast Cancer Research Foundation, the Marianne and Robert MacDonald Foundation, the Komen Foundation, the Helsinki University Central Hospital Research Fund, the Academy of Finland (110663), the Finnish Cancer Society, the Sigrid Juselius Foundation, and the EU FP7 (223175, HEALTH-F2-2009-223175)', 'Breast Cancer', 'C1287/A8874', None)\n",
      "(30, 'This work was funded by the Spanish Ministries of Health, and Science ane Innovation (CB07/02/2005; FIS 08/1120, 08/1359, 08/1635, and 09/02483; RTICCC RD06/0020/1060 and RD06/0020/0028; Transversal Action Against Cancer; the Spanish Biomedical Research Centre Networks for Epidemiology and Public Health, and Rare Diseases; and the \"Ramon y Cajal\" Young Investigator Program), the Spanish National Society of Medical Oncology (2010), the Spanish Association Against Cancer (AECC 2010), the AGAUR Catalan Government Agency (2009SGR1489 and 2009SGR293; and the Beatriu Pinos Postdoctoral Program), the Ramon Areces Foundation (XV), the \"Roses Contra el Cancer\" Foundation, the Michael Cuccione Foundation for Childhood Cancer Research, Cancer Research-UK (C490/A10119, C1287/A8874, C1287/A10118, C5047/A8385, and C8197/A10123), the National Institute for Health Research (UK), the Association for International Cancer Research (AICR-07-0454), the Ligue National Contre le Cancer (France), the Association \"Le cancer du sein, parlons-en!\", the Dutch Cancer Society (NKI 1998-1854, 2004-3088, and 2007-3756), the Fondazione Italiana per la Ricerca sul Cancro (\"Hereditary Tumors\"), the Associazione Italiana per la Ricerca sul Cancro (4017), the Italian Ministero della Salute (RFPS-2006-3-340203 and \"Progetto Tumori Femminili\"), the Italian Ministero dell\\'Universita\\' e Ricerca (RBLAO3-BETH), the Fondazione IRCCS Istituto Nazionale Tumori (INT \"5x1000\"), the Fondazione Cassa di Risparmio di Pisa (Istituto Toscano Tumori), the National Breast Cancer Foundation (Australia), the Australian National Health and Medical Research Council (145684, 288704, and 454508), the Queensland Cancer Fund, the Cancer Councils of New South Wales, Victoria, Tasmania, and South Australia, the Cancer Foundation of Western Australia, the German Cancer Aid (107054), the Center for Molecular Medicine Cologne (TV93), the National Cancer Institute (USA; CA128978 and CA122340), National Institutes of Health (RFA-CA-06-503, BCFR U01 CA69398, CA69417, CA69446, CA69467, CA69631, and CA69638), the Research Triangle Institute Informatics Support Center (RFP N02PC45022-46), the Specialized Program of Research Excellence (SPORE P50 CA83638 and CA113916), the Department of Defense Breast Cancer Research Program (05/0612), the Eileen Stein Jacoby Fund, the Breast Cancer Research Foundation, the Marianne and Robert MacDonald Foundation, the Komen Foundation, the Helsinki University Central Hospital Research Fund, the Academy of Finland (110663), the Finnish Cancer Society, the Sigrid Juselius Foundation, and the EU FP7 (223175, HEALTH-F2-2009-223175)', 'Breast Cancer', 'C1287/A8874', None)\n",
      "(31, 'Inconsistent association between the STK15 F31I genetic polymorphism and breast cancer risk.', 'breast cancer', 'F31I', None)\n"
     ]
    }
   ],
   "source": [
    "#store the dataframe into psql table\n",
    "import psycopg2\n",
    "import pandas as pd\n",
    "from sqlalchemy import create_engine\n",
    "\n",
    "conn_string = 'postgresql://postgres:sinkiru@localhost:5432/gdmbiobert'\n",
    "\n",
    "db = create_engine(conn_string)\n",
    "conn = db.connect()\n",
    "\n",
    "final_re_output.to_sql('dm', con=conn, if_exists='replace', index=False)\n",
    "conn = psycopg2.connect(conn_string)\n",
    "conn.autocommit = True\n",
    "cursor = conn.cursor()\n",
    "\n",
    "sql1 = '''select * from dm;'''\n",
    "cursor.execute(sql1)\n",
    "for i in cursor.fetchall():\n",
    "\tprint(i)\n",
    "\n",
    "# conn.commit()\n",
    "conn.close()\n"
   ]
  }
 ],
 "metadata": {
  "colab": {
   "provenance": []
  },
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.13"
  },
  "vscode": {
   "interpreter": {
    "hash": "6626a9a69e96ea8665bf0f933e0462e9e2599c09c430a57c21911de9c8d78cbc"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
