{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "executionInfo": {
     "elapsed": 3605,
     "status": "ok",
     "timestamp": 1679075791770,
     "user": {
      "displayName": "19PT24 - S KIRUPA",
      "userId": "12629547787028680648"
     },
     "user_tz": -330
    },
    "id": "RvaNWdrKgXjD",
    "outputId": "7fe66ab5-ecad-49ad-f8f1-6398703962bc"
   },
   "outputs": [],
   "source": [
    "#We extract the pubmed document in BioCJSON format\n",
    "import urllib3\n",
    "import json\n",
    "import csv\n",
    "import requests\n",
    "import pandas as pd\n",
    "import io\n",
    "import os\n",
    "#import nltk\n",
    "#nltk.download('punkt')\n",
    "#from nltk.tokenize import word_tokenize\n",
    "from itertools import combinations\n",
    "\n",
    "http = urllib3.PoolManager()\n",
    "\n",
    "pmcids = []\n",
    "\n",
    "\n",
    "# Read the PMC IDs from the input file\n",
    "with open(\"pmcid.txt\", \"r\") as file:\n",
    "    for line in file:\n",
    "        pmcids.append(line.strip())\n",
    "# Initialize an empty list to store the results\n",
    "results = []\n",
    "\n",
    "for pmcid in pmcids:\n",
    "    r = http.request('GET', f'https://www.ncbi.nlm.nih.gov/research/pubtator-api/publications/export/biocjson?pmcids={pmcid}')\n",
    "    data = json.loads(r.data.decode('utf-8'))\n",
    "\n",
    "    # Add the results to the list\n",
    "    results.append(data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 143
    },
    "executionInfo": {
     "elapsed": 410,
     "status": "ok",
     "timestamp": 1679075793852,
     "user": {
      "displayName": "19PT24 - S KIRUPA",
      "userId": "12629547787028680648"
     },
     "user_tz": -330
    },
    "id": "-3bUHWus1nUq",
    "outputId": "3f1c4784-c8ea-473b-d8bb-aefd779ea62b"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'\\nEvery PMC id has passages.\\nEvery passage has many {infons, offset, text, sentences, annotations, relations}.\\nHere, text is the actual text we have to annotate. \\nEvery annotations has {id, infons, text, locations}. \\nHere infons has {identifier, type} (optional: ncbi-homologene if type is gene).\\nAlso locations has {offset, length}.\\n\\nPassages:\\n    a)infons - data realted article id, author, etc..\\n    b)offset - location index\\n    c)text - whole medical data (sentence) in which medical terms (gene name or disease name) are to be annotated.\\n    d)sentences - not required here\\n    e)annotations:\\n        1)id - key index\\n        2)infons:\\n              a)identifier\\n              b)type - \"Gene\" or \"Disease\" etc.,\\n        3)text - gene name or disease name etc., (Eg: \"K-Ras\")\\n        4)locations:\\n            a)offset - location index\\n            b)length - length of text. (Eg: len(\"tumours\") = 7) \\n    f)relations\\n'"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "output = json.dumps(data, indent=4)\n",
    "#print(output)\n",
    "\n",
    "'''\n",
    "Every PMC id has passages.\n",
    "Every passage has many {infons, offset, text, sentences, annotations, relations}.\n",
    "Here, text is the actual text we have to annotate. \n",
    "Every annotations has {id, infons, text, locations}. \n",
    "Here infons has {identifier, type} (optional: ncbi-homologene if type is gene).\n",
    "Also locations has {offset, length}.\n",
    "\n",
    "Passages:\n",
    "    a)infons - data realted article id, author, etc..\n",
    "    b)offset - location index\n",
    "    c)text - whole medical data (sentence) in which medical terms (gene name or disease name) are to be annotated.\n",
    "    d)sentences - not required here\n",
    "    e)annotations:\n",
    "        1)id - key index\n",
    "        2)infons:\n",
    "              a)identifier\n",
    "              b)type - \"Gene\" or \"Disease\" etc.,\n",
    "        3)text - gene name or disease name etc., (Eg: \"K-Ras\")\n",
    "        4)locations:\n",
    "            a)offset - location index\n",
    "            b)length - length of text. (Eg: len(\"tumours\") = 7) \n",
    "    f)relations\n",
    "'''\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "id": "3EkWKX8wA1b5"
   },
   "outputs": [],
   "source": [
    "df1 = pd.DataFrame(columns=['index','sentence'])\n",
    "df2 = pd.DataFrame(columns=['index','sentence','Gene','Mutation'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "ZSMLCcRaAlL_"
   },
   "source": [
    "**Annotations for Gene and Mutation**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "id": "kbIqNKyCgkuF"
   },
   "outputs": [],
   "source": [
    "sentence_index = 0\n",
    "sentence_entities = {}\n",
    "for j in range(len(results)):\n",
    "  for i in results[j]['passages']:\n",
    "    if i['infons']['section_type'] != 'TABLE':\n",
    "      #filter the table segment\n",
    "      text = i['text']\n",
    "      # print(\"TEXT: \", text)\n",
    "      offset = i['offset']\n",
    "      # print(\"OFFSET: \", offset)    \n",
    "      annotations = i['annotations']\n",
    "      \n",
    "      annotations = sorted(annotations, key = lambda x: x['locations'][0]['offset'])\n",
    "      # print(\"ANNOTATIONS: \", annotations)\n",
    "      #Filter to only include gene-disease annotations\n",
    "      annotations = [annotation for annotation in annotations if ((annotation['infons']['type']=='Gene') or (annotation['infons']['type']=='Mutation'))]\n",
    "      #List all possible combinations of annotations\n",
    "      annots_combinations = list(combinations(annotations, 2))\n",
    "      # print(\"ANNOT COMBIMNATIONS: \", annots_combinations)\n",
    "      #Filter combinations to only include gene-disease combinations\n",
    "      annots_combinations = [annots for annots in annots_combinations if annots[0]['infons']['type'] != annots[1]['infons']['type']]\n",
    "      # print(\"ANNOT COMBINATIONS - GENE-DISEASE: \", annots_combinations)\n",
    "\n",
    "      #processing sentences\n",
    "      sentences = text.split('. ')\n",
    "      # print(sentences)\n",
    "      sentence_offset = {}\n",
    "      sentence_len = {}\n",
    "      prev_sent_offset = offset\n",
    "\n",
    "      for sentence in sentences:\n",
    "\n",
    "        sentence_offset[sentence] = prev_sent_offset\n",
    "        current_sentence_offset = prev_sent_offset\n",
    "        sentence_len[sentence] = len(sentence)\n",
    "        current_sentence_len = len(sentence)\n",
    "        prev_sent_offset += len(sentence) + 2\n",
    "        \n",
    "\n",
    "        #Point to note: Duplicate sentences for as many combinations as present. \n",
    "        for annots in annots_combinations:\n",
    "          difference = 0\n",
    "          current_sentence = sentence\n",
    "          #sort the tuple\n",
    "          annots = sorted(annots, key = lambda x: x['locations'][0]['offset'])\n",
    "\n",
    "          entity_1 = annots[0]\n",
    "          entity_2 = annots[1]\n",
    "          \n",
    "          entity_1_offset = entity_1['locations'][0]['offset']\n",
    "          entity_2_offset = entity_2['locations'][0]['offset']\n",
    "\n",
    "          entity_1_dist = entity_1_offset - current_sentence_offset\n",
    "          entity_2_dist = entity_2_offset - current_sentence_offset\n",
    "\n",
    "          if (0 <= entity_1_dist <= ((current_sentence_len - len(entity_1['text'])) + 1)) and (0 <= entity_2_dist <= ((current_sentence_len - len(entity_2['text'])) + 1)):\n",
    "            #the pair of annotations fall within the sentence\n",
    "            sentence_entities[sentence_index] = (entity_1['text'], entity_2['text'])\n",
    "\n",
    "            entity_1_type = entity_1['infons']['type']\n",
    "            entity_1_length = entity_1['locations'][0]['length']\n",
    "            temp = '@'+ entity_1_type +'$'\n",
    "            entity_1_final_off = entity_1_dist \n",
    "            current_sentence = current_sentence[:entity_1_final_off] + \"@\" + entity_1_type + \"$\" + current_sentence[(entity_1_final_off + entity_1_length):]\n",
    "            difference += (entity_1_length - len(temp))\n",
    "            entity_2_type = entity_2['infons']['type']\n",
    "            entity_2_length = entity_2['locations'][0]['length']\n",
    "            temp = '@'+ entity_2_type +'$'\n",
    "            entity_2_final_off = entity_2_dist - (difference)\n",
    "            current_sentence = current_sentence[:entity_2_final_off] + \"@\" + entity_2_type + \"$\" + current_sentence[(entity_2_final_off + entity_2_length):]\n",
    "            difference += (entity_2_length - len(temp))\n",
    "            # tsv_writer_1.writerow([sentence_index, current_sentence])\n",
    "            row1 = [sentence_index,current_sentence]\n",
    "            df1.loc[len(df1)] = row1\n",
    "            if (entity_1['infons']['type'] == 'Gene'):\n",
    "              # print('Writing...', [sentence_index, sentence, entity_1['text'], entity_2['text']])\n",
    "              # tsv_writer_2.writerow([sentence_index, sentence, entity_1['text'], entity_2['text']])\n",
    "              row2 = [sentence_index, sentence, entity_1['text'], entity_2['text']]\n",
    "              df2.loc[len(df2)] = row2\n",
    "            else:\n",
    "              # print('Writing')\n",
    "              # tsv_writer_2.writerow([sentence_index, sentence, entity_2['text'], entity_1['text']])\n",
    "              row2 = [sentence_index, sentence, entity_2['text'], entity_1['text']]\n",
    "              df2.loc[len(df2)] = row2\n",
    "            sentence_index += 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "id": "rax5aqQmrZp-"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/skirupa/Desktop/sem8/RSL-Lab/RSL Final/GDM-BioBert/project-sample\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "print(os.getcwd())\n",
    "os.chdir('../biobert-pytorch/relation-extraction')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "id": "Oe0kay_JgprA"
   },
   "outputs": [],
   "source": [
    "df2.to_csv('pub_original_sentences_GM.tsv', sep=\"\\t\", index=False)\n",
    "df1.to_csv('inputGM/test.tsv', sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "178"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(df1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "ZZSYZQAAi5db"
   },
   "source": [
    "**BioBert Training**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/skirupa/Desktop/sem8/RSL-Lab/RSL Final/GDM-BioBert/biobert-pytorch/relation-extraction\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "print(os.getcwd())\n",
    "os.chdir('..')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Download all datasets including NER/RE/QA\n",
    "#!bash ./download.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/skirupa/Desktop/sem8/RSL-Lab/RSL Final/GDM-BioBert/biobert-pytorch\n",
      "/home/skirupa/Desktop/sem8/RSL-Lab/RSL Final/GDM-BioBert/biobert-pytorch/relation-extraction\n",
      "*****  euadr  Preprocessing Start *****\n",
      "*****  euadr  Preprocessing Done *****\n",
      "*****  GAD  Preprocessing Start *****\n",
      "*****  GAD  Preprocessing Done *****\n"
     ]
    }
   ],
   "source": [
    "print(os.getcwd())\n",
    "os.chdir('relation-extraction')\n",
    "print(os.getcwd())\n",
    "#To preprocess the datasets downloaded\n",
    "!bash ./preprocess.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "executionInfo": {
     "elapsed": 8981,
     "status": "ok",
     "timestamp": 1679070353775,
     "user": {
      "displayName": "19PT24 - S KIRUPA",
      "userId": "12629547787028680648"
     },
     "user_tz": -330
    },
    "id": "jPCXvCJLkLsy",
    "outputId": "b2ffa5ef-ecb7-4bd9-dbe3-640d4f36ddcc"
   },
   "outputs": [],
   "source": [
    "#!pip install scikit-learn\n",
    "#!pip install pandas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "executionInfo": {
     "elapsed": 6,
     "status": "ok",
     "timestamp": 1679070381883,
     "user": {
      "displayName": "19PT24 - S KIRUPA",
      "userId": "12629547787028680648"
     },
     "user_tz": -330
    },
    "id": "uv7yHEmWkPk8",
    "outputId": "678d9197-a2d7-41d0-fa94-9dc9a2532ad6"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "env: SAVE_DIR=./outputGM\n",
      "env: DATA=\"GAD\"\n",
      "env: SPLIT=\"1\"\n",
      "env: DATA_DIR=./inputGM\n",
      "env: ENTITY=${DATA}-${SPLIT}\n",
      "env: MAX_LENGTH=128\n",
      "env: BATCH_SIZE=32\n",
      "env: NUM_EPOCHS=3\n",
      "env: SAVE_STEPS=1000\n",
      "env: SEED=1\n"
     ]
    }
   ],
   "source": [
    "%env SAVE_DIR=./outputGM\n",
    "%env DATA=\"GAD\"\n",
    "%env SPLIT=\"1\"\n",
    "%env DATA_DIR=./inputGM\n",
    "%env ENTITY=${DATA}-${SPLIT}\n",
    "\n",
    "%env MAX_LENGTH=128\n",
    "%env BATCH_SIZE=32\n",
    "%env NUM_EPOCHS=3\n",
    "%env SAVE_STEPS=1000\n",
    "%env SEED=1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "executionInfo": {
     "elapsed": 7095,
     "status": "ok",
     "timestamp": 1679070394106,
     "user": {
      "displayName": "19PT24 - S KIRUPA",
      "userId": "12629547787028680648"
     },
     "user_tz": -330
    },
    "id": "Nap7OsDNomOM",
    "outputId": "4ba15580-e24e-4619-f13d-fb35ad41f6ed"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "05/02/2023 11:29:02 - WARNING - __main__ -   Process rank: -1, device: cpu, n_gpu: 0, distributed training: False, 16-bits training: False\n",
      "05/02/2023 11:29:11 - INFO - __main__ -   Training/evaluation parameters TrainingArguments(output_dir='./outputGM', overwrite_output_dir=True, do_train=False, do_eval=False, do_predict=True, evaluate_during_training=False, evaluation_strategy=<EvaluationStrategy.NO: 'no'>, prediction_loss_only=False, per_device_train_batch_size=8, per_device_eval_batch_size=8, per_gpu_train_batch_size=None, per_gpu_eval_batch_size=None, gradient_accumulation_steps=1, eval_accumulation_steps=None, learning_rate=5e-05, weight_decay=0.0, adam_beta1=0.9, adam_beta2=0.999, adam_epsilon=1e-08, max_grad_norm=1.0, num_train_epochs=3.0, max_steps=-1, warmup_steps=0, logging_dir='runs/May02_11-29-02_skirupa', logging_first_step=False, logging_steps=500, save_steps=500, save_total_limit=None, no_cuda=False, seed=42, fp16=False, fp16_opt_level='O1', local_rank=-1, tpu_num_cores=None, tpu_metrics_debug=False, debug=False, dataloader_drop_last=False, eval_steps=500, dataloader_num_workers=0, past_index=-1, run_name='./outputGM', disable_tqdm=False, remove_unused_columns=True, label_names=None, load_best_model_at_end=False, metric_for_best_model=None, greater_is_better=None)\n",
      "Some weights of the model checkpoint at dmis-lab/biobert-base-cased-v1.1 were not used when initializing BertForSequenceClassification: ['cls.predictions.bias', 'cls.predictions.transform.dense.weight', 'cls.predictions.transform.dense.bias', 'cls.predictions.transform.LayerNorm.weight', 'cls.predictions.transform.LayerNorm.bias', 'cls.predictions.decoder.weight', 'cls.predictions.decoder.bias', 'cls.seq_relationship.weight', 'cls.seq_relationship.bias']\n",
      "- This IS expected if you are initializing BertForSequenceClassification from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPretraining model).\n",
      "- This IS NOT expected if you are initializing BertForSequenceClassification from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).\n",
      "Some weights of BertForSequenceClassification were not initialized from the model checkpoint at dmis-lab/biobert-base-cased-v1.1 and are newly initialized: ['classifier.weight', 'classifier.bias']\n",
      "You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.\n",
      "05/02/2023 11:29:28 - INFO - root -   *** Test ***\n",
      "100%|███████████████████████████████████████████| 15/15 [00:20<00:00,  1.33s/it]/home/skirupa/miniconda3/envs/env1/lib/python3.6/site-packages/transformers/trainer.py:1118: FutureWarning: This method is deprecated, use `Trainer.is_world_process_zero()` instead.\n",
      "  warnings.warn(\"This method is deprecated, use `Trainer.is_world_process_zero()` instead.\", FutureWarning)\n",
      "05/02/2023 11:29:58 - INFO - __main__ -   ***** Test results sst-2 *****\n",
      "100%|███████████████████████████████████████████| 15/15 [00:20<00:00,  1.37s/it]\n"
     ]
    }
   ],
   "source": [
    "!python run_re.py --task_name SST-2 --config_name bert-base-cased --model_name_or_path dmis-lab/biobert-base-cased-v1.1 \\\n",
    "        --do_predict --data_dir ${DATA_DIR} \\\n",
    "        --output_dir ${SAVE_DIR} \\\n",
    "        --overwrite_output_dir\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 328
    },
    "executionInfo": {
     "elapsed": 13,
     "status": "error",
     "timestamp": 1679064615235,
     "user": {
      "displayName": "19PT24 - S KIRUPA",
      "userId": "12629547787028680648"
     },
     "user_tz": -330
    },
    "id": "QgHoYa6Noo9w",
    "outputId": "6af921bc-1802-4a23-bb95-187243510d51"
   },
   "outputs": [],
   "source": [
    "original_sentences = pd.read_csv('pub_original_sentences_GM.tsv', sep=\"\\t\")\n",
    "predictions = pd.read_csv('outputGM/test_results.txt', sep='\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "id": "QX9DHYpyBbeC"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>sentence</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Mutation</th>\n",
       "      <th>prediction</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>Four additional K-Ras mutations (Leu19Phe (1 o...</td>\n",
       "      <td>K-Ras</td>\n",
       "      <td>Leu19Phe</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>Four additional K-Ras mutations (Leu19Phe (1 o...</td>\n",
       "      <td>K-Ras</td>\n",
       "      <td>Lys117Asn</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>Four additional K-Ras mutations (Leu19Phe (1 o...</td>\n",
       "      <td>K-Ras</td>\n",
       "      <td>Ala146Thr</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>Four additional K-Ras mutations (Leu19Phe (1 o...</td>\n",
       "      <td>K-Ras</td>\n",
       "      <td>Arg164Gln</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>Lys117Asn and Ala146Thr had phenotypes similar...</td>\n",
       "      <td>K-Ras</td>\n",
       "      <td>Lys117Asn</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>173</th>\n",
       "      <td>173</td>\n",
       "      <td>In order to reconstitute HCC1937 cells with wi...</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>S11N</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>174</th>\n",
       "      <td>174</td>\n",
       "      <td>This work was funded by the Spanish Ministries...</td>\n",
       "      <td>FIS</td>\n",
       "      <td>C1287/A8874</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>175</th>\n",
       "      <td>175</td>\n",
       "      <td>The 185delAG mutation (c.68_69delAG) in the BR...</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>185delAG</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>176</th>\n",
       "      <td>176</td>\n",
       "      <td>The 185delAG mutation (c.68_69delAG) in the BR...</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>c.68_69delAG</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>177</th>\n",
       "      <td>177</td>\n",
       "      <td>Inconsistent association between the STK15 F31...</td>\n",
       "      <td>STK15</td>\n",
       "      <td>F31I</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>178 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     index                                           sentence   Gene  \\\n",
       "0        0  Four additional K-Ras mutations (Leu19Phe (1 o...  K-Ras   \n",
       "1        1  Four additional K-Ras mutations (Leu19Phe (1 o...  K-Ras   \n",
       "2        2  Four additional K-Ras mutations (Leu19Phe (1 o...  K-Ras   \n",
       "3        3  Four additional K-Ras mutations (Leu19Phe (1 o...  K-Ras   \n",
       "4        4  Lys117Asn and Ala146Thr had phenotypes similar...  K-Ras   \n",
       "..     ...                                                ...    ...   \n",
       "173    173  In order to reconstitute HCC1937 cells with wi...  BRCA1   \n",
       "174    174  This work was funded by the Spanish Ministries...    FIS   \n",
       "175    175  The 185delAG mutation (c.68_69delAG) in the BR...  BRCA1   \n",
       "176    176  The 185delAG mutation (c.68_69delAG) in the BR...  BRCA1   \n",
       "177    177  Inconsistent association between the STK15 F31...  STK15   \n",
       "\n",
       "         Mutation  prediction  \n",
       "0        Leu19Phe         1.0  \n",
       "1       Lys117Asn         1.0  \n",
       "2       Ala146Thr         1.0  \n",
       "3       Arg164Gln         1.0  \n",
       "4       Lys117Asn         1.0  \n",
       "..            ...         ...  \n",
       "173          S11N         NaN  \n",
       "174   C1287/A8874         NaN  \n",
       "175      185delAG         NaN  \n",
       "176  c.68_69delAG         NaN  \n",
       "177          F31I         NaN  \n",
       "\n",
       "[178 rows x 5 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Merge the pub original sentences and test results (prediction of biobert model)\n",
    "\n",
    "final_re_output = pd.merge(original_sentences, predictions, on ='index', how='left')\n",
    "final_re_output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "#store the result into csv\n",
    "#os.chdir('outputGM')\n",
    "#final_re_output.to_csv('final_GM_output.tsv', sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(0, 'Four additional K-Ras mutations (Leu19Phe (1 out of 106 tumours), Lys117Asn (1 out of 106), Ala146Thr (7 out of 106) and Arg164Gln (1 out of 106)) were identified', 'K-Ras', 'Leu19Phe', 1.0)\n",
      "(1, 'Four additional K-Ras mutations (Leu19Phe (1 out of 106 tumours), Lys117Asn (1 out of 106), Ala146Thr (7 out of 106) and Arg164Gln (1 out of 106)) were identified', 'K-Ras', 'Lys117Asn', 1.0)\n",
      "(2, 'Four additional K-Ras mutations (Leu19Phe (1 out of 106 tumours), Lys117Asn (1 out of 106), Ala146Thr (7 out of 106) and Arg164Gln (1 out of 106)) were identified', 'K-Ras', 'Ala146Thr', 1.0)\n",
      "(3, 'Four additional K-Ras mutations (Leu19Phe (1 out of 106 tumours), Lys117Asn (1 out of 106), Ala146Thr (7 out of 106) and Arg164Gln (1 out of 106)) were identified', 'K-Ras', 'Arg164Gln', 1.0)\n",
      "(4, 'Lys117Asn and Ala146Thr had phenotypes similar to the hotspot mutations, whereas Leu19Phe had an attenuated phenotype and the Arg164Gln mutation was phenotypically equivalent to wt K-Ras', 'K-Ras', 'Lys117Asn', 1.0)\n",
      "(5, 'Lys117Asn and Ala146Thr had phenotypes similar to the hotspot mutations, whereas Leu19Phe had an attenuated phenotype and the Arg164Gln mutation was phenotypically equivalent to wt K-Ras', 'K-Ras', 'Ala146Thr', 1.0)\n",
      "(6, 'Lys117Asn and Ala146Thr had phenotypes similar to the hotspot mutations, whereas Leu19Phe had an attenuated phenotype and the Arg164Gln mutation was phenotypically equivalent to wt K-Ras', 'K-Ras', 'Leu19Phe', 1.0)\n",
      "(7, 'Lys117Asn and Ala146Thr had phenotypes similar to the hotspot mutations, whereas Leu19Phe had an attenuated phenotype and the Arg164Gln mutation was phenotypically equivalent to wt K-Ras', 'K-Ras', 'Arg164Gln', 1.0)\n",
      "(8, 'Ras proteins regulate signal transduction by activating a number of downstream effector proteins, including the cytoplasmic serine/threonine protein kinase B-Raf (Figure 1)', ' protein kinase B-Raf', 'serine/threonine', 0.0)\n",
      "(9, 'The most common B-Raf mutation is a T to A transversion, resulting in a valine to glutamic acid substitution (V600E), present in approximately 90% of B-Raf mutant tumours (Davies et al, 2002), which results in a 500-fold increase in in vitro kinase activity and the induction of cell proliferation and transformation (Wan et al, 2004)', 'B-Raf', 'T to A', 0.0)\n",
      "(10, 'The most common B-Raf mutation is a T to A transversion, resulting in a valine to glutamic acid substitution (V600E), present in approximately 90% of B-Raf mutant tumours (Davies et al, 2002), which results in a 500-fold increase in in vitro kinase activity and the induction of cell proliferation and transformation (Wan et al, 2004)', 'B-Raf', 'valine to glutamic acid', 1.0)\n",
      "(11, 'The most common B-Raf mutation is a T to A transversion, resulting in a valine to glutamic acid substitution (V600E), present in approximately 90% of B-Raf mutant tumours (Davies et al, 2002), which results in a 500-fold increase in in vitro kinase activity and the induction of cell proliferation and transformation (Wan et al, 2004)', 'B-Raf', 'V600E', 1.0)\n",
      "(12, 'The most common B-Raf mutation is a T to A transversion, resulting in a valine to glutamic acid substitution (V600E), present in approximately 90% of B-Raf mutant tumours (Davies et al, 2002), which results in a 500-fold increase in in vitro kinase activity and the induction of cell proliferation and transformation (Wan et al, 2004)', 'B-Raf', 'T to A', 0.0)\n",
      "(13, 'The most common B-Raf mutation is a T to A transversion, resulting in a valine to glutamic acid substitution (V600E), present in approximately 90% of B-Raf mutant tumours (Davies et al, 2002), which results in a 500-fold increase in in vitro kinase activity and the induction of cell proliferation and transformation (Wan et al, 2004)', 'B-Raf', 'valine to glutamic acid', 1.0)\n",
      "(14, 'The most common B-Raf mutation is a T to A transversion, resulting in a valine to glutamic acid substitution (V600E), present in approximately 90% of B-Raf mutant tumours (Davies et al, 2002), which results in a 500-fold increase in in vitro kinase activity and the induction of cell proliferation and transformation (Wan et al, 2004)', 'B-Raf', 'V600E', 1.0)\n",
      "(15, \"Site-directed mutagenesis (SDM) was carried out on the pEFHA wt K-Ras plasmid using the QuikChange II SDM kit (Agilent, South Queensferry, Edinburgh, UK) according to the manufacturer's instructions to generate the following K-Ras mutations: G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q, using the oligonucleotide primers described in Table 1.\", 'K-Ras', 'G12V', 0.0)\n",
      "(16, \"Site-directed mutagenesis (SDM) was carried out on the pEFHA wt K-Ras plasmid using the QuikChange II SDM kit (Agilent, South Queensferry, Edinburgh, UK) according to the manufacturer's instructions to generate the following K-Ras mutations: G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q, using the oligonucleotide primers described in Table 1.\", 'K-Ras', 'G12D', 0.0)\n",
      "(17, \"Site-directed mutagenesis (SDM) was carried out on the pEFHA wt K-Ras plasmid using the QuikChange II SDM kit (Agilent, South Queensferry, Edinburgh, UK) according to the manufacturer's instructions to generate the following K-Ras mutations: G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q, using the oligonucleotide primers described in Table 1.\", 'K-Ras', 'G13D', 0.0)\n",
      "(18, \"Site-directed mutagenesis (SDM) was carried out on the pEFHA wt K-Ras plasmid using the QuikChange II SDM kit (Agilent, South Queensferry, Edinburgh, UK) according to the manufacturer's instructions to generate the following K-Ras mutations: G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q, using the oligonucleotide primers described in Table 1.\", 'K-Ras', 'Q61H', 0.0)\n",
      "(19, \"Site-directed mutagenesis (SDM) was carried out on the pEFHA wt K-Ras plasmid using the QuikChange II SDM kit (Agilent, South Queensferry, Edinburgh, UK) according to the manufacturer's instructions to generate the following K-Ras mutations: G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q, using the oligonucleotide primers described in Table 1.\", 'K-Ras', 'L19F', 0.0)\n",
      "(20, \"Site-directed mutagenesis (SDM) was carried out on the pEFHA wt K-Ras plasmid using the QuikChange II SDM kit (Agilent, South Queensferry, Edinburgh, UK) according to the manufacturer's instructions to generate the following K-Ras mutations: G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q, using the oligonucleotide primers described in Table 1.\", 'K-Ras', 'K117N', 0.0)\n",
      "(21, \"Site-directed mutagenesis (SDM) was carried out on the pEFHA wt K-Ras plasmid using the QuikChange II SDM kit (Agilent, South Queensferry, Edinburgh, UK) according to the manufacturer's instructions to generate the following K-Ras mutations: G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q, using the oligonucleotide primers described in Table 1.\", 'K-Ras', 'A146T', 0.0)\n",
      "(22, \"Site-directed mutagenesis (SDM) was carried out on the pEFHA wt K-Ras plasmid using the QuikChange II SDM kit (Agilent, South Queensferry, Edinburgh, UK) according to the manufacturer's instructions to generate the following K-Ras mutations: G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q, using the oligonucleotide primers described in Table 1.\", 'K-Ras', 'R164Q', 0.0)\n",
      "(102, 'It is interesting to note that the codon 12 mutants were most similar to wt K-Ras, whereas the K117N and A146T mutants clustered with the activating codon 13 and codon 61 hotspot mutants', 'K-Ras', 'K117N', 1.0)\n",
      "(23, \"Site-directed mutagenesis (SDM) was carried out on the pEFHA wt K-Ras plasmid using the QuikChange II SDM kit (Agilent, South Queensferry, Edinburgh, UK) according to the manufacturer's instructions to generate the following K-Ras mutations: G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q, using the oligonucleotide primers described in Table 1.\", 'K-Ras', 'G12V', 0.0)\n",
      "(24, \"Site-directed mutagenesis (SDM) was carried out on the pEFHA wt K-Ras plasmid using the QuikChange II SDM kit (Agilent, South Queensferry, Edinburgh, UK) according to the manufacturer's instructions to generate the following K-Ras mutations: G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q, using the oligonucleotide primers described in Table 1.\", 'K-Ras', 'G12D', 0.0)\n",
      "(25, \"Site-directed mutagenesis (SDM) was carried out on the pEFHA wt K-Ras plasmid using the QuikChange II SDM kit (Agilent, South Queensferry, Edinburgh, UK) according to the manufacturer's instructions to generate the following K-Ras mutations: G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q, using the oligonucleotide primers described in Table 1.\", 'K-Ras', 'G13D', 0.0)\n",
      "(26, \"Site-directed mutagenesis (SDM) was carried out on the pEFHA wt K-Ras plasmid using the QuikChange II SDM kit (Agilent, South Queensferry, Edinburgh, UK) according to the manufacturer's instructions to generate the following K-Ras mutations: G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q, using the oligonucleotide primers described in Table 1.\", 'K-Ras', 'Q61H', 0.0)\n",
      "(27, \"Site-directed mutagenesis (SDM) was carried out on the pEFHA wt K-Ras plasmid using the QuikChange II SDM kit (Agilent, South Queensferry, Edinburgh, UK) according to the manufacturer's instructions to generate the following K-Ras mutations: G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q, using the oligonucleotide primers described in Table 1.\", 'K-Ras', 'L19F', 0.0)\n",
      "(28, \"Site-directed mutagenesis (SDM) was carried out on the pEFHA wt K-Ras plasmid using the QuikChange II SDM kit (Agilent, South Queensferry, Edinburgh, UK) according to the manufacturer's instructions to generate the following K-Ras mutations: G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q, using the oligonucleotide primers described in Table 1.\", 'K-Ras', 'K117N', 0.0)\n",
      "(29, \"Site-directed mutagenesis (SDM) was carried out on the pEFHA wt K-Ras plasmid using the QuikChange II SDM kit (Agilent, South Queensferry, Edinburgh, UK) according to the manufacturer's instructions to generate the following K-Ras mutations: G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q, using the oligonucleotide primers described in Table 1.\", 'K-Ras', 'A146T', 0.0)\n",
      "(30, \"Site-directed mutagenesis (SDM) was carried out on the pEFHA wt K-Ras plasmid using the QuikChange II SDM kit (Agilent, South Queensferry, Edinburgh, UK) according to the manufacturer's instructions to generate the following K-Ras mutations: G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q, using the oligonucleotide primers described in Table 1.\", 'K-Ras', 'R164Q', 1.0)\n",
      "(31, 'NIH3T3 cells were seeded at 2.5 x 105 cells in 35 mm culture dishes and allowed to adhere overnight before transfection with 250 ng of either pEF.6, pEFHA (wt K-Ras) or plasmids containing one of the following K-Ras mutants: G12V, G12D, G13D, Q61H, L19F, K117N, A146T or R164Q', 'K-Ras', 'G12V', 0.0)\n",
      "(32, 'NIH3T3 cells were seeded at 2.5 x 105 cells in 35 mm culture dishes and allowed to adhere overnight before transfection with 250 ng of either pEF.6, pEFHA (wt K-Ras) or plasmids containing one of the following K-Ras mutants: G12V, G12D, G13D, Q61H, L19F, K117N, A146T or R164Q', 'K-Ras', 'G12D', 0.0)\n",
      "(33, 'NIH3T3 cells were seeded at 2.5 x 105 cells in 35 mm culture dishes and allowed to adhere overnight before transfection with 250 ng of either pEF.6, pEFHA (wt K-Ras) or plasmids containing one of the following K-Ras mutants: G12V, G12D, G13D, Q61H, L19F, K117N, A146T or R164Q', 'K-Ras', 'G13D', 0.0)\n",
      "(34, 'NIH3T3 cells were seeded at 2.5 x 105 cells in 35 mm culture dishes and allowed to adhere overnight before transfection with 250 ng of either pEF.6, pEFHA (wt K-Ras) or plasmids containing one of the following K-Ras mutants: G12V, G12D, G13D, Q61H, L19F, K117N, A146T or R164Q', 'K-Ras', 'Q61H', 0.0)\n",
      "(35, 'NIH3T3 cells were seeded at 2.5 x 105 cells in 35 mm culture dishes and allowed to adhere overnight before transfection with 250 ng of either pEF.6, pEFHA (wt K-Ras) or plasmids containing one of the following K-Ras mutants: G12V, G12D, G13D, Q61H, L19F, K117N, A146T or R164Q', 'K-Ras', 'L19F', 0.0)\n",
      "(36, 'NIH3T3 cells were seeded at 2.5 x 105 cells in 35 mm culture dishes and allowed to adhere overnight before transfection with 250 ng of either pEF.6, pEFHA (wt K-Ras) or plasmids containing one of the following K-Ras mutants: G12V, G12D, G13D, Q61H, L19F, K117N, A146T or R164Q', 'K-Ras', 'K117N', 0.0)\n",
      "(37, 'NIH3T3 cells were seeded at 2.5 x 105 cells in 35 mm culture dishes and allowed to adhere overnight before transfection with 250 ng of either pEF.6, pEFHA (wt K-Ras) or plasmids containing one of the following K-Ras mutants: G12V, G12D, G13D, Q61H, L19F, K117N, A146T or R164Q', 'K-Ras', 'A146T', 0.0)\n",
      "(38, 'NIH3T3 cells were seeded at 2.5 x 105 cells in 35 mm culture dishes and allowed to adhere overnight before transfection with 250 ng of either pEF.6, pEFHA (wt K-Ras) or plasmids containing one of the following K-Ras mutants: G12V, G12D, G13D, Q61H, L19F, K117N, A146T or R164Q', 'K-Ras', 'R164Q', 1.0)\n",
      "(39, 'NIH3T3 cells were seeded at 2.5 x 105 cells in 35 mm culture dishes and allowed to adhere overnight before transfection with 250 ng of either pEF.6, pEFHA (wt K-Ras) or plasmids containing one of the following K-Ras mutants: G12V, G12D, G13D, Q61H, L19F, K117N, A146T or R164Q', 'K-Ras', 'G12V', 1.0)\n",
      "(40, 'NIH3T3 cells were seeded at 2.5 x 105 cells in 35 mm culture dishes and allowed to adhere overnight before transfection with 250 ng of either pEF.6, pEFHA (wt K-Ras) or plasmids containing one of the following K-Ras mutants: G12V, G12D, G13D, Q61H, L19F, K117N, A146T or R164Q', 'K-Ras', 'G12D', 1.0)\n",
      "(41, 'NIH3T3 cells were seeded at 2.5 x 105 cells in 35 mm culture dishes and allowed to adhere overnight before transfection with 250 ng of either pEF.6, pEFHA (wt K-Ras) or plasmids containing one of the following K-Ras mutants: G12V, G12D, G13D, Q61H, L19F, K117N, A146T or R164Q', 'K-Ras', 'G13D', 1.0)\n",
      "(42, 'NIH3T3 cells were seeded at 2.5 x 105 cells in 35 mm culture dishes and allowed to adhere overnight before transfection with 250 ng of either pEF.6, pEFHA (wt K-Ras) or plasmids containing one of the following K-Ras mutants: G12V, G12D, G13D, Q61H, L19F, K117N, A146T or R164Q', 'K-Ras', 'Q61H', 0.0)\n",
      "(43, 'NIH3T3 cells were seeded at 2.5 x 105 cells in 35 mm culture dishes and allowed to adhere overnight before transfection with 250 ng of either pEF.6, pEFHA (wt K-Ras) or plasmids containing one of the following K-Ras mutants: G12V, G12D, G13D, Q61H, L19F, K117N, A146T or R164Q', 'K-Ras', 'L19F', 1.0)\n",
      "(44, 'NIH3T3 cells were seeded at 2.5 x 105 cells in 35 mm culture dishes and allowed to adhere overnight before transfection with 250 ng of either pEF.6, pEFHA (wt K-Ras) or plasmids containing one of the following K-Ras mutants: G12V, G12D, G13D, Q61H, L19F, K117N, A146T or R164Q', 'K-Ras', 'K117N', 1.0)\n",
      "(45, 'NIH3T3 cells were seeded at 2.5 x 105 cells in 35 mm culture dishes and allowed to adhere overnight before transfection with 250 ng of either pEF.6, pEFHA (wt K-Ras) or plasmids containing one of the following K-Ras mutants: G12V, G12D, G13D, Q61H, L19F, K117N, A146T or R164Q', 'K-Ras', 'A146T', 1.0)\n",
      "(46, 'NIH3T3 cells were seeded at 2.5 x 105 cells in 35 mm culture dishes and allowed to adhere overnight before transfection with 250 ng of either pEF.6, pEFHA (wt K-Ras) or plasmids containing one of the following K-Ras mutants: G12V, G12D, G13D, Q61H, L19F, K117N, A146T or R164Q', 'K-Ras', 'R164Q', 1.0)\n",
      "(47, '500 mug of cell lysate from each sample was incubated with 10 mug of Raf-1 RBD agarose at 4 C for 30 min with gentle rocking', 'Raf-1', 'C for 30', 1.0)\n",
      "(48, 'NIH3T3 cells were plated at 2.5 x 105 cells per well in a 6-well plate and, following overnight attachment, transfected with 250 ng of pEF, wt K-Ras or one of the K-Ras mutants: G12V, G12D, G12C, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'G12V', 1.0)\n",
      "(49, 'NIH3T3 cells were plated at 2.5 x 105 cells per well in a 6-well plate and, following overnight attachment, transfected with 250 ng of pEF, wt K-Ras or one of the K-Ras mutants: G12V, G12D, G12C, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'G12D', 1.0)\n",
      "(50, 'NIH3T3 cells were plated at 2.5 x 105 cells per well in a 6-well plate and, following overnight attachment, transfected with 250 ng of pEF, wt K-Ras or one of the K-Ras mutants: G12V, G12D, G12C, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'G12C', 1.0)\n",
      "(51, 'NIH3T3 cells were plated at 2.5 x 105 cells per well in a 6-well plate and, following overnight attachment, transfected with 250 ng of pEF, wt K-Ras or one of the K-Ras mutants: G12V, G12D, G12C, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'G13D', 1.0)\n",
      "(52, 'NIH3T3 cells were plated at 2.5 x 105 cells per well in a 6-well plate and, following overnight attachment, transfected with 250 ng of pEF, wt K-Ras or one of the K-Ras mutants: G12V, G12D, G12C, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'Q61H', 0.0)\n",
      "(53, 'NIH3T3 cells were plated at 2.5 x 105 cells per well in a 6-well plate and, following overnight attachment, transfected with 250 ng of pEF, wt K-Ras or one of the K-Ras mutants: G12V, G12D, G12C, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'L19F', 1.0)\n",
      "(54, 'NIH3T3 cells were plated at 2.5 x 105 cells per well in a 6-well plate and, following overnight attachment, transfected with 250 ng of pEF, wt K-Ras or one of the K-Ras mutants: G12V, G12D, G12C, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'K117N', 1.0)\n",
      "(55, 'NIH3T3 cells were plated at 2.5 x 105 cells per well in a 6-well plate and, following overnight attachment, transfected with 250 ng of pEF, wt K-Ras or one of the K-Ras mutants: G12V, G12D, G12C, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'A146T', 1.0)\n",
      "(56, 'NIH3T3 cells were plated at 2.5 x 105 cells per well in a 6-well plate and, following overnight attachment, transfected with 250 ng of pEF, wt K-Ras or one of the K-Ras mutants: G12V, G12D, G12C, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'R164Q', 1.0)\n",
      "(57, 'NIH3T3 cells were plated at 2.5 x 105 cells per well in a 6-well plate and, following overnight attachment, transfected with 250 ng of pEF, wt K-Ras or one of the K-Ras mutants: G12V, G12D, G12C, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'G12V', 1.0)\n",
      "(58, 'NIH3T3 cells were plated at 2.5 x 105 cells per well in a 6-well plate and, following overnight attachment, transfected with 250 ng of pEF, wt K-Ras or one of the K-Ras mutants: G12V, G12D, G12C, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'G12D', 1.0)\n",
      "(59, 'NIH3T3 cells were plated at 2.5 x 105 cells per well in a 6-well plate and, following overnight attachment, transfected with 250 ng of pEF, wt K-Ras or one of the K-Ras mutants: G12V, G12D, G12C, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'G12C', 1.0)\n",
      "(60, 'NIH3T3 cells were plated at 2.5 x 105 cells per well in a 6-well plate and, following overnight attachment, transfected with 250 ng of pEF, wt K-Ras or one of the K-Ras mutants: G12V, G12D, G12C, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'G13D', 1.0)\n",
      "(61, 'NIH3T3 cells were plated at 2.5 x 105 cells per well in a 6-well plate and, following overnight attachment, transfected with 250 ng of pEF, wt K-Ras or one of the K-Ras mutants: G12V, G12D, G12C, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'Q61H', 0.0)\n",
      "(62, 'NIH3T3 cells were plated at 2.5 x 105 cells per well in a 6-well plate and, following overnight attachment, transfected with 250 ng of pEF, wt K-Ras or one of the K-Ras mutants: G12V, G12D, G12C, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'L19F', 1.0)\n",
      "(63, 'NIH3T3 cells were plated at 2.5 x 105 cells per well in a 6-well plate and, following overnight attachment, transfected with 250 ng of pEF, wt K-Ras or one of the K-Ras mutants: G12V, G12D, G12C, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'K117N', 1.0)\n",
      "(64, 'NIH3T3 cells were plated at 2.5 x 105 cells per well in a 6-well plate and, following overnight attachment, transfected with 250 ng of pEF, wt K-Ras or one of the K-Ras mutants: G12V, G12D, G12C, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'A146T', 1.0)\n",
      "(65, 'NIH3T3 cells were plated at 2.5 x 105 cells per well in a 6-well plate and, following overnight attachment, transfected with 250 ng of pEF, wt K-Ras or one of the K-Ras mutants: G12V, G12D, G12C, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'R164Q', 1.0)\n",
      "(66, 'Meanwhile, 1 mul chromosome 12p BAC RP11-707G18 (Red) (BlueGnome), 1 mul chromosome 12q BAC RP11-89H19 (Green) (BlueGnome) and 9 mul LSI-WCP hybridisation buffer (Vysis) was mixed together for each slide and pipetted onto a 22 x 22 mm coverslip which was placed onto a region of the slide containing tumour tissue and placed onto a heat block at 75 C for 5 min to co-denature the probes and tumour tissue', 'RP11', 'C for 5', 1.0)\n",
      "(67, 'Meanwhile, 1 mul chromosome 12p BAC RP11-707G18 (Red) (BlueGnome), 1 mul chromosome 12q BAC RP11-89H19 (Green) (BlueGnome) and 9 mul LSI-WCP hybridisation buffer (Vysis) was mixed together for each slide and pipetted onto a 22 x 22 mm coverslip which was placed onto a region of the slide containing tumour tissue and placed onto a heat block at 75 C for 5 min to co-denature the probes and tumour tissue', 'RP11', 'C for 5', 1.0)\n",
      "(68, 'Mutations at codon 19 (G57T, Leu19Phe) and in B-Raf (V600E) were found in single tumours (Table 2).', 'B-Raf', 'G57T', 1.0)\n",
      "(69, 'Mutations at codon 19 (G57T, Leu19Phe) and in B-Raf (V600E) were found in single tumours (Table 2).', 'B-Raf', 'Leu19Phe', 0.0)\n",
      "(70, 'Mutations at codon 19 (G57T, Leu19Phe) and in B-Raf (V600E) were found in single tumours (Table 2).', 'B-Raf', 'V600E', 0.0)\n",
      "(71, 'Together, these K-Ras mutations and the B-Raf V600E mutation increase overall Ras pathway mutation frequency from 27.4 to 37.8%', 'K-Ras', 'V600E', 1.0)\n",
      "(72, 'Together, these K-Ras mutations and the B-Raf V600E mutation increase overall Ras pathway mutation frequency from 27.4 to 37.8%', 'B-Raf', 'V600E', 1.0)\n",
      "(73, 'To compare the transformation potential of the K-Ras mutants, therefore, NIH3T3 cells were transiently transfected with plasmids expressing wt K-Ras and the K-Ras mutations G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'G12V', 1.0)\n",
      "(74, 'To compare the transformation potential of the K-Ras mutants, therefore, NIH3T3 cells were transiently transfected with plasmids expressing wt K-Ras and the K-Ras mutations G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'G12D', 1.0)\n",
      "(75, 'To compare the transformation potential of the K-Ras mutants, therefore, NIH3T3 cells were transiently transfected with plasmids expressing wt K-Ras and the K-Ras mutations G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'G13D', 0.0)\n",
      "(76, 'To compare the transformation potential of the K-Ras mutants, therefore, NIH3T3 cells were transiently transfected with plasmids expressing wt K-Ras and the K-Ras mutations G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'Q61H', 1.0)\n",
      "(77, 'To compare the transformation potential of the K-Ras mutants, therefore, NIH3T3 cells were transiently transfected with plasmids expressing wt K-Ras and the K-Ras mutations G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'L19F', 1.0)\n",
      "(78, 'To compare the transformation potential of the K-Ras mutants, therefore, NIH3T3 cells were transiently transfected with plasmids expressing wt K-Ras and the K-Ras mutations G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'K117N', 0.0)\n",
      "(79, 'To compare the transformation potential of the K-Ras mutants, therefore, NIH3T3 cells were transiently transfected with plasmids expressing wt K-Ras and the K-Ras mutations G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'A146T', 0.0)\n",
      "(80, 'To compare the transformation potential of the K-Ras mutants, therefore, NIH3T3 cells were transiently transfected with plasmids expressing wt K-Ras and the K-Ras mutations G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'R164Q', 0.0)\n",
      "(81, 'To compare the transformation potential of the K-Ras mutants, therefore, NIH3T3 cells were transiently transfected with plasmids expressing wt K-Ras and the K-Ras mutations G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'G12V', 1.0)\n",
      "(82, 'To compare the transformation potential of the K-Ras mutants, therefore, NIH3T3 cells were transiently transfected with plasmids expressing wt K-Ras and the K-Ras mutations G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'G12D', 1.0)\n",
      "(83, 'To compare the transformation potential of the K-Ras mutants, therefore, NIH3T3 cells were transiently transfected with plasmids expressing wt K-Ras and the K-Ras mutations G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'G13D', 0.0)\n",
      "(84, 'To compare the transformation potential of the K-Ras mutants, therefore, NIH3T3 cells were transiently transfected with plasmids expressing wt K-Ras and the K-Ras mutations G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'Q61H', 1.0)\n",
      "(85, 'To compare the transformation potential of the K-Ras mutants, therefore, NIH3T3 cells were transiently transfected with plasmids expressing wt K-Ras and the K-Ras mutations G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'L19F', 1.0)\n",
      "(86, 'To compare the transformation potential of the K-Ras mutants, therefore, NIH3T3 cells were transiently transfected with plasmids expressing wt K-Ras and the K-Ras mutations G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'K117N', 0.0)\n",
      "(87, 'To compare the transformation potential of the K-Ras mutants, therefore, NIH3T3 cells were transiently transfected with plasmids expressing wt K-Ras and the K-Ras mutations G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'A146T', 0.0)\n",
      "(88, 'To compare the transformation potential of the K-Ras mutants, therefore, NIH3T3 cells were transiently transfected with plasmids expressing wt K-Ras and the K-Ras mutations G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'R164Q', 0.0)\n",
      "(89, 'To compare the transformation potential of the K-Ras mutants, therefore, NIH3T3 cells were transiently transfected with plasmids expressing wt K-Ras and the K-Ras mutations G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'G12V', 1.0)\n",
      "(90, 'To compare the transformation potential of the K-Ras mutants, therefore, NIH3T3 cells were transiently transfected with plasmids expressing wt K-Ras and the K-Ras mutations G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'G12D', 1.0)\n",
      "(91, 'To compare the transformation potential of the K-Ras mutants, therefore, NIH3T3 cells were transiently transfected with plasmids expressing wt K-Ras and the K-Ras mutations G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'G13D', 1.0)\n",
      "(92, 'To compare the transformation potential of the K-Ras mutants, therefore, NIH3T3 cells were transiently transfected with plasmids expressing wt K-Ras and the K-Ras mutations G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'Q61H', 1.0)\n",
      "(93, 'To compare the transformation potential of the K-Ras mutants, therefore, NIH3T3 cells were transiently transfected with plasmids expressing wt K-Ras and the K-Ras mutations G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'L19F', 1.0)\n",
      "(94, 'To compare the transformation potential of the K-Ras mutants, therefore, NIH3T3 cells were transiently transfected with plasmids expressing wt K-Ras and the K-Ras mutations G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'K117N', 1.0)\n",
      "(95, 'To compare the transformation potential of the K-Ras mutants, therefore, NIH3T3 cells were transiently transfected with plasmids expressing wt K-Ras and the K-Ras mutations G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'A146T', 0.0)\n",
      "(96, 'To compare the transformation potential of the K-Ras mutants, therefore, NIH3T3 cells were transiently transfected with plasmids expressing wt K-Ras and the K-Ras mutations G12V, G12D, G13D, Q61H, L19F, K117N, A146T and R164Q', 'K-Ras', 'R164Q', 0.0)\n",
      "(97, 'Figure 4 illustrates the results of our analysis where, consistent with our focus formation experiments, the K-Ras G12V and L19F, K117N and A146T mutations are clearly in the active GTP-bound conformation', 'K-Ras', 'G12V', 1.0)\n",
      "(98, 'Figure 4 illustrates the results of our analysis where, consistent with our focus formation experiments, the K-Ras G12V and L19F, K117N and A146T mutations are clearly in the active GTP-bound conformation', 'K-Ras', 'L19F', 1.0)\n",
      "(99, 'Figure 4 illustrates the results of our analysis where, consistent with our focus formation experiments, the K-Ras G12V and L19F, K117N and A146T mutations are clearly in the active GTP-bound conformation', 'K-Ras', 'K117N', 1.0)\n",
      "(100, 'Figure 4 illustrates the results of our analysis where, consistent with our focus formation experiments, the K-Ras G12V and L19F, K117N and A146T mutations are clearly in the active GTP-bound conformation', 'K-Ras', 'A146T', 1.0)\n",
      "(101, 'In contrast, the R164Q mutation, like wt K-Ras, was not GTP-bound.', 'K-Ras', 'R164Q', 1.0)\n",
      "(103, 'It is interesting to note that the codon 12 mutants were most similar to wt K-Ras, whereas the K117N and A146T mutants clustered with the activating codon 13 and codon 61 hotspot mutants', 'K-Ras', 'A146T', 1.0)\n",
      "(104, \"The presence of R164Q in cluster 2, rather than in cluster 1 with wt K-Ras was therefore initially surprising, but suggests that the R164Q mutation also has an 'activating' phenotype, albeit attenuated relative to the other mutations studied.\", 'K-Ras', 'R164Q', 1.0)\n",
      "(105, \"The presence of R164Q in cluster 2, rather than in cluster 1 with wt K-Ras was therefore initially surprising, but suggests that the R164Q mutation also has an 'activating' phenotype, albeit attenuated relative to the other mutations studied.\", 'K-Ras', 'R164Q', 1.0)\n",
      "(106, 'However, there were examples of genes that were induced by all of the mutants studied, including the protein tyrosine phosphatase Ptpre and the Rho GTPase activating protein Arhgap6, and multiple examples of genes induced or repressed by all of the mutants with the exception of R164Q', 'Ptpre', 'R164Q', 1.0)\n",
      "(107, 'However, there were examples of genes that were induced by all of the mutants studied, including the protein tyrosine phosphatase Ptpre and the Rho GTPase activating protein Arhgap6, and multiple examples of genes induced or repressed by all of the mutants with the exception of R164Q', 'Arhgap6', 'R164Q', 1.0)\n",
      "(108, 'RFLP analysis of B-Raf V599E mutation', 'Raf', 'V599E', 1.0)\n",
      "(109, 'The B-Raf V599E mutation was detected using a novel RFLP assay, where the presence of the mutation destroyed a recognition site for the restriction enzyme TspRI', 'Raf', 'V599E', 1.0)\n",
      "(110, 'Samples without the B-Raf V599E mutation were digested to produce bands of 124, 78 and 22 bp, whereas samples homozygous for the V599E mutation produced bands of 202 and 22 bp', 'B-Raf', 'V599E', 1.0)\n",
      "(111, 'Samples without the B-Raf V599E mutation were digested to produce bands of 124, 78 and 22 bp, whereas samples homozygous for the V599E mutation produced bands of 202 and 22 bp', 'B-Raf', 'V599E', 1.0)\n",
      "(112, 'Mutation analysis in Costello syndrome: functional and structural characterization of the HRAS p.Lys117Arg mutation', 'HRAS', 'p.Lys117Arg', 1.0)\n",
      "(113, '(B) The transforming potential of L19F, K117N, A146T and R164Q mutations were compared with the hotspot mutation G12V and wt K-Ras', 'K-Ras', 'L19F', 1.0)\n",
      "(114, '(B) The transforming potential of L19F, K117N, A146T and R164Q mutations were compared with the hotspot mutation G12V and wt K-Ras', 'K-Ras', 'K117N', 1.0)\n",
      "(115, '(B) The transforming potential of L19F, K117N, A146T and R164Q mutations were compared with the hotspot mutation G12V and wt K-Ras', 'K-Ras', 'A146T', 1.0)\n",
      "(116, '(B) The transforming potential of L19F, K117N, A146T and R164Q mutations were compared with the hotspot mutation G12V and wt K-Ras', 'K-Ras', 'R164Q', 1.0)\n",
      "(117, '(B) The transforming potential of L19F, K117N, A146T and R164Q mutations were compared with the hotspot mutation G12V and wt K-Ras', 'K-Ras', 'G12V', 1.0)\n",
      "(118, 'The K-Ras G12V construct was included as a positive control for K-Ras protein in the active GTP-bound conformation.', 'K-Ras', 'G12V', 1.0)\n",
      "(119, 'The K-Ras G12V construct was included as a positive control for K-Ras protein in the active GTP-bound conformation.', 'K-Ras', 'G12V', 1.0)\n",
      "(120, 'The mitotic serine/threonine kinase Aurora2/AIK is regulated by phosphorylation and degradation', 'Aurora2', 'serine/threonine', None)\n",
      "(121, 'The mitotic serine/threonine kinase Aurora2/AIK is regulated by phosphorylation and degradation', 'AIK', 'serine/threonine', None)\n",
      "(122, 'Following a pilot study in Italy and Spain, analysis of carriers (n = 11,609) collected through 24 study groups participating in the Consortium of Investigators of Modifiers of BRCA1/2 (CIMBA) detected significant modification of breast cancer risk by HMMR rs299290 variant among BRCA1, but not BRCA2, mutation carriers: BRCA1 mutation carriers n = 7,584, Cox proportional-hazards regression model, hazard ratio (HR) = 1.08 (95% confidence interval (CI) 1.02-1.13), ptrend = 0.004 (p 2df = 0.014), in the same direction as originally detected in Ashkenazi Jewish populations; BRCA2 mutation carriers n = 3,965, HR = 1.03 (95% CI 0.96-1.10), ptrend = 0.42 (p 2df = 0.67)', 'BRCA1/2', 'rs299290', None)\n",
      "(123, 'Following a pilot study in Italy and Spain, analysis of carriers (n = 11,609) collected through 24 study groups participating in the Consortium of Investigators of Modifiers of BRCA1/2 (CIMBA) detected significant modification of breast cancer risk by HMMR rs299290 variant among BRCA1, but not BRCA2, mutation carriers: BRCA1 mutation carriers n = 7,584, Cox proportional-hazards regression model, hazard ratio (HR) = 1.08 (95% confidence interval (CI) 1.02-1.13), ptrend = 0.004 (p 2df = 0.014), in the same direction as originally detected in Ashkenazi Jewish populations; BRCA2 mutation carriers n = 3,965, HR = 1.03 (95% CI 0.96-1.10), ptrend = 0.42 (p 2df = 0.67)', 'HMMR', 'rs299290', None)\n",
      "(124, 'Following a pilot study in Italy and Spain, analysis of carriers (n = 11,609) collected through 24 study groups participating in the Consortium of Investigators of Modifiers of BRCA1/2 (CIMBA) detected significant modification of breast cancer risk by HMMR rs299290 variant among BRCA1, but not BRCA2, mutation carriers: BRCA1 mutation carriers n = 7,584, Cox proportional-hazards regression model, hazard ratio (HR) = 1.08 (95% confidence interval (CI) 1.02-1.13), ptrend = 0.004 (p 2df = 0.014), in the same direction as originally detected in Ashkenazi Jewish populations; BRCA2 mutation carriers n = 3,965, HR = 1.03 (95% CI 0.96-1.10), ptrend = 0.42 (p 2df = 0.67)', 'BRCA1', 'rs299290', None)\n",
      "(125, 'Following a pilot study in Italy and Spain, analysis of carriers (n = 11,609) collected through 24 study groups participating in the Consortium of Investigators of Modifiers of BRCA1/2 (CIMBA) detected significant modification of breast cancer risk by HMMR rs299290 variant among BRCA1, but not BRCA2, mutation carriers: BRCA1 mutation carriers n = 7,584, Cox proportional-hazards regression model, hazard ratio (HR) = 1.08 (95% confidence interval (CI) 1.02-1.13), ptrend = 0.004 (p 2df = 0.014), in the same direction as originally detected in Ashkenazi Jewish populations; BRCA2 mutation carriers n = 3,965, HR = 1.03 (95% CI 0.96-1.10), ptrend = 0.42 (p 2df = 0.67)', 'BRCA2', 'rs299290', None)\n",
      "(126, 'Following a pilot study in Italy and Spain, analysis of carriers (n = 11,609) collected through 24 study groups participating in the Consortium of Investigators of Modifiers of BRCA1/2 (CIMBA) detected significant modification of breast cancer risk by HMMR rs299290 variant among BRCA1, but not BRCA2, mutation carriers: BRCA1 mutation carriers n = 7,584, Cox proportional-hazards regression model, hazard ratio (HR) = 1.08 (95% confidence interval (CI) 1.02-1.13), ptrend = 0.004 (p 2df = 0.014), in the same direction as originally detected in Ashkenazi Jewish populations; BRCA2 mutation carriers n = 3,965, HR = 1.03 (95% CI 0.96-1.10), ptrend = 0.42 (p 2df = 0.67)', 'BRCA1', 'rs299290', None)\n",
      "(156, 'In MCF10A cells, AURKA abundance and activity determined total RHAMM as well as pT703-RHAMM levels (Figure S10A)', 'AURKA', 'S10A', None)\n",
      "(157, 'In MCF10A cells, AURKA abundance and activity determined total RHAMM as well as pT703-RHAMM levels (Figure S10A)', 'RHAMM', 'S10A', None)\n",
      "(127, 'Following a pilot study in Italy and Spain, analysis of carriers (n = 11,609) collected through 24 study groups participating in the Consortium of Investigators of Modifiers of BRCA1/2 (CIMBA) detected significant modification of breast cancer risk by HMMR rs299290 variant among BRCA1, but not BRCA2, mutation carriers: BRCA1 mutation carriers n = 7,584, Cox proportional-hazards regression model, hazard ratio (HR) = 1.08 (95% confidence interval (CI) 1.02-1.13), ptrend = 0.004 (p 2df = 0.014), in the same direction as originally detected in Ashkenazi Jewish populations; BRCA2 mutation carriers n = 3,965, HR = 1.03 (95% CI 0.96-1.10), ptrend = 0.42 (p 2df = 0.67)', 'Cox', 'rs299290', None)\n",
      "(128, 'Following a pilot study in Italy and Spain, analysis of carriers (n = 11,609) collected through 24 study groups participating in the Consortium of Investigators of Modifiers of BRCA1/2 (CIMBA) detected significant modification of breast cancer risk by HMMR rs299290 variant among BRCA1, but not BRCA2, mutation carriers: BRCA1 mutation carriers n = 7,584, Cox proportional-hazards regression model, hazard ratio (HR) = 1.08 (95% confidence interval (CI) 1.02-1.13), ptrend = 0.004 (p 2df = 0.014), in the same direction as originally detected in Ashkenazi Jewish populations; BRCA2 mutation carriers n = 3,965, HR = 1.03 (95% CI 0.96-1.10), ptrend = 0.42 (p 2df = 0.67)', 'BRCA2', 'rs299290', None)\n",
      "(129, 'Effect of HMMR rs299290 variation on breast cancer risk among BRCA1 and BRCA2 mutation carriers.', 'HMMR', 'rs299290', None)\n",
      "(130, 'Effect of HMMR rs299290 variation on breast cancer risk among BRCA1 and BRCA2 mutation carriers.', 'BRCA1', 'rs299290', None)\n",
      "(131, 'Effect of HMMR rs299290 variation on breast cancer risk among BRCA1 and BRCA2 mutation carriers.', 'BRCA2', 'rs299290', None)\n",
      "(132, 'Forrest plots show HRs and 95% CIs of the additive model (rs299290 C allele) for all participating centers ordered by sample size (n>30) of BRCA1 mutation carriers (left panel, wHR per study center are shown in Table S2; right panel, effect on BRCA2 mutation carriers)', 'BRCA1', 'rs299290', None)\n",
      "(133, 'Forrest plots show HRs and 95% CIs of the additive model (rs299290 C allele) for all participating centers ordered by sample size (n>30) of BRCA1 mutation carriers (left panel, wHR per study center are shown in Table S2; right panel, effect on BRCA2 mutation carriers)', 'BRCA2', 'rs299290', None)\n",
      "(134, 'Analysis of public gene expression datasets suggests that the rs299290 risk allele is associated with HMMR germline over-expression (see also Table S3)', 'HMMR', 'rs299290', None)\n",
      "(135, 'Since rs299284 is at the fourth base position of HMMR exon 5, we evaluated the potential alteration of the splicing pattern of this exon or the ratio of the alternative exon 4', 'HMMR', 'rs299284', None)\n",
      "(136, 'In data provided by several CIMBA groups (Text S1), no ER-positive tumors were observed among rare rs299290 homozygotes in BRCA1 mutation carriers (p interaction = 0.006), whereas this bias was not observed in BRCA2 mutation carriers (p interaction = 0.95) (Table S4)', 'ER', 'rs299290', None)\n",
      "(137, 'In data provided by several CIMBA groups (Text S1), no ER-positive tumors were observed among rare rs299290 homozygotes in BRCA1 mutation carriers (p interaction = 0.006), whereas this bias was not observed in BRCA2 mutation carriers (p interaction = 0.95) (Table S4)', 'BRCA1', 'rs299290', None)\n",
      "(138, 'In data provided by several CIMBA groups (Text S1), no ER-positive tumors were observed among rare rs299290 homozygotes in BRCA1 mutation carriers (p interaction = 0.006), whereas this bias was not observed in BRCA2 mutation carriers (p interaction = 0.95) (Table S4)', 'BRCA2', 'rs299290', None)\n",
      "(139, 'That is, despite the expected differences in the frequency of tumor types between the two sets of carriers, heterogeneity was observed in the distribution of rs299290 genotypes in BRCA1, but not BRCA2, mutation carriers', 'BRCA1', 'rs299290', None)\n",
      "(140, 'That is, despite the expected differences in the frequency of tumor types between the two sets of carriers, heterogeneity was observed in the distribution of rs299290 genotypes in BRCA1, but not BRCA2, mutation carriers', 'BRCA2', 'rs299290', None)\n",
      "(141, 'Importantly, both proteasome inhibition and BRCA1 depletion increased the abundance of RHAMM (Figure S7), which is also consistent with observed RHAMM over-expression in breast cancer cell lines derived from BRCA1 mutation carriers', 'BRCA1', 'S7', None)\n",
      "(142, 'Importantly, both proteasome inhibition and BRCA1 depletion increased the abundance of RHAMM (Figure S7), which is also consistent with observed RHAMM over-expression in breast cancer cell lines derived from BRCA1 mutation carriers', 'RHAMM', 'S7', None)\n",
      "(143, 'Importantly, both proteasome inhibition and BRCA1 depletion increased the abundance of RHAMM (Figure S7), which is also consistent with observed RHAMM over-expression in breast cancer cell lines derived from BRCA1 mutation carriers', 'RHAMM', 'S7', None)\n",
      "(144, 'Importantly, both proteasome inhibition and BRCA1 depletion increased the abundance of RHAMM (Figure S7), which is also consistent with observed RHAMM over-expression in breast cancer cell lines derived from BRCA1 mutation carriers', 'BRCA1', 'S7', None)\n",
      "(145, 'BRCA1 depletion, however, did not alter AURKA levels (Figure S7)', 'BRCA1', 'S7', None)\n",
      "(146, 'BRCA1 depletion, however, did not alter AURKA levels (Figure S7)', 'AURKA', 'S7', None)\n",
      "(147, 'In contrast to single depletions, simultaneous reduction of AURKA and RHAMM levels recovered normal acini formation (Figure 4C-G), possibly implying a negative regulatory relationship between RHAMM abundance and AURKA activity', 'AURKA', '4C-G', None)\n",
      "(148, 'In contrast to single depletions, simultaneous reduction of AURKA and RHAMM levels recovered normal acini formation (Figure 4C-G), possibly implying a negative regulatory relationship between RHAMM abundance and AURKA activity', 'RHAMM', '4C-G', None)\n",
      "(149, 'In contrast to single depletions, simultaneous reduction of AURKA and RHAMM levels recovered normal acini formation (Figure 4C-G), possibly implying a negative regulatory relationship between RHAMM abundance and AURKA activity', 'RHAMM', '4C-G', None)\n",
      "(150, 'In contrast to single depletions, simultaneous reduction of AURKA and RHAMM levels recovered normal acini formation (Figure 4C-G), possibly implying a negative regulatory relationship between RHAMM abundance and AURKA activity', 'AURKA', '4C-G', None)\n",
      "(151, 'Consistently, AURKA depletion reduced RHAMM levels (Figure S7C), while simultaneous depletion of AURKA and BRCA1 recovered RHAMM to control levels (Figure S7C)', 'AURKA', 'S7C', None)\n",
      "(152, 'Consistently, AURKA depletion reduced RHAMM levels (Figure S7C), while simultaneous depletion of AURKA and BRCA1 recovered RHAMM to control levels (Figure S7C)', 'RHAMM', 'S7C', None)\n",
      "(153, 'Consistently, AURKA depletion reduced RHAMM levels (Figure S7C), while simultaneous depletion of AURKA and BRCA1 recovered RHAMM to control levels (Figure S7C)', 'AURKA', 'S7C', None)\n",
      "(154, 'Consistently, AURKA depletion reduced RHAMM levels (Figure S7C), while simultaneous depletion of AURKA and BRCA1 recovered RHAMM to control levels (Figure S7C)', 'BRCA1', 'S7C', None)\n",
      "(155, 'Consistently, AURKA depletion reduced RHAMM levels (Figure S7C), while simultaneous depletion of AURKA and BRCA1 recovered RHAMM to control levels (Figure S7C)', 'RHAMM', 'S7C', None)\n",
      "(158, 'In MCF10A cells, AURKA abundance and activity determined total RHAMM as well as pT703-RHAMM levels (Figure S10A)', 'RHAMM', 'S10A', None)\n",
      "(159, 'This model is consistent with reduced expression of AURKA, TPX2, and HMMR, but to a lesser extent BRCA1, with polarization and growth arrest of nonmalignant mammary epithelial cells, as measured by gene expression profiling (Figure S11A)', 'AURKA', 'S11A', None)\n",
      "(160, 'This model is consistent with reduced expression of AURKA, TPX2, and HMMR, but to a lesser extent BRCA1, with polarization and growth arrest of nonmalignant mammary epithelial cells, as measured by gene expression profiling (Figure S11A)', 'TPX2', 'S11A', None)\n",
      "(161, 'This model is consistent with reduced expression of AURKA, TPX2, and HMMR, but to a lesser extent BRCA1, with polarization and growth arrest of nonmalignant mammary epithelial cells, as measured by gene expression profiling (Figure S11A)', 'HMMR', 'S11A', None)\n",
      "(162, 'This model is consistent with reduced expression of AURKA, TPX2, and HMMR, but to a lesser extent BRCA1, with polarization and growth arrest of nonmalignant mammary epithelial cells, as measured by gene expression profiling (Figure S11A)', 'BRCA1', 'S11A', None)\n",
      "(163, 'According to the model and as stated above, analysis of public gene expression datasets suggests that the rs299290 risk allele is associated with HMMR germline over-expression (Table S3)', 'HMMR', 'rs299290', None)\n",
      "(164, 'As the potential splicing alteration by rs299284 might be tissue specific and RHAMM-R92 was used in the over-expression assays, further work may be warranted to define the causal mutation(s) and the alteration of RHAMM function and/or expression level according to the depicted model.', 'RHAMM', 'rs299284', None)\n",
      "(165, 'As the potential splicing alteration by rs299284 might be tissue specific and RHAMM-R92 was used in the over-expression assays, further work may be warranted to define the causal mutation(s) and the alteration of RHAMM function and/or expression level according to the depicted model.', 'RHAMM', 'rs299284', None)\n",
      "(166, 'As the potential splicing alteration by rs299284 might be tissue specific and RHAMM-R92 was used in the over-expression assays, further work may be warranted to define the causal mutation(s) and the alteration of RHAMM function and/or expression level according to the depicted model.', 'RHAMM', 'R92', None)\n",
      "(167, 'As the potential splicing alteration by rs299284 might be tissue specific and RHAMM-R92 was used in the over-expression assays, further work may be warranted to define the causal mutation(s) and the alteration of RHAMM function and/or expression level according to the depicted model.', 'RHAMM', 'R92', None)\n",
      "(168, 'In preparation for mitotic spindle assembly, AURKA promotes centrosome-dependent microtubule assembly by suppressing BRCA1-dependent ubiquitination, which involves BRCA1 phosphorylation at S308', 'AURKA', 'S308', None)\n",
      "(169, 'In preparation for mitotic spindle assembly, AURKA promotes centrosome-dependent microtubule assembly by suppressing BRCA1-dependent ubiquitination, which involves BRCA1 phosphorylation at S308', 'BRCA1', 'S308', None)\n",
      "(170, 'In preparation for mitotic spindle assembly, AURKA promotes centrosome-dependent microtubule assembly by suppressing BRCA1-dependent ubiquitination, which involves BRCA1 phosphorylation at S308', 'BRCA1', 'S308', None)\n",
      "(171, 'Accordingly, BRCA1 depletion increases the clonogenic potential of mammary epithelia, while a BRCA1 S308A mutant alters embryonic stem cell differentiation', 'BRCA1', 'S308A', None)\n",
      "(172, 'Accordingly, BRCA1 depletion increases the clonogenic potential of mammary epithelia, while a BRCA1 S308A mutant alters embryonic stem cell differentiation', 'BRCA1', 'S308A', None)\n",
      "(173, 'In order to reconstitute HCC1937 cells with wild-type BRCA1, the corresponding full-length open-reading frame was cloned into a retroviral vector S11N and transduced', 'BRCA1', 'S11N', None)\n",
      "(174, 'This work was funded by the Spanish Ministries of Health, and Science ane Innovation (CB07/02/2005; FIS 08/1120, 08/1359, 08/1635, and 09/02483; RTICCC RD06/0020/1060 and RD06/0020/0028; Transversal Action Against Cancer; the Spanish Biomedical Research Centre Networks for Epidemiology and Public Health, and Rare Diseases; and the \"Ramon y Cajal\" Young Investigator Program), the Spanish National Society of Medical Oncology (2010), the Spanish Association Against Cancer (AECC 2010), the AGAUR Catalan Government Agency (2009SGR1489 and 2009SGR293; and the Beatriu Pinos Postdoctoral Program), the Ramon Areces Foundation (XV), the \"Roses Contra el Cancer\" Foundation, the Michael Cuccione Foundation for Childhood Cancer Research, Cancer Research-UK (C490/A10119, C1287/A8874, C1287/A10118, C5047/A8385, and C8197/A10123), the National Institute for Health Research (UK), the Association for International Cancer Research (AICR-07-0454), the Ligue National Contre le Cancer (France), the Association \"Le cancer du sein, parlons-en!\", the Dutch Cancer Society (NKI 1998-1854, 2004-3088, and 2007-3756), the Fondazione Italiana per la Ricerca sul Cancro (\"Hereditary Tumors\"), the Associazione Italiana per la Ricerca sul Cancro (4017), the Italian Ministero della Salute (RFPS-2006-3-340203 and \"Progetto Tumori Femminili\"), the Italian Ministero dell\\'Universita\\' e Ricerca (RBLAO3-BETH), the Fondazione IRCCS Istituto Nazionale Tumori (INT \"5x1000\"), the Fondazione Cassa di Risparmio di Pisa (Istituto Toscano Tumori), the National Breast Cancer Foundation (Australia), the Australian National Health and Medical Research Council (145684, 288704, and 454508), the Queensland Cancer Fund, the Cancer Councils of New South Wales, Victoria, Tasmania, and South Australia, the Cancer Foundation of Western Australia, the German Cancer Aid (107054), the Center for Molecular Medicine Cologne (TV93), the National Cancer Institute (USA; CA128978 and CA122340), National Institutes of Health (RFA-CA-06-503, BCFR U01 CA69398, CA69417, CA69446, CA69467, CA69631, and CA69638), the Research Triangle Institute Informatics Support Center (RFP N02PC45022-46), the Specialized Program of Research Excellence (SPORE P50 CA83638 and CA113916), the Department of Defense Breast Cancer Research Program (05/0612), the Eileen Stein Jacoby Fund, the Breast Cancer Research Foundation, the Marianne and Robert MacDonald Foundation, the Komen Foundation, the Helsinki University Central Hospital Research Fund, the Academy of Finland (110663), the Finnish Cancer Society, the Sigrid Juselius Foundation, and the EU FP7 (223175, HEALTH-F2-2009-223175)', 'FIS', 'C1287/A8874', None)\n",
      "(175, 'The 185delAG mutation (c.68_69delAG) in the BRCA1 gene triggers translation reinitiation at a downstream AUG codon.', 'BRCA1', '185delAG', None)\n",
      "(176, 'The 185delAG mutation (c.68_69delAG) in the BRCA1 gene triggers translation reinitiation at a downstream AUG codon.', 'BRCA1', 'c.68_69delAG', None)\n",
      "(177, 'Inconsistent association between the STK15 F31I genetic polymorphism and breast cancer risk.', 'STK15', 'F31I', None)\n"
     ]
    }
   ],
   "source": [
    "#store the dataframe into psql table\n",
    "import psycopg2\n",
    "import pandas as pd\n",
    "from sqlalchemy import create_engine\n",
    "\n",
    "conn_string = 'postgresql://postgres:sinkiru@localhost:5432/gdmbiobert'\n",
    "\n",
    "db = create_engine(conn_string)\n",
    "conn = db.connect()\n",
    "\n",
    "final_re_output.to_sql('gm', con=conn, if_exists='replace', index=False)\n",
    "conn = psycopg2.connect(conn_string)\n",
    "conn.autocommit = True\n",
    "cursor = conn.cursor()\n",
    "\n",
    "sql1 = '''select * from gm;'''\n",
    "cursor.execute(sql1)\n",
    "for i in cursor.fetchall():\n",
    "\tprint(i)\n",
    "\n",
    "# conn.commit()\n",
    "conn.close()\n"
   ]
  }
 ],
 "metadata": {
  "colab": {
   "provenance": []
  },
  "kernelspec": {
   "display_name": "Python 3.6.13 ('env1')",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.13"
  },
  "vscode": {
   "interpreter": {
    "hash": "6626a9a69e96ea8665bf0f933e0462e9e2599c09c430a57c21911de9c8d78cbc"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
