{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "executionInfo": {
     "elapsed": 1994,
     "status": "ok",
     "timestamp": 1679115294967,
     "user": {
      "displayName": "19PT24 - S KIRUPA",
      "userId": "12629547787028680648"
     },
     "user_tz": -330
    },
    "id": "RvaNWdrKgXjD",
    "outputId": "42afdbf9-c1a0-474c-d094-a237c1063f70"
   },
   "outputs": [],
   "source": [
    "#We extract the pubmed document in BioCJSON format\n",
    "import urllib3\n",
    "import json\n",
    "import csv\n",
    "import requests\n",
    "import pandas as pd\n",
    "import io\n",
    "import os\n",
    "#import nltk\n",
    "#nltk.download('punkt')\n",
    "#from nltk.tokenize import word_tokenize\n",
    "from itertools import combinations\n",
    "\n",
    "http = urllib3.PoolManager()\n",
    "\n",
    "pmcids = []\n",
    "\n",
    "\n",
    "# Read the PMC IDs from the input file\n",
    "with open(\"pmcid.txt\", \"r\") as file:\n",
    "    for line in file:\n",
    "        pmcids.append(line.strip())\n",
    "# Initialize an empty list to store the results\n",
    "results = []\n",
    "\n",
    "for pmcid in pmcids:\n",
    "    r = http.request('GET', f'https://www.ncbi.nlm.nih.gov/research/pubtator-api/publications/export/biocjson?pmcids={pmcid}')\n",
    "    data = json.loads(r.data.decode('utf-8'))\n",
    "\n",
    "    # Add the results to the list\n",
    "    results.append(data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 122
    },
    "executionInfo": {
     "elapsed": 6,
     "status": "ok",
     "timestamp": 1679115294968,
     "user": {
      "displayName": "19PT24 - S KIRUPA",
      "userId": "12629547787028680648"
     },
     "user_tz": -330
    },
    "id": "-3bUHWus1nUq",
    "outputId": "6459e92e-d045-4841-eb58-35ac0f68d993"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'\\nEvery PMC id has passages.\\nEvery passage has many {infons, offset, text, sentences, annotations, relations}.\\nHere, text is the actual text we have to annotate. \\nEvery annotations has {id, infons, text, locations}. \\nHere infons has {identifier, type} (optional: ncbi-homologene if type is gene).\\nAlso locations has {offset, length}.\\n\\nPassages:\\n    a)infons - data realted article id, author, etc..\\n    b)offset - location index\\n    c)text - whole medical data (sentence) in which medical terms (gene name or disease name) are to be annotated.\\n    d)sentences - not required here\\n    e)annotations:\\n        1)id - key index\\n        2)infons:\\n              a)identifier\\n              b)type - \"Gene\" or \"Disease\" etc.,\\n        3)text - gene name or disease name etc., (Eg: \"K-Ras\")\\n        4)locations:\\n            a)offset - location index\\n            b)length - length of text. (Eg: len(\"tumours\") = 7) \\n    f)relations\\n'"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "output = json.dumps(data, indent=4)\n",
    "#print(output)\n",
    "\n",
    "'''\n",
    "Every PMC id has passages.\n",
    "Every passage has many {infons, offset, text, sentences, annotations, relations}.\n",
    "Here, text is the actual text we have to annotate. \n",
    "Every annotations has {id, infons, text, locations}. \n",
    "Here infons has {identifier, type} (optional: ncbi-homologene if type is gene).\n",
    "Also locations has {offset, length}.\n",
    "\n",
    "Passages:\n",
    "    a)infons - data realted article id, author, etc..\n",
    "    b)offset - location index\n",
    "    c)text - whole medical data (sentence) in which medical terms (gene name or disease name) are to be annotated.\n",
    "    d)sentences - not required here\n",
    "    e)annotations:\n",
    "        1)id - key index\n",
    "        2)infons:\n",
    "              a)identifier\n",
    "              b)type - \"Gene\" or \"Disease\" etc.,\n",
    "        3)text - gene name or disease name etc., (Eg: \"K-Ras\")\n",
    "        4)locations:\n",
    "            a)offset - location index\n",
    "            b)length - length of text. (Eg: len(\"tumours\") = 7) \n",
    "    f)relations\n",
    "'''\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "id": "3EkWKX8wA1b5"
   },
   "outputs": [],
   "source": [
    "df1 = pd.DataFrame(columns=['index','sentence'])\n",
    "df2 = pd.DataFrame(columns=['index','sentence','Gene','Disease'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "ZSMLCcRaAlL_"
   },
   "source": [
    "**Annotations for Gene and Mutation**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "id": "kbIqNKyCgkuF"
   },
   "outputs": [],
   "source": [
    "sentence_index = 0\n",
    "sentence_entities = {}\n",
    "for j in range(len(results)):\n",
    "  for i in results[j]['passages']:\n",
    "    if i['infons']['section_type'] != 'TABLE':\n",
    "      #filter the table segment\n",
    "      text = i['text']\n",
    "      # print(\"TEXT: \", text)\n",
    "      offset = i['offset']\n",
    "      # print(\"OFFSET: \", offset)    \n",
    "      annotations = i['annotations']\n",
    "      \n",
    "      annotations = sorted(annotations, key = lambda x: x['locations'][0]['offset'])\n",
    "      # print(\"ANNOTATIONS: \", annotations)\n",
    "      #Filter to only include gene-disease annotations\n",
    "      annotations = [annotation for annotation in annotations if ((annotation['infons']['type']=='Gene') or (annotation['infons']['type']=='Disease'))]\n",
    "      #List all possible combinations of annotations\n",
    "      annots_combinations = list(combinations(annotations, 2))\n",
    "      # print(\"ANNOT COMBIMNATIONS: \", annots_combinations)\n",
    "      #Filter combinations to only include gene-disease combinations\n",
    "      annots_combinations = [annots for annots in annots_combinations if annots[0]['infons']['type'] != annots[1]['infons']['type']]\n",
    "      # print(\"ANNOT COMBINATIONS - GENE-DISEASE: \", annots_combinations)\n",
    "\n",
    "      #processing sentences\n",
    "      sentences = text.split('. ')\n",
    "      # print(sentences)\n",
    "      sentence_offset = {}\n",
    "      sentence_len = {}\n",
    "      prev_sent_offset = offset\n",
    "\n",
    "      for sentence in sentences:\n",
    "\n",
    "        sentence_offset[sentence] = prev_sent_offset\n",
    "        current_sentence_offset = prev_sent_offset\n",
    "        sentence_len[sentence] = len(sentence)\n",
    "        current_sentence_len = len(sentence)\n",
    "        prev_sent_offset += len(sentence) + 2\n",
    "        \n",
    "\n",
    "        #Point to note: Duplicate sentences for as many combinations as present. \n",
    "        for annots in annots_combinations:\n",
    "          difference = 0\n",
    "          current_sentence = sentence\n",
    "          #sort the tuple\n",
    "          annots = sorted(annots, key = lambda x: x['locations'][0]['offset'])\n",
    "\n",
    "          entity_1 = annots[0]\n",
    "          entity_2 = annots[1]\n",
    "          \n",
    "          entity_1_offset = entity_1['locations'][0]['offset']\n",
    "          entity_2_offset = entity_2['locations'][0]['offset']\n",
    "\n",
    "          entity_1_dist = entity_1_offset - current_sentence_offset\n",
    "          entity_2_dist = entity_2_offset - current_sentence_offset\n",
    "\n",
    "          if (0 <= entity_1_dist <= ((current_sentence_len - len(entity_1['text'])) + 1)) and (0 <= entity_2_dist <= ((current_sentence_len - len(entity_2['text'])) + 1)):\n",
    "            #the pair of annotations fall within the sentence\n",
    "            sentence_entities[sentence_index] = (entity_1['text'], entity_2['text'])\n",
    "\n",
    "            entity_1_type = entity_1['infons']['type']\n",
    "            entity_1_length = entity_1['locations'][0]['length']\n",
    "            temp = '@'+ entity_1_type +'$'\n",
    "            entity_1_final_off = entity_1_dist \n",
    "            current_sentence = current_sentence[:entity_1_final_off] + \"@\" + entity_1_type + \"$\" + current_sentence[(entity_1_final_off + entity_1_length):]\n",
    "            difference += (entity_1_length - len(temp))\n",
    "            entity_2_type = entity_2['infons']['type']\n",
    "            entity_2_length = entity_2['locations'][0]['length']\n",
    "            temp = '@'+ entity_2_type +'$'\n",
    "            entity_2_final_off = entity_2_dist - (difference)\n",
    "            current_sentence = current_sentence[:entity_2_final_off] + \"@\" + entity_2_type + \"$\" + current_sentence[(entity_2_final_off + entity_2_length):]\n",
    "            difference += (entity_2_length - len(temp))\n",
    "            # tsv_writer_1.writerow([sentence_index, current_sentence])\n",
    "            row1 = [sentence_index,current_sentence]\n",
    "            df1.loc[len(df1)] = row1\n",
    "            if (entity_1['infons']['type'] == 'Gene'):\n",
    "              # print('Writing...', [sentence_index, sentence, entity_1['text'], entity_2['text']])\n",
    "              # tsv_writer_2.writerow([sentence_index, sentence, entity_1['text'], entity_2['text']])\n",
    "              row2 = [sentence_index, sentence, entity_1['text'], entity_2['text']]\n",
    "              df2.loc[len(df2)] = row2\n",
    "            else:\n",
    "              # print('Writing')\n",
    "              # tsv_writer_2.writerow([sentence_index, sentence, entity_2['text'], entity_1['text']])\n",
    "              row2 = [sentence_index, sentence, entity_2['text'], entity_1['text']]\n",
    "              df2.loc[len(df2)] = row2\n",
    "            sentence_index += 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "id": "rax5aqQmrZp-"
   },
   "outputs": [],
   "source": [
    "import os\n",
    "os.chdir('../biobert-pytorch/relation-extraction')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "id": "Oe0kay_JgprA"
   },
   "outputs": [],
   "source": [
    "df2.to_csv('pub_original_sentences_GD.tsv', sep=\"\\t\", index=False)\n",
    "df1.to_csv('inputGD/test.tsv', sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "523"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(df1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "ZZSYZQAAi5db"
   },
   "source": [
    "**BioBert Training**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "colab": {
     "background_save": true
    },
    "id": "ETmbMG8sDB2U"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/skirupa/Desktop/sem8/RSL-Lab/RSL Final/GDM-BioBert/biobert-pytorch/relation-extraction\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "print(os.getcwd())\n",
    "os.chdir('..')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "colab": {
     "background_save": true,
     "base_uri": "https://localhost:8080/"
    },
    "executionInfo": {
     "elapsed": 49572,
     "status": "ok",
     "timestamp": 1679113913000,
     "user": {
      "displayName": "19PT24 - S KIRUPA",
      "userId": "12629547787028680648"
     },
     "user_tz": -330
    },
    "id": "zIu8GyzMjUoE",
    "outputId": "fd85fe47-e028-48ca-c80d-d9d344329c9e"
   },
   "outputs": [],
   "source": [
    "# Download all datasets including NER/RE/QA\n",
    "#!bash ./download.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "colab": {
     "background_save": true,
     "base_uri": "https://localhost:8080/"
    },
    "executionInfo": {
     "elapsed": 7241,
     "status": "ok",
     "timestamp": 1679113920218,
     "user": {
      "displayName": "19PT24 - S KIRUPA",
      "userId": "12629547787028680648"
     },
     "user_tz": -330
    },
    "id": "lfbbpV5lkJry",
    "outputId": "3c42dba5-3879-4891-c84f-ffcca7fbe599"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "*****  euadr  Preprocessing Start *****\n",
      "*****  euadr  Preprocessing Done *****\n",
      "*****  GAD  Preprocessing Start *****\n",
      "*****  GAD  Preprocessing Done *****\n"
     ]
    }
   ],
   "source": [
    "os.chdir('relation-extraction')\n",
    "\n",
    "#To preprocess the datasets downloaded\n",
    "!bash ./preprocess.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "colab": {
     "background_save": true,
     "base_uri": "https://localhost:8080/"
    },
    "executionInfo": {
     "elapsed": 9321,
     "status": "ok",
     "timestamp": 1679113929525,
     "user": {
      "displayName": "19PT24 - S KIRUPA",
      "userId": "12629547787028680648"
     },
     "user_tz": -330
    },
    "id": "jPCXvCJLkLsy",
    "outputId": "5b07e932-5654-4efb-cd39-55e358b64933"
   },
   "outputs": [],
   "source": [
    "#!pip install scikit-learn\n",
    "#!pip install pandas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!pip install torch torchvision torchaudio"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "executionInfo": {
     "elapsed": 831,
     "status": "ok",
     "timestamp": 1679120234782,
     "user": {
      "displayName": "19PT24 - S KIRUPA",
      "userId": "12629547787028680648"
     },
     "user_tz": -330
    },
    "id": "uv7yHEmWkPk8",
    "outputId": "83a27745-c3c5-486f-f279-91040342140c"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "env: SAVE_DIR=./outputGD\n",
      "env: DATA=\"GAD\"\n",
      "env: SPLIT=\"1\"\n",
      "env: DATA_DIR=./inputGD\n",
      "env: ENTITY=${DATA}-${SPLIT}\n",
      "env: MAX_LENGTH=128\n",
      "env: BATCH_SIZE=32\n",
      "env: NUM_EPOCHS=3\n",
      "env: SAVE_STEPS=1000\n",
      "env: SEED=1\n"
     ]
    }
   ],
   "source": [
    "%env SAVE_DIR=./outputGD\n",
    "%env DATA=\"GAD\"\n",
    "%env SPLIT=\"1\"\n",
    "%env DATA_DIR=./inputGD\n",
    "%env ENTITY=${DATA}-${SPLIT}\n",
    "\n",
    "%env MAX_LENGTH=128\n",
    "%env BATCH_SIZE=32\n",
    "%env NUM_EPOCHS=3\n",
    "%env SAVE_STEPS=1000\n",
    "%env SEED=1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "executionInfo": {
     "elapsed": 6962,
     "status": "ok",
     "timestamp": 1679120242449,
     "user": {
      "displayName": "19PT24 - S KIRUPA",
      "userId": "12629547787028680648"
     },
     "user_tz": -330
    },
    "id": "Nap7OsDNomOM",
    "outputId": "e92aad7a-3cdc-4402-ce07-ed5660a090a3"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "05/02/2023 11:00:26 - WARNING - __main__ -   Process rank: -1, device: cpu, n_gpu: 0, distributed training: False, 16-bits training: False\n",
      "05/02/2023 11:00:34 - INFO - __main__ -   Training/evaluation parameters TrainingArguments(output_dir='./outputGD', overwrite_output_dir=True, do_train=False, do_eval=False, do_predict=True, evaluate_during_training=False, evaluation_strategy=<EvaluationStrategy.NO: 'no'>, prediction_loss_only=False, per_device_train_batch_size=8, per_device_eval_batch_size=8, per_gpu_train_batch_size=None, per_gpu_eval_batch_size=None, gradient_accumulation_steps=1, eval_accumulation_steps=None, learning_rate=5e-05, weight_decay=0.0, adam_beta1=0.9, adam_beta2=0.999, adam_epsilon=1e-08, max_grad_norm=1.0, num_train_epochs=3.0, max_steps=-1, warmup_steps=0, logging_dir='runs/May02_11-00-26_skirupa', logging_first_step=False, logging_steps=500, save_steps=500, save_total_limit=None, no_cuda=False, seed=42, fp16=False, fp16_opt_level='O1', local_rank=-1, tpu_num_cores=None, tpu_metrics_debug=False, debug=False, dataloader_drop_last=False, eval_steps=500, dataloader_num_workers=0, past_index=-1, run_name='./outputGD', disable_tqdm=False, remove_unused_columns=True, label_names=None, load_best_model_at_end=False, metric_for_best_model=None, greater_is_better=None)\n",
      "Some weights of the model checkpoint at dmis-lab/biobert-base-cased-v1.1 were not used when initializing BertForSequenceClassification: ['cls.predictions.bias', 'cls.predictions.transform.dense.weight', 'cls.predictions.transform.dense.bias', 'cls.predictions.transform.LayerNorm.weight', 'cls.predictions.transform.LayerNorm.bias', 'cls.predictions.decoder.weight', 'cls.predictions.decoder.bias', 'cls.seq_relationship.weight', 'cls.seq_relationship.bias']\n",
      "- This IS expected if you are initializing BertForSequenceClassification from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPretraining model).\n",
      "- This IS NOT expected if you are initializing BertForSequenceClassification from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).\n",
      "Some weights of BertForSequenceClassification were not initialized from the model checkpoint at dmis-lab/biobert-base-cased-v1.1 and are newly initialized: ['classifier.weight', 'classifier.bias']\n",
      "You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.\n",
      "05/02/2023 11:00:50 - INFO - root -   *** Test ***\n",
      "100%|███████████████████████████████████████████| 27/27 [00:36<00:00,  1.20s/it]/home/skirupa/miniconda3/envs/env1/lib/python3.6/site-packages/transformers/trainer.py:1118: FutureWarning: This method is deprecated, use `Trainer.is_world_process_zero()` instead.\n",
      "  warnings.warn(\"This method is deprecated, use `Trainer.is_world_process_zero()` instead.\", FutureWarning)\n",
      "05/02/2023 11:01:32 - INFO - __main__ -   ***** Test results sst-2 *****\n",
      "100%|███████████████████████████████████████████| 27/27 [00:36<00:00,  1.36s/it]\n"
     ]
    }
   ],
   "source": [
    "!python run_re.py --task_name SST-2 --config_name bert-base-cased --model_name_or_path dmis-lab/biobert-base-cased-v1.1 \\\n",
    "        --do_predict --data_dir ${DATA_DIR} \\\n",
    "        --output_dir ${SAVE_DIR} \\\n",
    "        --overwrite_output_dir\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 328
    },
    "executionInfo": {
     "elapsed": 13,
     "status": "error",
     "timestamp": 1679064615235,
     "user": {
      "displayName": "19PT24 - S KIRUPA",
      "userId": "12629547787028680648"
     },
     "user_tz": -330
    },
    "id": "QgHoYa6Noo9w",
    "outputId": "6af921bc-1802-4a23-bb95-187243510d51"
   },
   "outputs": [],
   "source": [
    "original_sentences = pd.read_csv('pub_original_sentences_GD.tsv', sep=\"\\t\")\n",
    "predictions = pd.read_csv('outputGD/test_results.txt', sep='\\t')\n",
    "\n",
    "#df = pd.read_csv('pub_original_sentences.tsv', sep=\"\\t\")\n",
    "#df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "id": "QX9DHYpyBbeC"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>sentence</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Disease</th>\n",
       "      <th>prediction</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>Activating K-Ras mutations outwith 'hotspot' c...</td>\n",
       "      <td>K-Ras</td>\n",
       "      <td>colorectal tumours</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>Activating K-Ras mutations outwith 'hotspot' c...</td>\n",
       "      <td>K-Ras</td>\n",
       "      <td>cancer</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>Response to EGFR-targeted therapies in colorec...</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>colorectal cancer</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>Response to EGFR-targeted therapies in colorec...</td>\n",
       "      <td>Kirsten-Ras</td>\n",
       "      <td>colorectal cancer</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>Response to EGFR-targeted therapies in colorec...</td>\n",
       "      <td>K-Ras</td>\n",
       "      <td>colorectal cancer</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>518</th>\n",
       "      <td>518</td>\n",
       "      <td>Identification of a new APC/C recognition doma...</td>\n",
       "      <td>Cdh1</td>\n",
       "      <td>APC</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>519</th>\n",
       "      <td>519</td>\n",
       "      <td>Identification of a new APC/C recognition doma...</td>\n",
       "      <td>Aurora-A</td>\n",
       "      <td>APC</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>520</th>\n",
       "      <td>520</td>\n",
       "      <td>Aurora-A kinase regulates breast cancer-associ...</td>\n",
       "      <td>Aurora-A</td>\n",
       "      <td>breast cancer</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>521</th>\n",
       "      <td>521</td>\n",
       "      <td>Hyaluronan-mediated motility receptor gene si...</td>\n",
       "      <td>Hyaluronan-mediated motility receptor</td>\n",
       "      <td>breast cancer</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522</th>\n",
       "      <td>522</td>\n",
       "      <td>Inconsistent association between the STK15 F31...</td>\n",
       "      <td>STK15</td>\n",
       "      <td>breast cancer</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>523 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     index                                           sentence  \\\n",
       "0        0  Activating K-Ras mutations outwith 'hotspot' c...   \n",
       "1        1  Activating K-Ras mutations outwith 'hotspot' c...   \n",
       "2        2  Response to EGFR-targeted therapies in colorec...   \n",
       "3        3  Response to EGFR-targeted therapies in colorec...   \n",
       "4        4  Response to EGFR-targeted therapies in colorec...   \n",
       "..     ...                                                ...   \n",
       "518    518  Identification of a new APC/C recognition doma...   \n",
       "519    519  Identification of a new APC/C recognition doma...   \n",
       "520    520  Aurora-A kinase regulates breast cancer-associ...   \n",
       "521    521   Hyaluronan-mediated motility receptor gene si...   \n",
       "522    522  Inconsistent association between the STK15 F31...   \n",
       "\n",
       "                                      Gene             Disease  prediction  \n",
       "0                                    K-Ras  colorectal tumours         1.0  \n",
       "1                                    K-Ras              cancer         1.0  \n",
       "2                                     EGFR   colorectal cancer         0.0  \n",
       "3                              Kirsten-Ras   colorectal cancer         1.0  \n",
       "4                                    K-Ras   colorectal cancer         0.0  \n",
       "..                                     ...                 ...         ...  \n",
       "518                                   Cdh1                 APC         NaN  \n",
       "519                               Aurora-A                 APC         NaN  \n",
       "520                               Aurora-A       breast cancer         NaN  \n",
       "521  Hyaluronan-mediated motility receptor       breast cancer         NaN  \n",
       "522                                  STK15       breast cancer         NaN  \n",
       "\n",
       "[523 rows x 5 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Merge the pub original sentences and test results (prediction of biobert model)\n",
    "\n",
    "final_re_output = pd.merge(original_sentences, predictions, on ='index', how='left')\n",
    "final_re_output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "#store result in a csv\n",
    "#os.chdir('outputGD')\n",
    "#final_re_output.to_csv('final_GD_output.tsv', sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(0, \"Activating K-Ras mutations outwith 'hotspot' codons in sporadic colorectal tumours - implications for personalised cancer medicine\", 'K-Ras', 'colorectal tumours', 1.0)\n",
      "(1, \"Activating K-Ras mutations outwith 'hotspot' codons in sporadic colorectal tumours - implications for personalised cancer medicine\", 'K-Ras', 'cancer', 1.0)\n",
      "(2, 'Response to EGFR-targeted therapies in colorectal cancer patients has been convincingly associated with Kirsten-Ras (K-Ras) mutation status', 'EGFR', 'colorectal cancer', 0.0)\n",
      "(3, 'Response to EGFR-targeted therapies in colorectal cancer patients has been convincingly associated with Kirsten-Ras (K-Ras) mutation status', 'Kirsten-Ras', 'colorectal cancer', 1.0)\n",
      "(4, 'Response to EGFR-targeted therapies in colorectal cancer patients has been convincingly associated with Kirsten-Ras (K-Ras) mutation status', 'K-Ras', 'colorectal cancer', 0.0)\n",
      "(5, 'Colorectal tumours (n=106) were screened for additional K-Ras mutations, phenotypes compared in transformation and Ras GTPase activating assays and gene and pathway changes induced by individual K-Ras mutants identified by microarray analysis', 'K-Ras', 'Colorectal tumours', 1.0)\n",
      "(6, 'Colorectal tumours (n=106) were screened for additional K-Ras mutations, phenotypes compared in transformation and Ras GTPase activating assays and gene and pathway changes induced by individual K-Ras mutants identified by microarray analysis', 'K-Ras', 'Colorectal tumours', 1.0)\n",
      "(7, 'Four additional K-Ras mutations (Leu19Phe (1 out of 106 tumours), Lys117Asn (1 out of 106), Ala146Thr (7 out of 106) and Arg164Gln (1 out of 106)) were identified', 'K-Ras', 'tumours', 1.0)\n",
      "(8, 'We additionally identified a new K-Ras gene amplification event, present in approximately 2% of tumours.', 'K-Ras', 'tumours', 1.0)\n",
      "(9, 'The identification of mutations outwith previously described hotspot codons increases the K-Ras mutation burden in colorectal tumours by one-third', 'K-Ras', 'colorectal tumours', 1.0)\n",
      "(10, 'A genetic model for colorectal cancer highlighted key genes, including the tumour suppressor genes adenomatous polyposis coli (APC) and p53 and the oncogene Kirsten-Ras (K-Ras), the progressive acquisition of mutations in which was proposed to regulate the adenoma-carcinoma transition (Vogelstein et al, 1988)', 'p53', 'colorectal cancer', 1.0)\n",
      "(11, 'A genetic model for colorectal cancer highlighted key genes, including the tumour suppressor genes adenomatous polyposis coli (APC) and p53 and the oncogene Kirsten-Ras (K-Ras), the progressive acquisition of mutations in which was proposed to regulate the adenoma-carcinoma transition (Vogelstein et al, 1988)', 'Kirsten-Ras', 'colorectal cancer', 1.0)\n",
      "(12, 'A genetic model for colorectal cancer highlighted key genes, including the tumour suppressor genes adenomatous polyposis coli (APC) and p53 and the oncogene Kirsten-Ras (K-Ras), the progressive acquisition of mutations in which was proposed to regulate the adenoma-carcinoma transition (Vogelstein et al, 1988)', 'K-Ras', 'colorectal cancer', 1.0)\n",
      "(13, 'A genetic model for colorectal cancer highlighted key genes, including the tumour suppressor genes adenomatous polyposis coli (APC) and p53 and the oncogene Kirsten-Ras (K-Ras), the progressive acquisition of mutations in which was proposed to regulate the adenoma-carcinoma transition (Vogelstein et al, 1988)', 'p53', 'tumour', 1.0)\n",
      "(14, 'A genetic model for colorectal cancer highlighted key genes, including the tumour suppressor genes adenomatous polyposis coli (APC) and p53 and the oncogene Kirsten-Ras (K-Ras), the progressive acquisition of mutations in which was proposed to regulate the adenoma-carcinoma transition (Vogelstein et al, 1988)', 'Kirsten-Ras', 'tumour', 1.0)\n",
      "(15, 'A genetic model for colorectal cancer highlighted key genes, including the tumour suppressor genes adenomatous polyposis coli (APC) and p53 and the oncogene Kirsten-Ras (K-Ras), the progressive acquisition of mutations in which was proposed to regulate the adenoma-carcinoma transition (Vogelstein et al, 1988)', 'K-Ras', 'tumour', 1.0)\n",
      "(16, 'A genetic model for colorectal cancer highlighted key genes, including the tumour suppressor genes adenomatous polyposis coli (APC) and p53 and the oncogene Kirsten-Ras (K-Ras), the progressive acquisition of mutations in which was proposed to regulate the adenoma-carcinoma transition (Vogelstein et al, 1988)', 'p53', 'adenomatous polyposis coli', 1.0)\n",
      "(17, 'A genetic model for colorectal cancer highlighted key genes, including the tumour suppressor genes adenomatous polyposis coli (APC) and p53 and the oncogene Kirsten-Ras (K-Ras), the progressive acquisition of mutations in which was proposed to regulate the adenoma-carcinoma transition (Vogelstein et al, 1988)', 'Kirsten-Ras', 'adenomatous polyposis coli', 1.0)\n",
      "(18, 'A genetic model for colorectal cancer highlighted key genes, including the tumour suppressor genes adenomatous polyposis coli (APC) and p53 and the oncogene Kirsten-Ras (K-Ras), the progressive acquisition of mutations in which was proposed to regulate the adenoma-carcinoma transition (Vogelstein et al, 1988)', 'K-Ras', 'adenomatous polyposis coli', 1.0)\n",
      "(19, 'A genetic model for colorectal cancer highlighted key genes, including the tumour suppressor genes adenomatous polyposis coli (APC) and p53 and the oncogene Kirsten-Ras (K-Ras), the progressive acquisition of mutations in which was proposed to regulate the adenoma-carcinoma transition (Vogelstein et al, 1988)', 'p53', 'APC', 1.0)\n",
      "(20, 'A genetic model for colorectal cancer highlighted key genes, including the tumour suppressor genes adenomatous polyposis coli (APC) and p53 and the oncogene Kirsten-Ras (K-Ras), the progressive acquisition of mutations in which was proposed to regulate the adenoma-carcinoma transition (Vogelstein et al, 1988)', 'Kirsten-Ras', 'APC', 1.0)\n",
      "(21, 'A genetic model for colorectal cancer highlighted key genes, including the tumour suppressor genes adenomatous polyposis coli (APC) and p53 and the oncogene Kirsten-Ras (K-Ras), the progressive acquisition of mutations in which was proposed to regulate the adenoma-carcinoma transition (Vogelstein et al, 1988)', 'K-Ras', 'APC', 1.0)\n",
      "(22, 'A genetic model for colorectal cancer highlighted key genes, including the tumour suppressor genes adenomatous polyposis coli (APC) and p53 and the oncogene Kirsten-Ras (K-Ras), the progressive acquisition of mutations in which was proposed to regulate the adenoma-carcinoma transition (Vogelstein et al, 1988)', 'p53', 'adenoma-carcinoma', 1.0)\n",
      "(23, 'A genetic model for colorectal cancer highlighted key genes, including the tumour suppressor genes adenomatous polyposis coli (APC) and p53 and the oncogene Kirsten-Ras (K-Ras), the progressive acquisition of mutations in which was proposed to regulate the adenoma-carcinoma transition (Vogelstein et al, 1988)', 'Kirsten-Ras', 'adenoma-carcinoma', 1.0)\n",
      "(24, 'A genetic model for colorectal cancer highlighted key genes, including the tumour suppressor genes adenomatous polyposis coli (APC) and p53 and the oncogene Kirsten-Ras (K-Ras), the progressive acquisition of mutations in which was proposed to regulate the adenoma-carcinoma transition (Vogelstein et al, 1988)', 'K-Ras', 'adenoma-carcinoma', 1.0)\n",
      "(25, \"We subsequently refined this model (Smith et al, 2002), highlighting a significant increase in K-Ras mutation frequency in Dukes' C tumours, suggesting that K-Ras mutation status may be an important determinant of tumour progression\", 'K-Ras', \"Dukes' C tumours\", 1.0)\n",
      "(26, \"We subsequently refined this model (Smith et al, 2002), highlighting a significant increase in K-Ras mutation frequency in Dukes' C tumours, suggesting that K-Ras mutation status may be an important determinant of tumour progression\", 'K-Ras', 'tumour', 1.0)\n",
      "(27, \"We subsequently refined this model (Smith et al, 2002), highlighting a significant increase in K-Ras mutation frequency in Dukes' C tumours, suggesting that K-Ras mutation status may be an important determinant of tumour progression\", 'K-Ras', \"Dukes' C tumours\", 1.0)\n",
      "(28, \"We subsequently refined this model (Smith et al, 2002), highlighting a significant increase in K-Ras mutation frequency in Dukes' C tumours, suggesting that K-Ras mutation status may be an important determinant of tumour progression\", 'K-Ras', 'tumour', 1.0)\n",
      "(29, 'We additionally used comparative genomic hybridisation (CGH) analysis to identify common chromosomal aberrations in colorectal tumours, and highlighted an amplification of the region of chromosome 12p where the K-Ras gene is localised (Leslie et al, 2003)', 'K-Ras', 'colorectal tumours', 1.0)\n",
      "(30, 'We and others have shown that K-Ras mutations are associated with significantly reduced survival in colorectal cancer patients (Andreyev et al, 2001; Conlin et al, 2005), although previous data is not entirely consistent (Etienne-Grimaldi et al, 2008; Winder et al, 2009)', 'K-Ras', 'colorectal cancer', 0.0)\n",
      "(31, 'K-Ras mutation status has recently been convincingly associated with response to the new generation EGFR antagonists cetuximab (Erbitux) and panitumumab (Vectibix), where response is preferentially observed in wt K-Ras tumours (Lievre et al, 2006; Benvenuti et al, 2007; Khambata-Ford et al, 2007; Freeman et al, 2008; Karapetis et al, 2008; Ramos et al, 2008; Loupakis et al, 2009; Van Cutsem et al, 2009)', 'K-Ras', 'K-Ras tumours', 1.0)\n",
      "(32, 'K-Ras mutation status has recently been convincingly associated with response to the new generation EGFR antagonists cetuximab (Erbitux) and panitumumab (Vectibix), where response is preferentially observed in wt K-Ras tumours (Lievre et al, 2006; Benvenuti et al, 2007; Khambata-Ford et al, 2007; Freeman et al, 2008; Karapetis et al, 2008; Ramos et al, 2008; Loupakis et al, 2009; Van Cutsem et al, 2009)', 'EGFR', 'K-Ras tumours', 1.0)\n",
      "(33, 'K-Ras mutation testing is therefore increasingly recommended to facilitate selection of the most appropriate patients for treatment with EGFR antagonists (McNeill, 2008; van Krieken and Tol, 2009).', 'K-Ras', 'van Krieken', 1.0)\n",
      "(34, 'K-Ras mutation testing is therefore increasingly recommended to facilitate selection of the most appropriate patients for treatment with EGFR antagonists (McNeill, 2008; van Krieken and Tol, 2009).', 'EGFR', 'van Krieken', 1.0)\n",
      "(35, 'Previous analyses of K-Ras mutations in human tumours have consistently focussed on single-point mutations in codons 12, 13 and 61, where mutation has been shown to result in reduced Ras GAP GTPase activity, locking the protein in the active Ras-GTP conformation (Ellis and Clark, 2000), although additional mutations at codons 19, 22 and 146 have been described in single colorectal tumour case reports (Orita et al, 1991; Miyakura et al, 2002; Akagi et al, 2007).', 'K-Ras', 'tumours', 1.0)\n",
      "(36, 'Previous analyses of K-Ras mutations in human tumours have consistently focussed on single-point mutations in codons 12, 13 and 61, where mutation has been shown to result in reduced Ras GAP GTPase activity, locking the protein in the active Ras-GTP conformation (Ellis and Clark, 2000), although additional mutations at codons 19, 22 and 146 have been described in single colorectal tumour case reports (Orita et al, 1991; Miyakura et al, 2002; Akagi et al, 2007).', 'K-Ras', 'colorectal tumour', 1.0)\n",
      "(37, 'B-Raf mutations have also been identified in human cancers, including melanoma and thyroid, ovarian and colorectal tumours, although mutation frequency estimates in colorectal tumours vary from 1 to 20%, and have been particularly associated with tumours deficient in mismatch-repair activity (Wellbrock et al, 2004)', 'B-Raf', 'cancers', 0.0)\n",
      "(38, 'B-Raf mutations have also been identified in human cancers, including melanoma and thyroid, ovarian and colorectal tumours, although mutation frequency estimates in colorectal tumours vary from 1 to 20%, and have been particularly associated with tumours deficient in mismatch-repair activity (Wellbrock et al, 2004)', 'B-Raf', 'melanoma', 0.0)\n",
      "(39, 'B-Raf mutations have also been identified in human cancers, including melanoma and thyroid, ovarian and colorectal tumours, although mutation frequency estimates in colorectal tumours vary from 1 to 20%, and have been particularly associated with tumours deficient in mismatch-repair activity (Wellbrock et al, 2004)', 'B-Raf', 'ovarian and colorectal tumours', 0.0)\n",
      "(40, 'B-Raf mutations have also been identified in human cancers, including melanoma and thyroid, ovarian and colorectal tumours, although mutation frequency estimates in colorectal tumours vary from 1 to 20%, and have been particularly associated with tumours deficient in mismatch-repair activity (Wellbrock et al, 2004)', 'B-Raf', 'colorectal tumours', 0.0)\n",
      "(41, 'B-Raf mutations have also been identified in human cancers, including melanoma and thyroid, ovarian and colorectal tumours, although mutation frequency estimates in colorectal tumours vary from 1 to 20%, and have been particularly associated with tumours deficient in mismatch-repair activity (Wellbrock et al, 2004)', 'B-Raf', 'tumours deficient', 0.0)\n",
      "(42, 'The most common B-Raf mutation is a T to A transversion, resulting in a valine to glutamic acid substitution (V600E), present in approximately 90% of B-Raf mutant tumours (Davies et al, 2002), which results in a 500-fold increase in in vitro kinase activity and the induction of cell proliferation and transformation (Wan et al, 2004)', 'B-Raf', 'tumours', 1.0)\n",
      "(43, 'The most common B-Raf mutation is a T to A transversion, resulting in a valine to glutamic acid substitution (V600E), present in approximately 90% of B-Raf mutant tumours (Davies et al, 2002), which results in a 500-fold increase in in vitro kinase activity and the induction of cell proliferation and transformation (Wan et al, 2004)', 'B-Raf', 'tumours', 0.0)\n",
      "(44, 'Although K-Ras and B-Raf mutations are found in the same tumour types, they are thought to be mutually exclusive (Wellbrock et al, 2004), providing distinct but related mechanisms for the activation of K-Ras signalling pathways', 'K-Ras', 'tumour', 1.0)\n",
      "(45, 'Although K-Ras and B-Raf mutations are found in the same tumour types, they are thought to be mutually exclusive (Wellbrock et al, 2004), providing distinct but related mechanisms for the activation of K-Ras signalling pathways', 'B-Raf', 'tumour', 1.0)\n",
      "(46, 'Although K-Ras and B-Raf mutations are found in the same tumour types, they are thought to be mutually exclusive (Wellbrock et al, 2004), providing distinct but related mechanisms for the activation of K-Ras signalling pathways', 'K-Ras', 'tumour', 1.0)\n",
      "(47, 'B-Raf mutation status has also recently been associated with response to cetuximab and panitumumab where, like K-Ras, clinical response is limited to wt B-Raf tumours (Di Nicolantonio et al, 2008).', 'B-Raf', 'B-Raf tumours', 1.0)\n",
      "(48, 'B-Raf mutation status has also recently been associated with response to cetuximab and panitumumab where, like K-Ras, clinical response is limited to wt B-Raf tumours (Di Nicolantonio et al, 2008).', 'K-Ras', 'B-Raf tumours', 1.0)\n",
      "(49, 'As K-Ras mutations in colorectal tumours are clearly important biomarkers of tumour progression and patient survival and also influence response to new generation EGFR antagonists, we have used a variety of experimental approaches to further investigate inter-individual differences in K-Ras mutation burden.', 'K-Ras', 'colorectal tumours', 0.0)\n",
      "(50, 'As K-Ras mutations in colorectal tumours are clearly important biomarkers of tumour progression and patient survival and also influence response to new generation EGFR antagonists, we have used a variety of experimental approaches to further investigate inter-individual differences in K-Ras mutation burden.', 'K-Ras', 'tumour', 0.0)\n",
      "(51, 'As K-Ras mutations in colorectal tumours are clearly important biomarkers of tumour progression and patient survival and also influence response to new generation EGFR antagonists, we have used a variety of experimental approaches to further investigate inter-individual differences in K-Ras mutation burden.', 'EGFR', 'colorectal tumours', 0.0)\n",
      "(52, 'As K-Ras mutations in colorectal tumours are clearly important biomarkers of tumour progression and patient survival and also influence response to new generation EGFR antagonists, we have used a variety of experimental approaches to further investigate inter-individual differences in K-Ras mutation burden.', 'K-Ras', 'colorectal tumours', 0.0)\n",
      "(53, 'As K-Ras mutations in colorectal tumours are clearly important biomarkers of tumour progression and patient survival and also influence response to new generation EGFR antagonists, we have used a variety of experimental approaches to further investigate inter-individual differences in K-Ras mutation burden.', 'EGFR', 'tumour', 0.0)\n",
      "(54, 'As K-Ras mutations in colorectal tumours are clearly important biomarkers of tumour progression and patient survival and also influence response to new generation EGFR antagonists, we have used a variety of experimental approaches to further investigate inter-individual differences in K-Ras mutation burden.', 'K-Ras', 'tumour', 0.0)\n",
      "(55, \"K-Ras was amplified from genomic DNA extracted from normal and tumour tissues (n=96 normal/tumour pairs) from our colorectal patient series (Smith et al, 2002) with the oligonucleotide primers 5'-TTTAATACTTTTTATGTATTTCAGGGTGTTG-3' (300 nM) and 5'-TTACCATCTTTGCTCATCTTTTCTTTAT-3' (300 nM) and Taqman probe 5'-FAM-TGATGCCTTCTATACATTAGTTCGAGAAATTCGAAAA-TAMRA-3' (100 nM), and expression compared with RNAse P by the comparative Ct method, according to the manufacturer's instructions (Applied Biosystems, Warrington, UK).\", 'K-Ras', 'tumour', 1.0)\n",
      "(56, \"K-Ras was amplified from genomic DNA extracted from normal and tumour tissues (n=96 normal/tumour pairs) from our colorectal patient series (Smith et al, 2002) with the oligonucleotide primers 5'-TTTAATACTTTTTATGTATTTCAGGGTGTTG-3' (300 nM) and 5'-TTACCATCTTTGCTCATCTTTTCTTTAT-3' (300 nM) and Taqman probe 5'-FAM-TGATGCCTTCTATACATTAGTTCGAGAAATTCGAAAA-TAMRA-3' (100 nM), and expression compared with RNAse P by the comparative Ct method, according to the manufacturer's instructions (Applied Biosystems, Warrington, UK).\", 'K-Ras', 'tumour', 1.0)\n",
      "(57, \"K-Ras was amplified from genomic DNA extracted from normal and tumour tissues (n=96 normal/tumour pairs) from our colorectal patient series (Smith et al, 2002) with the oligonucleotide primers 5'-TTTAATACTTTTTATGTATTTCAGGGTGTTG-3' (300 nM) and 5'-TTACCATCTTTGCTCATCTTTTCTTTAT-3' (300 nM) and Taqman probe 5'-FAM-TGATGCCTTCTATACATTAGTTCGAGAAATTCGAAAA-TAMRA-3' (100 nM), and expression compared with RNAse P by the comparative Ct method, according to the manufacturer's instructions (Applied Biosystems, Warrington, UK).\", 'K-Ras', 'colorectal', 1.0)\n",
      "(58, 'Meanwhile, 1 mul chromosome 12p BAC RP11-707G18 (Red) (BlueGnome), 1 mul chromosome 12q BAC RP11-89H19 (Green) (BlueGnome) and 9 mul LSI-WCP hybridisation buffer (Vysis) was mixed together for each slide and pipetted onto a 22 x 22 mm coverslip which was placed onto a region of the slide containing tumour tissue and placed onto a heat block at 75 C for 5 min to co-denature the probes and tumour tissue', 'RP11', 'tumour', 1.0)\n",
      "(59, 'Meanwhile, 1 mul chromosome 12p BAC RP11-707G18 (Red) (BlueGnome), 1 mul chromosome 12q BAC RP11-89H19 (Green) (BlueGnome) and 9 mul LSI-WCP hybridisation buffer (Vysis) was mixed together for each slide and pipetted onto a 22 x 22 mm coverslip which was placed onto a region of the slide containing tumour tissue and placed onto a heat block at 75 C for 5 min to co-denature the probes and tumour tissue', 'RP11', 'tumour', 1.0)\n",
      "(60, 'Meanwhile, 1 mul chromosome 12p BAC RP11-707G18 (Red) (BlueGnome), 1 mul chromosome 12q BAC RP11-89H19 (Green) (BlueGnome) and 9 mul LSI-WCP hybridisation buffer (Vysis) was mixed together for each slide and pipetted onto a 22 x 22 mm coverslip which was placed onto a region of the slide containing tumour tissue and placed onto a heat block at 75 C for 5 min to co-denature the probes and tumour tissue', 'RP11', 'tumour', 1.0)\n",
      "(61, 'Meanwhile, 1 mul chromosome 12p BAC RP11-707G18 (Red) (BlueGnome), 1 mul chromosome 12q BAC RP11-89H19 (Green) (BlueGnome) and 9 mul LSI-WCP hybridisation buffer (Vysis) was mixed together for each slide and pipetted onto a 22 x 22 mm coverslip which was placed onto a region of the slide containing tumour tissue and placed onto a heat block at 75 C for 5 min to co-denature the probes and tumour tissue', 'RP11', 'tumour', 1.0)\n",
      "(62, 'We have previously described K-Ras mutations at codons 12, 13 and 61 in a series of 106 unselected colorectal tumours (Smith et al, 2002)', 'K-Ras', 'colorectal tumours', 0.0)\n",
      "(63, 'Mutations at codon 19 (G57T, Leu19Phe) and in B-Raf (V600E) were found in single tumours (Table 2).', 'B-Raf', 'tumours', 1.0)\n",
      "(64, 'To identify additional K-Ras mutations, and to establish the relative frequencies of individual K-Ras mutations in human colorectal tumours, we used WAVE analysis followed by direct sequencing to screen the same tumour series for mutations in K-Ras exon 3 and exon 4B, the most common exon 4 splice variant', 'K-Ras', 'colorectal tumours', 1.0)\n",
      "(65, 'To identify additional K-Ras mutations, and to establish the relative frequencies of individual K-Ras mutations in human colorectal tumours, we used WAVE analysis followed by direct sequencing to screen the same tumour series for mutations in K-Ras exon 3 and exon 4B, the most common exon 4 splice variant', 'K-Ras', 'tumour', 1.0)\n",
      "(66, 'To identify additional K-Ras mutations, and to establish the relative frequencies of individual K-Ras mutations in human colorectal tumours, we used WAVE analysis followed by direct sequencing to screen the same tumour series for mutations in K-Ras exon 3 and exon 4B, the most common exon 4 splice variant', 'K-Ras', 'colorectal tumours', 1.0)\n",
      "(156, 'Characterization of a novel oncogenic K-ras mutation in colon cancer', 'K-ras', 'colon cancer', 1.0)\n",
      "(67, 'To identify additional K-Ras mutations, and to establish the relative frequencies of individual K-Ras mutations in human colorectal tumours, we used WAVE analysis followed by direct sequencing to screen the same tumour series for mutations in K-Ras exon 3 and exon 4B, the most common exon 4 splice variant', 'K-Ras', 'tumour', 1.0)\n",
      "(68, 'To identify additional K-Ras mutations, and to establish the relative frequencies of individual K-Ras mutations in human colorectal tumours, we used WAVE analysis followed by direct sequencing to screen the same tumour series for mutations in K-Ras exon 3 and exon 4B, the most common exon 4 splice variant', 'K-Ras', 'colorectal tumours', 1.0)\n",
      "(69, 'To identify additional K-Ras mutations, and to establish the relative frequencies of individual K-Ras mutations in human colorectal tumours, we used WAVE analysis followed by direct sequencing to screen the same tumour series for mutations in K-Ras exon 3 and exon 4B, the most common exon 4 splice variant', 'K-Ras', 'tumour', 1.0)\n",
      "(70, 'To determine the frequencies of the codon 117, 146 and 164 K-Ras mutations and the codon 173 SNP, PCR-RFLP assays were designed to permit rapid screening of blood and tumour DNA (Appendix A)', 'K-Ras', 'tumour', 1.0)\n",
      "(71, 'No tumours with codon 19, 117, 146 or 146 mutations had additional K-Ras mutations in codons 12, 13 or 61', 'K-Ras', 'tumours', 1.0)\n",
      "(72, 'To investigate whether altered K-Ras activity could additionally result from gene amplification, a K-Ras gene copy number assay was designed and carried out as described in Materials and Methods and K-Ras copy number compared in normal and tumour tissues', 'K-Ras', 'tumour', 1.0)\n",
      "(73, 'To investigate whether altered K-Ras activity could additionally result from gene amplification, a K-Ras gene copy number assay was designed and carried out as described in Materials and Methods and K-Ras copy number compared in normal and tumour tissues', 'K-Ras', 'tumour', 1.0)\n",
      "(74, 'To investigate whether altered K-Ras activity could additionally result from gene amplification, a K-Ras gene copy number assay was designed and carried out as described in Materials and Methods and K-Ras copy number compared in normal and tumour tissues', 'K-Ras', 'tumour', 1.0)\n",
      "(75, 'In total, 2 of the 96 tumour pairs analysed (2.1%) showed tumour-specific copy number increases (Figure 6), one (study code 1264, wt K-Ras) with 4 tumour copies of K-Ras and the other (study code 1233, wt K-Ras) with 27 copies of the K-Ras gene', 'K-Ras', 'tumour', 1.0)\n",
      "(76, 'In total, 2 of the 96 tumour pairs analysed (2.1%) showed tumour-specific copy number increases (Figure 6), one (study code 1264, wt K-Ras) with 4 tumour copies of K-Ras and the other (study code 1233, wt K-Ras) with 27 copies of the K-Ras gene', 'K-Ras', 'tumour', 1.0)\n",
      "(77, 'In total, 2 of the 96 tumour pairs analysed (2.1%) showed tumour-specific copy number increases (Figure 6), one (study code 1264, wt K-Ras) with 4 tumour copies of K-Ras and the other (study code 1233, wt K-Ras) with 27 copies of the K-Ras gene', 'K-Ras', 'tumour', 1.0)\n",
      "(78, 'In total, 2 of the 96 tumour pairs analysed (2.1%) showed tumour-specific copy number increases (Figure 6), one (study code 1264, wt K-Ras) with 4 tumour copies of K-Ras and the other (study code 1233, wt K-Ras) with 27 copies of the K-Ras gene', 'K-Ras', 'tumour', 1.0)\n",
      "(79, 'In total, 2 of the 96 tumour pairs analysed (2.1%) showed tumour-specific copy number increases (Figure 6), one (study code 1264, wt K-Ras) with 4 tumour copies of K-Ras and the other (study code 1233, wt K-Ras) with 27 copies of the K-Ras gene', 'K-Ras', 'tumour', 1.0)\n",
      "(80, 'In total, 2 of the 96 tumour pairs analysed (2.1%) showed tumour-specific copy number increases (Figure 6), one (study code 1264, wt K-Ras) with 4 tumour copies of K-Ras and the other (study code 1233, wt K-Ras) with 27 copies of the K-Ras gene', 'K-Ras', 'tumour', 1.0)\n",
      "(81, 'In total, 2 of the 96 tumour pairs analysed (2.1%) showed tumour-specific copy number increases (Figure 6), one (study code 1264, wt K-Ras) with 4 tumour copies of K-Ras and the other (study code 1233, wt K-Ras) with 27 copies of the K-Ras gene', 'K-Ras', 'tumour', 1.0)\n",
      "(82, 'In total, 2 of the 96 tumour pairs analysed (2.1%) showed tumour-specific copy number increases (Figure 6), one (study code 1264, wt K-Ras) with 4 tumour copies of K-Ras and the other (study code 1233, wt K-Ras) with 27 copies of the K-Ras gene', 'K-Ras', 'tumour', 1.0)\n",
      "(83, 'In total, 2 of the 96 tumour pairs analysed (2.1%) showed tumour-specific copy number increases (Figure 6), one (study code 1264, wt K-Ras) with 4 tumour copies of K-Ras and the other (study code 1233, wt K-Ras) with 27 copies of the K-Ras gene', 'K-Ras', 'tumour', 1.0)\n",
      "(84, 'In total, 2 of the 96 tumour pairs analysed (2.1%) showed tumour-specific copy number increases (Figure 6), one (study code 1264, wt K-Ras) with 4 tumour copies of K-Ras and the other (study code 1233, wt K-Ras) with 27 copies of the K-Ras gene', 'K-Ras', 'tumour', 1.0)\n",
      "(85, 'In total, 2 of the 96 tumour pairs analysed (2.1%) showed tumour-specific copy number increases (Figure 6), one (study code 1264, wt K-Ras) with 4 tumour copies of K-Ras and the other (study code 1233, wt K-Ras) with 27 copies of the K-Ras gene', 'K-Ras', 'tumour', 1.0)\n",
      "(86, 'In total, 2 of the 96 tumour pairs analysed (2.1%) showed tumour-specific copy number increases (Figure 6), one (study code 1264, wt K-Ras) with 4 tumour copies of K-Ras and the other (study code 1233, wt K-Ras) with 27 copies of the K-Ras gene', 'K-Ras', 'tumour', 1.0)\n",
      "(87, 'Figure 7C illustrates a tumour sample (study code 1271) in which the K-Ras gene is not amplified (red fluorescence=green fluorescence), whereas Figure 7D illustrates a colorectal tumour (study code 1264) with K-Ras gene amplification (red fluorescence>green fluorescence)', 'K-Ras', 'tumour', 1.0)\n",
      "(88, 'Figure 7C illustrates a tumour sample (study code 1271) in which the K-Ras gene is not amplified (red fluorescence=green fluorescence), whereas Figure 7D illustrates a colorectal tumour (study code 1264) with K-Ras gene amplification (red fluorescence>green fluorescence)', 'K-Ras', 'tumour', 1.0)\n",
      "(89, 'Figure 7C illustrates a tumour sample (study code 1271) in which the K-Ras gene is not amplified (red fluorescence=green fluorescence), whereas Figure 7D illustrates a colorectal tumour (study code 1264) with K-Ras gene amplification (red fluorescence>green fluorescence)', 'K-Ras', 'colorectal tumour', 1.0)\n",
      "(90, 'Figure 7C illustrates a tumour sample (study code 1271) in which the K-Ras gene is not amplified (red fluorescence=green fluorescence), whereas Figure 7D illustrates a colorectal tumour (study code 1264) with K-Ras gene amplification (red fluorescence>green fluorescence)', 'K-Ras', 'colorectal tumour', 1.0)\n",
      "(91, 'Recent clinical data has shown a compelling association between K-Ras and B-Raf mutation status and response to EGFR blockade by cetuximab and panitumumab in the treatment of metastatic colorectal tumours (Benvenuti et al, 2007; Khambata-Ford et al, 2007; Freeman et al, 2008; Karapetis et al, 2008; Ramos et al, 2008)', 'K-Ras', 'colorectal tumours', 0.0)\n",
      "(203, 'Glucose deprivation contributes to the development of KRAS pathway mutations in tumor cells', 'KRAS', 'Glucose deprivation', 1.0)\n",
      "(92, 'Recent clinical data has shown a compelling association between K-Ras and B-Raf mutation status and response to EGFR blockade by cetuximab and panitumumab in the treatment of metastatic colorectal tumours (Benvenuti et al, 2007; Khambata-Ford et al, 2007; Freeman et al, 2008; Karapetis et al, 2008; Ramos et al, 2008)', 'B-Raf', 'colorectal tumours', 0.0)\n",
      "(93, 'Recent clinical data has shown a compelling association between K-Ras and B-Raf mutation status and response to EGFR blockade by cetuximab and panitumumab in the treatment of metastatic colorectal tumours (Benvenuti et al, 2007; Khambata-Ford et al, 2007; Freeman et al, 2008; Karapetis et al, 2008; Ramos et al, 2008)', 'EGFR', 'colorectal tumours', 0.0)\n",
      "(94, 'K-Ras mutations in colorectal tumours have been extensively documented, where recent meta analysis of more than 3000 tumours estimates an average K-Ras mutation frequency of 34.8% (Andreyev et al, 2001)', 'K-Ras', 'colorectal tumours', 0.0)\n",
      "(95, 'K-Ras mutations in colorectal tumours have been extensively documented, where recent meta analysis of more than 3000 tumours estimates an average K-Ras mutation frequency of 34.8% (Andreyev et al, 2001)', 'K-Ras', 'tumours', 0.0)\n",
      "(96, 'K-Ras mutations in colorectal tumours have been extensively documented, where recent meta analysis of more than 3000 tumours estimates an average K-Ras mutation frequency of 34.8% (Andreyev et al, 2001)', 'K-Ras', 'colorectal tumours', 0.0)\n",
      "(97, 'K-Ras mutations in colorectal tumours have been extensively documented, where recent meta analysis of more than 3000 tumours estimates an average K-Ras mutation frequency of 34.8% (Andreyev et al, 2001)', 'K-Ras', 'tumours', 0.0)\n",
      "(98, \"Although additional K-Ras mutations have been reported outwith these 'hotspot' codons, to our knowledge, there has been no systematic analysis of K-Ras mutation frequency in human colorectal tumours\", 'K-Ras', 'colorectal tumours', 0.0)\n",
      "(99, \"Although additional K-Ras mutations have been reported outwith these 'hotspot' codons, to our knowledge, there has been no systematic analysis of K-Ras mutation frequency in human colorectal tumours\", 'K-Ras', 'colorectal tumours', 0.0)\n",
      "(100, \"Our use of 'WAVE' denaturing HPLC (dHPLC) analysis (Kuklin et al, 1997) allowed us to carry out a rapid, systematic analysis of the entire K-Ras coding sequence to identify colorectal tumours with sequence changes, the nature which were subsequently confirmed by conventional dideoxy sequencing.\", 'K-Ras', 'colorectal tumours', 1.0)\n",
      "(101, 'Our data clearly demonstrates that K-Ras mutations in human colorectal tumours are likely to be significantly underestimated if only hotspot codons are analysed', 'K-Ras', 'colorectal tumours', 1.0)\n",
      "(102, 'Current data from the COSMIC database, which summarises literature data and the ongoing Sanger cancer-genome project describes 29 K-Ras colorectal tumour mutations outwith codons 12, 13 and 61, and more than 11 800 mutations at these hotspot codons (http://www.sanger.ac.uk/genetics/CGP/), but does not describe associated phenotypes', 'K-Ras', 'cancer', 1.0)\n",
      "(103, 'Current data from the COSMIC database, which summarises literature data and the ongoing Sanger cancer-genome project describes 29 K-Ras colorectal tumour mutations outwith codons 12, 13 and 61, and more than 11 800 mutations at these hotspot codons (http://www.sanger.ac.uk/genetics/CGP/), but does not describe associated phenotypes', 'K-Ras', 'colorectal tumour', 1.0)\n",
      "(104, 'K-Ras codon 146 mutations have been described in a human colorectal tumour cell line (Higashi et al, 1990), in mice with thymic lymphomas (Sloan et al, 1990) and in a single human colorectal tumour case report (Orita et al, 1991)', 'K-Ras', 'colorectal tumour', 0.0)\n",
      "(105, 'K-Ras codon 146 mutations have been described in a human colorectal tumour cell line (Higashi et al, 1990), in mice with thymic lymphomas (Sloan et al, 1990) and in a single human colorectal tumour case report (Orita et al, 1991)', 'K-Ras', 'thymic lymphomas', 0.0)\n",
      "(106, 'K-Ras codon 146 mutations have been described in a human colorectal tumour cell line (Higashi et al, 1990), in mice with thymic lymphomas (Sloan et al, 1990) and in a single human colorectal tumour case report (Orita et al, 1991)', 'K-Ras', 'colorectal tumour', 0.0)\n",
      "(107, 'Consistent with our own data, Tyner et al (2009) reported an oncogenic phenotype associated with the K-Ras 146 mutation in leukaemia and Loupakis et al (2009) have recently reported that a patient with metastatic colorectal cancer with a K-Ras 146 mutation was resistant to cetuximab.', 'K-Ras', 'leukaemia', 1.0)\n",
      "(108, 'Consistent with our own data, Tyner et al (2009) reported an oncogenic phenotype associated with the K-Ras 146 mutation in leukaemia and Loupakis et al (2009) have recently reported that a patient with metastatic colorectal cancer with a K-Ras 146 mutation was resistant to cetuximab.', 'K-Ras', 'colorectal cancer', 1.0)\n",
      "(109, 'Consistent with our own data, Tyner et al (2009) reported an oncogenic phenotype associated with the K-Ras 146 mutation in leukaemia and Loupakis et al (2009) have recently reported that a patient with metastatic colorectal cancer with a K-Ras 146 mutation was resistant to cetuximab.', 'K-Ras', 'leukaemia', 1.0)\n",
      "(110, 'Consistent with our own data, Tyner et al (2009) reported an oncogenic phenotype associated with the K-Ras 146 mutation in leukaemia and Loupakis et al (2009) have recently reported that a patient with metastatic colorectal cancer with a K-Ras 146 mutation was resistant to cetuximab.', 'K-Ras', 'colorectal cancer', 1.0)\n",
      "(111, 'There is only a single report of K-Ras codon 19 mutations in human colorectal tumours (Akagi et al, 2007), in which in vitro elevation of active Ras-GTP levels, anchorage-independent growth and increased tumourigenicity in nude mice was demonstrated', 'K-Ras', 'colorectal tumours', 1.0)\n",
      "(112, 'A codon 19 mutation has additionally been described in H-Ras in a human pituitary carcinoma metastases (Pei et al, 1994)', 'H-Ras', 'pituitary carcinoma metastases', 1.0)\n",
      "(113, 'There are two recent reports of codon 117 mutations - the first in K-Ras in human colorectal tumours (Wojcik et al, 2008) and the second in H-Ras, associated with activation of the RAS-MAPK pathway and the mental retardation Costello syndrome (Denayer et al, 2008)', 'K-Ras', 'colorectal tumours', 0.0)\n",
      "(114, 'There are two recent reports of codon 117 mutations - the first in K-Ras in human colorectal tumours (Wojcik et al, 2008) and the second in H-Ras, associated with activation of the RAS-MAPK pathway and the mental retardation Costello syndrome (Denayer et al, 2008)', 'K-Ras', 'mental retardation Costello syndrome', 0.0)\n",
      "(115, 'There are two recent reports of codon 117 mutations - the first in K-Ras in human colorectal tumours (Wojcik et al, 2008) and the second in H-Ras, associated with activation of the RAS-MAPK pathway and the mental retardation Costello syndrome (Denayer et al, 2008)', 'H-Ras', 'colorectal tumours', 0.0)\n",
      "(116, 'There are two recent reports of codon 117 mutations - the first in K-Ras in human colorectal tumours (Wojcik et al, 2008) and the second in H-Ras, associated with activation of the RAS-MAPK pathway and the mental retardation Costello syndrome (Denayer et al, 2008)', 'MAPK', 'colorectal tumours', 0.0)\n",
      "(117, 'There are two recent reports of codon 117 mutations - the first in K-Ras in human colorectal tumours (Wojcik et al, 2008) and the second in H-Ras, associated with activation of the RAS-MAPK pathway and the mental retardation Costello syndrome (Denayer et al, 2008)', 'H-Ras', 'mental retardation Costello syndrome', 0.0)\n",
      "(118, 'There are two recent reports of codon 117 mutations - the first in K-Ras in human colorectal tumours (Wojcik et al, 2008) and the second in H-Ras, associated with activation of the RAS-MAPK pathway and the mental retardation Costello syndrome (Denayer et al, 2008)', 'MAPK', 'mental retardation Costello syndrome', 0.0)\n",
      "(119, 'There are no reports of K-Ras 164 mutations, and mutations at this codon were not identified by Sanger cancer genome project investigators.', 'K-Ras', 'cancer', 1.0)\n",
      "(120, 'Recent clinical data convincingly associates response to anti-EGFR therapies with K-Ras mutation status in patients with metastatic colorectal cancer where, logically, response is preferentially observed in K-Ras wt tumours (Lievre et al, 2006; Benvenuti et al, 2007; Khambata-Ford et al, 2007; Freeman et al, 2008; Karapetis et al, 2008; Ramos et al, 2008; Loupakis et al, 2009; Van Cutsem et al, 2009) which retain the ability to respond to EGFR blockade', 'EGFR', 'colorectal cancer', 1.0)\n",
      "(121, 'Recent clinical data convincingly associates response to anti-EGFR therapies with K-Ras mutation status in patients with metastatic colorectal cancer where, logically, response is preferentially observed in K-Ras wt tumours (Lievre et al, 2006; Benvenuti et al, 2007; Khambata-Ford et al, 2007; Freeman et al, 2008; Karapetis et al, 2008; Ramos et al, 2008; Loupakis et al, 2009; Van Cutsem et al, 2009) which retain the ability to respond to EGFR blockade', 'EGFR', 'K-Ras wt tumours', 1.0)\n",
      "(122, 'Recent clinical data convincingly associates response to anti-EGFR therapies with K-Ras mutation status in patients with metastatic colorectal cancer where, logically, response is preferentially observed in K-Ras wt tumours (Lievre et al, 2006; Benvenuti et al, 2007; Khambata-Ford et al, 2007; Freeman et al, 2008; Karapetis et al, 2008; Ramos et al, 2008; Loupakis et al, 2009; Van Cutsem et al, 2009) which retain the ability to respond to EGFR blockade', 'K-Ras', 'colorectal cancer', 1.0)\n",
      "(123, 'Recent clinical data convincingly associates response to anti-EGFR therapies with K-Ras mutation status in patients with metastatic colorectal cancer where, logically, response is preferentially observed in K-Ras wt tumours (Lievre et al, 2006; Benvenuti et al, 2007; Khambata-Ford et al, 2007; Freeman et al, 2008; Karapetis et al, 2008; Ramos et al, 2008; Loupakis et al, 2009; Van Cutsem et al, 2009) which retain the ability to respond to EGFR blockade', 'K-Ras', 'K-Ras wt tumours', 1.0)\n",
      "(124, 'Recent clinical data convincingly associates response to anti-EGFR therapies with K-Ras mutation status in patients with metastatic colorectal cancer where, logically, response is preferentially observed in K-Ras wt tumours (Lievre et al, 2006; Benvenuti et al, 2007; Khambata-Ford et al, 2007; Freeman et al, 2008; Karapetis et al, 2008; Ramos et al, 2008; Loupakis et al, 2009; Van Cutsem et al, 2009) which retain the ability to respond to EGFR blockade', 'EGFR', 'colorectal cancer', 0.0)\n",
      "(125, 'Recent clinical data convincingly associates response to anti-EGFR therapies with K-Ras mutation status in patients with metastatic colorectal cancer where, logically, response is preferentially observed in K-Ras wt tumours (Lievre et al, 2006; Benvenuti et al, 2007; Khambata-Ford et al, 2007; Freeman et al, 2008; Karapetis et al, 2008; Ramos et al, 2008; Loupakis et al, 2009; Van Cutsem et al, 2009) which retain the ability to respond to EGFR blockade', 'EGFR', 'K-Ras wt tumours', 1.0)\n",
      "(126, \"Our description of K-Ras gene amplification in a subset of wt tumours is of particular interest in this regard as increased K-Ras gene copy number may lead to a more active 'mutation'-like phenotype\", 'K-Ras', 'tumours', 1.0)\n",
      "(127, \"Our description of K-Ras gene amplification in a subset of wt tumours is of particular interest in this regard as increased K-Ras gene copy number may lead to a more active 'mutation'-like phenotype\", 'K-Ras', 'tumours', 1.0)\n",
      "(128, 'Consistent with our own data, DUSP4 expression has previously been shown to be increased in pancreatic tumours with K-Ras mutations (Yip-Schneider et al, 2001) and DUSP6 expression increased in a variety of tumour types with mutations in Ras or Raf pathway genes (Croonquist et al, 2003; Warmka et al, 2004; Bloethner et al, 2005)', 'DUSP4', 'pancreatic tumours', 0.0)\n",
      "(129, 'Consistent with our own data, DUSP4 expression has previously been shown to be increased in pancreatic tumours with K-Ras mutations (Yip-Schneider et al, 2001) and DUSP6 expression increased in a variety of tumour types with mutations in Ras or Raf pathway genes (Croonquist et al, 2003; Warmka et al, 2004; Bloethner et al, 2005)', 'DUSP4', 'tumour', 0.0)\n",
      "(130, 'Consistent with our own data, DUSP4 expression has previously been shown to be increased in pancreatic tumours with K-Ras mutations (Yip-Schneider et al, 2001) and DUSP6 expression increased in a variety of tumour types with mutations in Ras or Raf pathway genes (Croonquist et al, 2003; Warmka et al, 2004; Bloethner et al, 2005)', 'K-Ras', 'pancreatic tumours', 0.0)\n",
      "(131, 'Consistent with our own data, DUSP4 expression has previously been shown to be increased in pancreatic tumours with K-Ras mutations (Yip-Schneider et al, 2001) and DUSP6 expression increased in a variety of tumour types with mutations in Ras or Raf pathway genes (Croonquist et al, 2003; Warmka et al, 2004; Bloethner et al, 2005)', 'DUSP6', 'pancreatic tumours', 0.0)\n",
      "(132, 'Consistent with our own data, DUSP4 expression has previously been shown to be increased in pancreatic tumours with K-Ras mutations (Yip-Schneider et al, 2001) and DUSP6 expression increased in a variety of tumour types with mutations in Ras or Raf pathway genes (Croonquist et al, 2003; Warmka et al, 2004; Bloethner et al, 2005)', 'Raf', 'pancreatic tumours', 0.0)\n",
      "(133, 'Consistent with our own data, DUSP4 expression has previously been shown to be increased in pancreatic tumours with K-Ras mutations (Yip-Schneider et al, 2001) and DUSP6 expression increased in a variety of tumour types with mutations in Ras or Raf pathway genes (Croonquist et al, 2003; Warmka et al, 2004; Bloethner et al, 2005)', 'K-Ras', 'tumour', 0.0)\n",
      "(134, 'Consistent with our own data, DUSP4 expression has previously been shown to be increased in pancreatic tumours with K-Ras mutations (Yip-Schneider et al, 2001) and DUSP6 expression increased in a variety of tumour types with mutations in Ras or Raf pathway genes (Croonquist et al, 2003; Warmka et al, 2004; Bloethner et al, 2005)', 'DUSP6', 'tumour', 0.0)\n",
      "(135, 'Consistent with our own data, DUSP4 expression has previously been shown to be increased in pancreatic tumours with K-Ras mutations (Yip-Schneider et al, 2001) and DUSP6 expression increased in a variety of tumour types with mutations in Ras or Raf pathway genes (Croonquist et al, 2003; Warmka et al, 2004; Bloethner et al, 2005)', 'Raf', 'tumour', 0.0)\n",
      "(157, 'Oncogenic activation of the RAS/RAF signaling pathway impairs the response of metastatic colorectal cancers to anti-epidermal growth factor receptor antibody therapies', 'RAF', 'colorectal cancers', 1.0)\n",
      "(136, 'Of particular interest, however, was our observation that certain signalling cascades, including those mediated by Igf1 and Vegf, were differentially activated by cluster 1 and cluster 2 K-Ras mutants, suggesting that not only the presence but the specific molecular characteristics of individual K-Ras mutations may be important determinants of both tumour progression and treatment response', 'Igf1', 'tumour', 1.0)\n",
      "(137, 'Of particular interest, however, was our observation that certain signalling cascades, including those mediated by Igf1 and Vegf, were differentially activated by cluster 1 and cluster 2 K-Ras mutants, suggesting that not only the presence but the specific molecular characteristics of individual K-Ras mutations may be important determinants of both tumour progression and treatment response', 'Vegf', 'tumour', 1.0)\n",
      "(138, 'Of particular interest, however, was our observation that certain signalling cascades, including those mediated by Igf1 and Vegf, were differentially activated by cluster 1 and cluster 2 K-Ras mutants, suggesting that not only the presence but the specific molecular characteristics of individual K-Ras mutations may be important determinants of both tumour progression and treatment response', 'K-Ras', 'tumour', 1.0)\n",
      "(139, 'Of particular interest, however, was our observation that certain signalling cascades, including those mediated by Igf1 and Vegf, were differentially activated by cluster 1 and cluster 2 K-Ras mutants, suggesting that not only the presence but the specific molecular characteristics of individual K-Ras mutations may be important determinants of both tumour progression and treatment response', 'K-Ras', 'tumour', 1.0)\n",
      "(140, 'For example, recent data suggests that glucose deprivation can promote the acquisition of K-Ras mutations in human tumours and reports upregulation of the glucose transporter GLUT-1 in K-Ras mutant tumours (Yun et al, 2009)', 'K-Ras', 'glucose deprivation', 1.0)\n",
      "(141, 'For example, recent data suggests that glucose deprivation can promote the acquisition of K-Ras mutations in human tumours and reports upregulation of the glucose transporter GLUT-1 in K-Ras mutant tumours (Yun et al, 2009)', 'GLUT-1', 'glucose deprivation', 1.0)\n",
      "(142, 'For example, recent data suggests that glucose deprivation can promote the acquisition of K-Ras mutations in human tumours and reports upregulation of the glucose transporter GLUT-1 in K-Ras mutant tumours (Yun et al, 2009)', 'K-Ras', 'glucose deprivation', 1.0)\n",
      "(143, 'For example, recent data suggests that glucose deprivation can promote the acquisition of K-Ras mutations in human tumours and reports upregulation of the glucose transporter GLUT-1 in K-Ras mutant tumours (Yun et al, 2009)', 'K-Ras', 'tumours', 1.0)\n",
      "(144, 'For example, recent data suggests that glucose deprivation can promote the acquisition of K-Ras mutations in human tumours and reports upregulation of the glucose transporter GLUT-1 in K-Ras mutant tumours (Yun et al, 2009)', 'K-Ras', 'tumours', 1.0)\n",
      "(145, 'For example, recent data suggests that glucose deprivation can promote the acquisition of K-Ras mutations in human tumours and reports upregulation of the glucose transporter GLUT-1 in K-Ras mutant tumours (Yun et al, 2009)', 'GLUT-1', 'tumours', 1.0)\n",
      "(146, 'For example, recent data suggests that glucose deprivation can promote the acquisition of K-Ras mutations in human tumours and reports upregulation of the glucose transporter GLUT-1 in K-Ras mutant tumours (Yun et al, 2009)', 'K-Ras', 'tumours', 1.0)\n",
      "(147, 'For example, recent data suggests that glucose deprivation can promote the acquisition of K-Ras mutations in human tumours and reports upregulation of the glucose transporter GLUT-1 in K-Ras mutant tumours (Yun et al, 2009)', 'GLUT-1', 'tumours', 1.0)\n",
      "(148, 'For example, recent data suggests that glucose deprivation can promote the acquisition of K-Ras mutations in human tumours and reports upregulation of the glucose transporter GLUT-1 in K-Ras mutant tumours (Yun et al, 2009)', 'K-Ras', 'tumours', 1.0)\n",
      "(149, 'Current clinical data is sufficiently compelling that K-Ras mutation testing to facilitate the selection of patients most likely to benefit from anti-EGFR therapy is increasingly recommended in clinical practise (McNeill, 2008; van Krieken and Tol, 2009), and has resulted in a recent formal recommendation from the National Comprehensive Cancer Network that patients with K-Ras tumour mutations should not be treated with cetuximab or panitumumab (McNeill, 2008)', 'K-Ras', 'van Krieken', 0.0)\n",
      "(150, 'Current clinical data is sufficiently compelling that K-Ras mutation testing to facilitate the selection of patients most likely to benefit from anti-EGFR therapy is increasingly recommended in clinical practise (McNeill, 2008; van Krieken and Tol, 2009), and has resulted in a recent formal recommendation from the National Comprehensive Cancer Network that patients with K-Ras tumour mutations should not be treated with cetuximab or panitumumab (McNeill, 2008)', 'K-Ras', 'K-Ras tumour', 0.0)\n",
      "(151, 'Current clinical data is sufficiently compelling that K-Ras mutation testing to facilitate the selection of patients most likely to benefit from anti-EGFR therapy is increasingly recommended in clinical practise (McNeill, 2008; van Krieken and Tol, 2009), and has resulted in a recent formal recommendation from the National Comprehensive Cancer Network that patients with K-Ras tumour mutations should not be treated with cetuximab or panitumumab (McNeill, 2008)', 'EGFR', 'van Krieken', 0.0)\n",
      "(152, 'Current clinical data is sufficiently compelling that K-Ras mutation testing to facilitate the selection of patients most likely to benefit from anti-EGFR therapy is increasingly recommended in clinical practise (McNeill, 2008; van Krieken and Tol, 2009), and has resulted in a recent formal recommendation from the National Comprehensive Cancer Network that patients with K-Ras tumour mutations should not be treated with cetuximab or panitumumab (McNeill, 2008)', 'EGFR', 'K-Ras tumour', 0.0)\n",
      "(153, 'Although the clinical implications of non-hotspot K-Ras mutations requires further validation both as prognostic markers and therapeutic targets, our data suggests that future analysis of K-Ras mutations and quantitation of mutation burden in colorectal tumours should not be limited to previously described mutation hotspots and should additionally consider the unique molecular signatures associated with individual K-Ras mutations.', 'K-Ras', 'colorectal tumours', 1.0)\n",
      "(154, 'Although the clinical implications of non-hotspot K-Ras mutations requires further validation both as prognostic markers and therapeutic targets, our data suggests that future analysis of K-Ras mutations and quantitation of mutation burden in colorectal tumours should not be limited to previously described mutation hotspots and should additionally consider the unique molecular signatures associated with individual K-Ras mutations.', 'K-Ras', 'colorectal tumours', 1.0)\n",
      "(155, 'Although the clinical implications of non-hotspot K-Ras mutations requires further validation both as prognostic markers and therapeutic targets, our data suggests that future analysis of K-Ras mutations and quantitation of mutation burden in colorectal tumours should not be limited to previously described mutation hotspots and should additionally consider the unique molecular signatures associated with individual K-Ras mutations.', 'K-Ras', 'colorectal tumours', 1.0)\n",
      "(158, 'Oncogenic activation of the RAS/RAF signaling pathway impairs the response of metastatic colorectal cancers to anti-epidermal growth factor receptor antibody therapies', 'epidermal growth factor receptor', 'colorectal cancers', 1.0)\n",
      "(159, 'Effect of common B-RAF and N-RAS mutations on global gene expression in melanoma cell lines', 'B-RAF', 'melanoma', 1.0)\n",
      "(160, 'Effect of common B-RAF and N-RAS mutations on global gene expression in melanoma cell lines', 'N-RAS', 'melanoma', 1.0)\n",
      "(161, 'The prognostic significance of K-ras, p53, and APC mutations in colorectal carcinoma', 'K-ras', 'APC', 1.0)\n",
      "(162, 'The prognostic significance of K-ras, p53, and APC mutations in colorectal carcinoma', 'K-ras', 'colorectal carcinoma', 1.0)\n",
      "(163, 'The prognostic significance of K-ras, p53, and APC mutations in colorectal carcinoma', 'p53', 'APC', 1.0)\n",
      "(164, 'The prognostic significance of K-ras, p53, and APC mutations in colorectal carcinoma', 'p53', 'colorectal carcinoma', 1.0)\n",
      "(165, 'Gene profiling of a myeloma cell line reveals similarities and unique signatures among IL-6 response, N-ras-activating mutations, and coculture with bone marrow stromal cells', 'IL-6', 'myeloma', 1.0)\n",
      "(166, 'Gene profiling of a myeloma cell line reveals similarities and unique signatures among IL-6 response, N-ras-activating mutations, and coculture with bone marrow stromal cells', 'N-ras', 'myeloma', 1.0)\n",
      "(167, 'Mutations of the BRAF gene in human cancer', 'BRAF', 'cancer', 1.0)\n",
      "(168, 'Mutation analysis in Costello syndrome: functional and structural characterization of the HRAS p.Lys117Arg mutation', 'HRAS', 'Costello syndrome', 1.0)\n",
      "(169, 'Wild-type BRAF is required for response to panitumumab or cetuximab in metastatic colorectal cancer', 'BRAF', 'colorectal cancer', 1.0)\n",
      "(170, 'Recurrent KRAS codon 146 mutations in human colorectal cancer', 'KRAS', 'colorectal cancer', 1.0)\n",
      "(171, 'K-Ras mutations and treatment outcome in colorectal cancer patients receiving exclusive fluoropyrimidine therapy', 'K-Ras', 'colorectal cancer', 1.0)\n",
      "(172, 'Association of K-ras mutational status and clinical outcomes in patients with metastatic colorectal cancer receiving panitumumab alone', 'K-ras', 'colorectal cancer', 1.0)\n",
      "(173, 'Genome profiling of chronic myelomonocytic leukemia: frequent alterations of RAS and RUNX1 genes', 'RUNX1', 'chronic myelomonocytic leukemia', 1.0)\n",
      "(174, 'K-ras mutations and benefit from cetuximab in advanced colorectal cancer', 'K-ras', 'colorectal cancer', 1.0)\n",
      "(175, 'Expression of epiregulin and amphiregulin and K-ras mutation status predict disease control in metastatic colorectal cancer patients treated with cetuximab', 'epiregulin', 'colorectal cancer', 1.0)\n",
      "(176, 'Expression of epiregulin and amphiregulin and K-ras mutation status predict disease control in metastatic colorectal cancer patients treated with cetuximab', 'amphiregulin', 'colorectal cancer', 1.0)\n",
      "(177, 'Expression of epiregulin and amphiregulin and K-ras mutation status predict disease control in metastatic colorectal cancer patients treated with cetuximab', 'K-ras', 'colorectal cancer', 1.0)\n",
      "(178, 'Mutations of APC, K-ras, and p53 are associated with specific chromosomal aberrations in colorectal adenocarcinomas', 'K-ras', 'APC', 1.0)\n",
      "(179, 'Mutations of APC, K-ras, and p53 are associated with specific chromosomal aberrations in colorectal adenocarcinomas', 'p53', 'APC', 1.0)\n",
      "(180, 'Mutations of APC, K-ras, and p53 are associated with specific chromosomal aberrations in colorectal adenocarcinomas', 'K-ras', 'colorectal adenocarcinomas', 1.0)\n",
      "(181, 'Mutations of APC, K-ras, and p53 are associated with specific chromosomal aberrations in colorectal adenocarcinomas', 'p53', 'colorectal adenocarcinomas', 1.0)\n",
      "(182, 'KRAS mutation status is predictive of response to cetuximab therapy in colorectal cancer', 'KRAS', 'colorectal cancer', 1.0)\n",
      "(183, 'KRAS codon 61, 146 and BRAF mutations predict resistance to cetuximab plus irinotecan in KRAS codon 12 and 13 wild-type metastatic colorectal cancer', 'KRAS', 'colorectal cancer', 1.0)\n",
      "(184, 'KRAS codon 61, 146 and BRAF mutations predict resistance to cetuximab plus irinotecan in KRAS codon 12 and 13 wild-type metastatic colorectal cancer', 'BRAF', 'colorectal cancer', 1.0)\n",
      "(185, 'KRAS codon 61, 146 and BRAF mutations predict resistance to cetuximab plus irinotecan in KRAS codon 12 and 13 wild-type metastatic colorectal cancer', 'KRAS', 'colorectal cancer', 1.0)\n",
      "(186, 'K-Ras mutations are changingpractice in advanced colorectal cancer', 'K-Ras', 'colorectal cancer', 1.0)\n",
      "(187, 'Concurrent mutations of K-ras oncogene at codons 12 and 22 in colon cancer', 'K-ras', 'colon cancer', 1.0)\n",
      "(188, 'A novel point mutation at codon 146 of the K-ras gene in a human colorectal cancer identified by the polymerase chain reaction', 'K-ras', 'colorectal cancer', 1.0)\n",
      "(189, 'H-ras mutations in human pituitary carcinoma metastases', 'H-ras', 'pituitary carcinoma metastases', 1.0)\n",
      "(190, 'Understanding the predictive role of K-ras for epidermal growth factor receptor-targeted therapies in colorectal cancer', 'K-ras', 'colorectal cancer', 1.0)\n",
      "(191, 'Understanding the predictive role of K-ras for epidermal growth factor receptor-targeted therapies in colorectal cancer', 'epidermal growth factor receptor', 'colorectal cancer', 1.0)\n",
      "(192, 'Mutations in APC, Kirsten-ras, and p53-alternative genetic pathways to colorectal cancer', 'Kirsten-ras', 'APC', 1.0)\n",
      "(193, 'Mutations in APC, Kirsten-ras, and p53-alternative genetic pathways to colorectal cancer', 'p53', 'APC', 1.0)\n",
      "(194, 'Mutations in APC, Kirsten-ras, and p53-alternative genetic pathways to colorectal cancer', 'Kirsten-ras', 'colorectal cancer', 1.0)\n",
      "(195, 'Mutations in APC, Kirsten-ras, and p53-alternative genetic pathways to colorectal cancer', 'p53', 'colorectal cancer', 1.0)\n",
      "(196, 'Setting future standards for KRAS testing in colorectal cancer', 'KRAS', 'colorectal cancer', 1.0)\n",
      "(197, 'Mitogen-activated protein kinase phosphatase-3 is a tumor promoter target in initiated cells that express oncogenic Ras', 'Mitogen-activated protein kinase phosphatase-3', 'tumor', 1.0)\n",
      "(198, 'Different types of K-Ras mutations are conversely associated with overall survival in patients with colorectal cancer', 'K-Ras', 'colorectal cancer', 1.0)\n",
      "(199, 'KRAS mutation profile in colorectal carcinoma and novel mutation-internal tandem duplication in KRAS', 'KRAS', 'colorectal carcinoma', 1.0)\n",
      "(200, 'Pancreatic tumor cells with mutant K-ras suppress ERK activity by MEK-dependent induction of MAP kinase phosphatase-2', 'K-ras', 'Pancreatic tumor', 1.0)\n",
      "(201, 'Pancreatic tumor cells with mutant K-ras suppress ERK activity by MEK-dependent induction of MAP kinase phosphatase-2', 'ERK', 'Pancreatic tumor', 1.0)\n",
      "(202, 'Pancreatic tumor cells with mutant K-ras suppress ERK activity by MEK-dependent induction of MAP kinase phosphatase-2', 'MEK', 'Pancreatic tumor', 1.0)\n",
      "(204, 'Glucose deprivation contributes to the development of KRAS pathway mutations in tumor cells', 'KRAS', 'tumor', 1.0)\n",
      "(205, 'Active Ras, where tumour-specific mutations lock Ras in the GTP-bound conformation, stimulates the RAS-RAF-MEK-ERK-MAP kinase signalling pathway.', 'RAF', 'tumour', 1.0)\n",
      "(206, 'Active Ras, where tumour-specific mutations lock Ras in the GTP-bound conformation, stimulates the RAS-RAF-MEK-ERK-MAP kinase signalling pathway.', 'MEK', 'tumour', 1.0)\n",
      "(207, 'Active Ras, where tumour-specific mutations lock Ras in the GTP-bound conformation, stimulates the RAS-RAF-MEK-ERK-MAP kinase signalling pathway.', 'ERK', 'tumour', 1.0)\n",
      "(208, 'A novel K-Ras gene copy number assay was designed as described in Materials and Methods and used to screen genomic DNA samples extracted from 96 paired normal (N) and colorectal tumour (T) tissues', 'K-Ras', 'colorectal tumour', 1.0)\n",
      "(209, 'Representative colorectal tumour sections (C) without and (D) with a K-Ras gene amplification are illustrated.', 'K-Ras', 'colorectal tumour', 1.0)\n",
      "(210, 'Relationship of increased aurora kinase A gene copy number, prognosis and response to chemotherapy in patients with metastatic colorectal cancer', 'aurora kinase A', 'colorectal cancer', None)\n",
      "(211, 'Increased Aurora kinase A gene copy number (AURKA-CN) has been reported in metastatic colorectal cancer (mCRC), with unknown relationship to clinical outcome', 'Aurora kinase A', 'colorectal cancer', None)\n",
      "(212, 'Increased Aurora kinase A gene copy number (AURKA-CN) has been reported in metastatic colorectal cancer (mCRC), with unknown relationship to clinical outcome', 'AURKA', 'colorectal cancer', None)\n",
      "(213, 'We correlated increased AURKA-CN in mCRC tumours with KRAS mutation status, overall and progression-free survival (OS, PFS).', 'AURKA', 'tumours', None)\n",
      "(214, 'We correlated increased AURKA-CN in mCRC tumours with KRAS mutation status, overall and progression-free survival (OS, PFS).', 'KRAS', 'tumours', None)\n",
      "(215, 'Sixty-one mCRC tumours were analysed for AURKA-CN using q-PCR, and KRAS mutation status by direct sequencing', 'AURKA', 'tumours', None)\n",
      "(216, 'Sixty-one mCRC tumours were analysed for AURKA-CN using q-PCR, and KRAS mutation status by direct sequencing', 'KRAS', 'tumours', None)\n",
      "(217, 'In all, 68% of tumour exhibited high AURKA-CN, and 29% had a KRAS mutation, without correlation between the two', 'AURKA', 'tumour', None)\n",
      "(218, 'In all, 68% of tumour exhibited high AURKA-CN, and 29% had a KRAS mutation, without correlation between the two', 'KRAS', 'tumour', None)\n",
      "(219, 'Patients with high AURKA-CN tumours had longer median OS (48.6 vs 18.8 months, P=0.01), with stronger trend among KRAS wild-type tumours (median OS not reached vs 18.8 months, P=0.003)', 'AURKA', 'tumours', None)\n",
      "(220, 'Patients with high AURKA-CN tumours had longer median OS (48.6 vs 18.8 months, P=0.01), with stronger trend among KRAS wild-type tumours (median OS not reached vs 18.8 months, P=0.003)', 'AURKA', 'tumours', None)\n",
      "(221, 'Patients with high AURKA-CN tumours had longer median OS (48.6 vs 18.8 months, P=0.01), with stronger trend among KRAS wild-type tumours (median OS not reached vs 18.8 months, P=0.003)', 'KRAS', 'tumours', None)\n",
      "(222, 'Patients with high AURKA-CN tumours had longer median OS (48.6 vs 18.8 months, P=0.01), with stronger trend among KRAS wild-type tumours (median OS not reached vs 18.8 months, P=0.003)', 'KRAS', 'tumours', None)\n",
      "(223, 'AURKA-CN level did not affect outcomes among patients with KRAS mutant tumours.', 'AURKA', 'tumours', None)\n",
      "(224, 'AURKA-CN level did not affect outcomes among patients with KRAS mutant tumours.', 'KRAS', 'tumours', None)\n",
      "(225, 'Increased AURKA-CN is common in mCRC tumours and is associated with longer OS and longer PFS during chemotherapy, particularly in KRAS wild-type tumours.', 'AURKA', 'tumours', None)\n",
      "(226, 'Increased AURKA-CN is common in mCRC tumours and is associated with longer OS and longer PFS during chemotherapy, particularly in KRAS wild-type tumours.', 'AURKA', 'tumours', None)\n",
      "(227, 'Increased AURKA-CN is common in mCRC tumours and is associated with longer OS and longer PFS during chemotherapy, particularly in KRAS wild-type tumours.', 'KRAS', 'tumours', None)\n",
      "(228, 'Increased AURKA-CN is common in mCRC tumours and is associated with longer OS and longer PFS during chemotherapy, particularly in KRAS wild-type tumours.', 'KRAS', 'tumours', None)\n",
      "(229, 'Aurora kinase A (AURKA) (also known as Aurora-2, BTAK/STK15) regulates mitotic entry, centrosome maturation, bipolar spindle assembly, chromosome alignment, cytokinesis and exit from mitosis', 'Aurora kinase A', 'mitosis', None)\n",
      "(230, 'Aurora kinase A (AURKA) (also known as Aurora-2, BTAK/STK15) regulates mitotic entry, centrosome maturation, bipolar spindle assembly, chromosome alignment, cytokinesis and exit from mitosis', 'AURKA', 'mitosis', None)\n",
      "(231, 'Aurora kinase A (AURKA) (also known as Aurora-2, BTAK/STK15) regulates mitotic entry, centrosome maturation, bipolar spindle assembly, chromosome alignment, cytokinesis and exit from mitosis', 'Aurora-2', 'mitosis', None)\n",
      "(232, 'Aurora kinase A (AURKA) (also known as Aurora-2, BTAK/STK15) regulates mitotic entry, centrosome maturation, bipolar spindle assembly, chromosome alignment, cytokinesis and exit from mitosis', 'BTAK', 'mitosis', None)\n",
      "(233, 'Aurora kinase A (AURKA) (also known as Aurora-2, BTAK/STK15) regulates mitotic entry, centrosome maturation, bipolar spindle assembly, chromosome alignment, cytokinesis and exit from mitosis', 'STK15', 'mitosis', None)\n",
      "(234, 'Aberrations in the function of Aurora kinases can result in abnormal cell division and aneuploidy due to losses or gains of whole chromosomes', 'Aurora kinases', 'aneuploidy', None)\n",
      "(235, 'In colorectal cancer (CRC), AURKA protein overexpression and amplification have been frequently observed', 'AURKA', 'colorectal cancer', None)\n",
      "(236, 'were the first to report overexpression of AURKA mRNA in >50% of CRC tumours', 'AURKA', 'CRC tumours', None)\n",
      "(237, 'The group led by was the first to demonstrate an increase in AURKA gene copy number (AURKA-CN) in 29% of a small sample of colorectal tumours', 'AURKA', 'colorectal tumours', None)\n",
      "(238, 'The group led by was the first to demonstrate an increase in AURKA gene copy number (AURKA-CN) in 29% of a small sample of colorectal tumours', 'AURKA', 'colorectal tumours', None)\n",
      "(239, 'Recently, the group led by reported increased AURKA-CN in 32% of CRC tumour samples and particularly in higher stage tumours, suggesting that AURKA may have a role in tumour progression', 'AURKA', 'CRC tumour', None)\n",
      "(240, 'Recently, the group led by reported increased AURKA-CN in 32% of CRC tumour samples and particularly in higher stage tumours, suggesting that AURKA may have a role in tumour progression', 'AURKA', 'stage tumours', None)\n",
      "(241, 'Recently, the group led by reported increased AURKA-CN in 32% of CRC tumour samples and particularly in higher stage tumours, suggesting that AURKA may have a role in tumour progression', 'AURKA', 'tumour', None)\n",
      "(242, 'Recently, the group led by reported increased AURKA-CN in 32% of CRC tumour samples and particularly in higher stage tumours, suggesting that AURKA may have a role in tumour progression', 'AURKA', 'CRC tumour', None)\n",
      "(243, 'Recently, the group led by reported increased AURKA-CN in 32% of CRC tumour samples and particularly in higher stage tumours, suggesting that AURKA may have a role in tumour progression', 'AURKA', 'stage tumours', None)\n",
      "(244, 'Recently, the group led by reported increased AURKA-CN in 32% of CRC tumour samples and particularly in higher stage tumours, suggesting that AURKA may have a role in tumour progression', 'AURKA', 'tumour', None)\n",
      "(245, 'The goal in this study was to assess the frequency of increased AURKA-CN in archival tumour tissues of patients with metastatic CRC (mCRC) and correlate this finding with progression-free survival (PFS), overall survival (OS), and KRAS mutation status.', 'AURKA', 'archival tumour', None)\n",
      "(246, 'The goal in this study was to assess the frequency of increased AURKA-CN in archival tumour tissues of patients with metastatic CRC (mCRC) and correlate this finding with progression-free survival (PFS), overall survival (OS), and KRAS mutation status.', 'KRAS', 'archival tumour', None)\n",
      "(247, 'We analysed 61 consecutive mCRC tumour samples submitted between 2006 and 2009 to the Fox Chase Cancer Center (FCCC) molecular genetic facility for evaluation of KRAS mutation status, with remaining tissue available for AURKA-CN evaluation', 'KRAS', 'tumour', None)\n",
      "(248, 'We analysed 61 consecutive mCRC tumour samples submitted between 2006 and 2009 to the Fox Chase Cancer Center (FCCC) molecular genetic facility for evaluation of KRAS mutation status, with remaining tissue available for AURKA-CN evaluation', 'AURKA', 'tumour', None)\n",
      "(249, 'We analysed 61 consecutive mCRC tumour samples submitted between 2006 and 2009 to the Fox Chase Cancer Center (FCCC) molecular genetic facility for evaluation of KRAS mutation status, with remaining tissue available for AURKA-CN evaluation', 'KRAS', 'Cancer', None)\n",
      "(250, 'We analysed 61 consecutive mCRC tumour samples submitted between 2006 and 2009 to the Fox Chase Cancer Center (FCCC) molecular genetic facility for evaluation of KRAS mutation status, with remaining tissue available for AURKA-CN evaluation', 'AURKA', 'Cancer', None)\n",
      "(251, 'Most of the patients had their tumour samples submitted by their treating oncologist for KRAS mutation status evaluation after the test became routinely used in clinical practice (2008)', 'KRAS', 'tumour', None)\n",
      "(252, 'Of the 61 tissue samples obtained for AURKA-CN analysis, 62% originated from primary tumours, while 38% originated from metastatic sites', 'AURKA', 'tumours', None)\n",
      "(253, 'For patients with high AURKA-CN tumours, the median PFS on first-line chemotherapy was 11.5 months vs 7.7 months for patients with low AURKA-CN tumours (HR=0.56, 95% CI: 0.28-1.1, P=0.094; Figure 3A)', 'AURKA', 'tumours', None)\n",
      "(254, 'For patients with high AURKA-CN tumours, the median PFS on first-line chemotherapy was 11.5 months vs 7.7 months for patients with low AURKA-CN tumours (HR=0.56, 95% CI: 0.28-1.1, P=0.094; Figure 3A)', 'AURKA', 'low AURKA-CN tumours', None)\n",
      "(255, 'Among patients with KRAS wild-type tumours, those with high AURKA-CN tumours had prolonged PFS compared with patients with low AURKA-CN tumours (HR=0.43, 95% CI: 0.19-0.94, P=0.04; Figure 3B)', 'AURKA', 'KRAS wild-type tumours', None)\n",
      "(256, 'Among patients with KRAS wild-type tumours, those with high AURKA-CN tumours had prolonged PFS compared with patients with low AURKA-CN tumours (HR=0.43, 95% CI: 0.19-0.94, P=0.04; Figure 3B)', 'AURKA', 'tumours', None)\n",
      "(257, 'Among patients with KRAS wild-type tumours, those with high AURKA-CN tumours had prolonged PFS compared with patients with low AURKA-CN tumours (HR=0.43, 95% CI: 0.19-0.94, P=0.04; Figure 3B)', 'AURKA', 'low AURKA-CN tumours', None)\n",
      "(258, 'Conversely, PFS did not differ by AURKA-CN expression among patients with KRAS mutant tumours (HR=1.06, 95% CI: 0.28-3.93, P=0.93), although the small sample size limits definitive conclusions regarding this population.', 'AURKA', 'tumours', None)\n",
      "(259, 'Conversely, PFS did not differ by AURKA-CN expression among patients with KRAS mutant tumours (HR=1.06, 95% CI: 0.28-3.93, P=0.93), although the small sample size limits definitive conclusions regarding this population.', 'KRAS', 'tumours', None)\n",
      "(260, 'No statistically significant difference in PFS was noted among patients with high vs low AURKA-CN tumours in the group overall (10.4 months vs 7.7 months; HR=0.54, 95% CI: 0.24-1.19, P=0.13; Figure 3C)', 'AURKA', 'tumours', None)\n",
      "(261, 'Table 3 outlines median PFS and HRs for high and low AURKA-CN tumours by KRAS mutational status and the use of cetuximab', 'AURKA', 'tumours', None)\n",
      "(262, 'Table 3 outlines median PFS and HRs for high and low AURKA-CN tumours by KRAS mutational status and the use of cetuximab', 'KRAS', 'tumours', None)\n",
      "(263, 'Interestingly, among patients with KRAS wild-type tumours, high AURKA-CN appeared to have the largest association with outcome among patients who did not receive cetuximab', 'AURKA', 'KRAS wild-type tumours', None)\n",
      "(264, 'A significantly longer OS was noted among patients with high vs low AURKA-CN (median OS 48.6 months for patients with high AURKA-CN tumours compared with 18.8 months for patients with low AURKA-CN tumours, HR=0.28, 95% CI: 0.10-0.73, P=0.01; Figure 4A)', 'AURKA', 'tumours', None)\n",
      "(265, 'A significantly longer OS was noted among patients with high vs low AURKA-CN (median OS 48.6 months for patients with high AURKA-CN tumours compared with 18.8 months for patients with low AURKA-CN tumours, HR=0.28, 95% CI: 0.10-0.73, P=0.01; Figure 4A)', 'AURKA', 'low AURKA-CN tumours', None)\n",
      "(266, 'A significantly longer OS was noted among patients with high vs low AURKA-CN (median OS 48.6 months for patients with high AURKA-CN tumours compared with 18.8 months for patients with low AURKA-CN tumours, HR=0.28, 95% CI: 0.10-0.73, P=0.01; Figure 4A)', 'AURKA', 'tumours', None)\n",
      "(267, 'A significantly longer OS was noted among patients with high vs low AURKA-CN (median OS 48.6 months for patients with high AURKA-CN tumours compared with 18.8 months for patients with low AURKA-CN tumours, HR=0.28, 95% CI: 0.10-0.73, P=0.01; Figure 4A)', 'AURKA', 'low AURKA-CN tumours', None)\n",
      "(268, 'In all, 1-year and 2-year survival were also longer among patients with high AURKA-CN tumours compared with those with low AURKA-CN tumours (1 year: 92.5% vs 82.2% P<0.001; 2 year: 80.9% vs 29.9%, P<0.001).', 'AURKA', 'tumours', None)\n",
      "(269, 'In all, 1-year and 2-year survival were also longer among patients with high AURKA-CN tumours compared with those with low AURKA-CN tumours (1 year: 92.5% vs 82.2% P<0.001; 2 year: 80.9% vs 29.9%, P<0.001).', 'AURKA', 'AURKA-CN tumours', None)\n",
      "(270, 'Similarly to PFS, the longer OS for patients with high AURKA-CN tumours was particularly pronounced in the KRAS wild-type patient population (Figure 4B)', 'AURKA', 'tumours', None)\n",
      "(271, 'Similarly to PFS, the longer OS for patients with high AURKA-CN tumours was particularly pronounced in the KRAS wild-type patient population (Figure 4B)', 'KRAS', 'tumours', None)\n",
      "(272, 'Among patients with KRAS wild-type tumours, the median OS was not reached for high AURKA-CN compared with 18.8 months for the group with low AURKA-CN tumours (HR=0.14, 95% CI: 0.038-0.514, P=0.003)', 'AURKA', 'KRAS wild-type tumours', None)\n",
      "(310, 'Quantitation of Aurora kinase A gene copy number in urine sediments and bladder cancer detection', 'Aurora kinase A', 'bladder cancer', None)\n",
      "(273, 'Among patients with KRAS wild-type tumours, the median OS was not reached for high AURKA-CN compared with 18.8 months for the group with low AURKA-CN tumours (HR=0.14, 95% CI: 0.038-0.514, P=0.003)', 'AURKA', 'low AURKA-CN tumours', None)\n",
      "(274, 'Similarly, increased 1-year and 2-year survival was seen among those patients with KRAS wild-type tumours and high AURKA-CN compared with patients with KRAS wild-type tumours and low AURKA-CN (1 year: 96% vs 75% P<0.0001; 2 year: 93% vs 28% P<0.0001)', 'KRAS', 'tumours', None)\n",
      "(275, 'Similarly, increased 1-year and 2-year survival was seen among those patients with KRAS wild-type tumours and high AURKA-CN compared with patients with KRAS wild-type tumours and low AURKA-CN (1 year: 96% vs 75% P<0.0001; 2 year: 93% vs 28% P<0.0001)', 'KRAS', 'tumours', None)\n",
      "(276, 'Similarly, increased 1-year and 2-year survival was seen among those patients with KRAS wild-type tumours and high AURKA-CN compared with patients with KRAS wild-type tumours and low AURKA-CN (1 year: 96% vs 75% P<0.0001; 2 year: 93% vs 28% P<0.0001)', 'AURKA', 'tumours', None)\n",
      "(277, 'Similarly, increased 1-year and 2-year survival was seen among those patients with KRAS wild-type tumours and high AURKA-CN compared with patients with KRAS wild-type tumours and low AURKA-CN (1 year: 96% vs 75% P<0.0001; 2 year: 93% vs 28% P<0.0001)', 'KRAS', 'tumours', None)\n",
      "(278, 'Similarly, increased 1-year and 2-year survival was seen among those patients with KRAS wild-type tumours and high AURKA-CN compared with patients with KRAS wild-type tumours and low AURKA-CN (1 year: 96% vs 75% P<0.0001; 2 year: 93% vs 28% P<0.0001)', 'AURKA', 'tumours', None)\n",
      "(279, 'Similarly, increased 1-year and 2-year survival was seen among those patients with KRAS wild-type tumours and high AURKA-CN compared with patients with KRAS wild-type tumours and low AURKA-CN (1 year: 96% vs 75% P<0.0001; 2 year: 93% vs 28% P<0.0001)', 'AURKA', 'tumours', None)\n",
      "(280, 'Similarly, increased 1-year and 2-year survival was seen among those patients with KRAS wild-type tumours and high AURKA-CN compared with patients with KRAS wild-type tumours and low AURKA-CN (1 year: 96% vs 75% P<0.0001; 2 year: 93% vs 28% P<0.0001)', 'KRAS', 'tumours', None)\n",
      "(281, 'Similarly, increased 1-year and 2-year survival was seen among those patients with KRAS wild-type tumours and high AURKA-CN compared with patients with KRAS wild-type tumours and low AURKA-CN (1 year: 96% vs 75% P<0.0001; 2 year: 93% vs 28% P<0.0001)', 'AURKA', 'tumours', None)\n",
      "(282, 'IHC evaluation of AURKA hyperactivity', 'AURKA', 'hyperactivity', None)\n",
      "(283, 'Immunohistochemical evaluation of phosphorylated AURKA protein was performed on 27 available tumour samples (20 with high AURKA-CN and 7 with low AURKA-CN)', 'AURKA', 'tumour', None)\n",
      "(284, 'Immunohistochemical evaluation of phosphorylated AURKA protein was performed on 27 available tumour samples (20 with high AURKA-CN and 7 with low AURKA-CN)', 'AURKA', 'tumour', None)\n",
      "(285, 'Immunohistochemical evaluation of phosphorylated AURKA protein was performed on 27 available tumour samples (20 with high AURKA-CN and 7 with low AURKA-CN)', 'AURKA', 'tumour', None)\n",
      "(286, 'Among the seven tumours with low AURKA-CN, none were positive for phosphorylated T-288', 'AURKA', 'tumours', None)\n",
      "(287, 'However, among the 20 tumours with high AURKA-CN, 9 were found to have positive staining (P=0.06 for difference between high and low AURKA-CN tumours)', 'AURKA', 'tumours', None)\n",
      "(288, 'However, among the 20 tumours with high AURKA-CN, 9 were found to have positive staining (P=0.06 for difference between high and low AURKA-CN tumours)', 'AURKA', 'AURKA-CN tumours', None)\n",
      "(289, 'The groups led by and reported increased AURKA-CN in ~30% of CRC tumours', 'AURKA', 'CRC tumours', None)\n",
      "(290, 'The role of Aurora kinase in tumourigenesis is well documented, although its utility as a prognostic marker is still under investigation in many cancers', 'Aurora kinase', 'cancers', None)\n",
      "(291, 'reported a high expression of Aurora A by AQUA to be a poor prognostic marker in patients with breast cancer', 'Aurora A', 'breast cancer', None)\n",
      "(292, 'Similarly overexpression of Aurora kinase B was found to be a poor prognostic marker in non-small cell lung cancer', 'Aurora kinase B', 'non-small cell lung cancer', None)\n",
      "(293, 'Similarly to our findings, Lam and colleagues have demonstrated higher detection of the AURKA protein by IHC in low-grade (well or moderately differentiated) CRC samples compared with high-grade (poorly differentiated) tumours', 'AURKA', 'tumours', None)\n",
      "(294, 'Our data demonstrating increased PFS for patients with high AURKA-CN tumours support this hypothesis', 'AURKA', 'tumours', None)\n",
      "(295, 'The improved OS for patients with high AURKA-CN was most pronounced in our study among patients with KRAS wild-type tumours', 'AURKA', 'KRAS wild-type tumours', None)\n",
      "(296, 'The improved response of patients with KRAS wild-type tumours treated with anti-EGFR therapy has been well documented in the literature', 'KRAS', 'tumours', None)\n",
      "(297, 'The improved response of patients with KRAS wild-type tumours treated with anti-EGFR therapy has been well documented in the literature', 'EGFR', 'tumours', None)\n",
      "(298, 'However in our study, among patients with KRAS wild-type tumours, the impact of high AURKA-CN level was most pronounced in patients who did not receive cetuximab', 'AURKA', 'KRAS wild-type tumours', None)\n",
      "(299, 'Many groups have studied AURKA protein expression by IHC in CRC and other cancers', 'AURKA', 'cancers', None)\n",
      "(300, 'Moreover, our study is the first to demonstrate an association between high AURKA-CN and improved clinical outcome among patients with mCRC receiving chemotherapy, with a more pronounced association noted among patients with KRAS wild-type tumours', 'AURKA', 'KRAS wild-type tumours', None)\n",
      "(301, 'Aurora-A expression is independently associated with chromosomal instability in colorectal cancer', 'Aurora-A', 'colorectal cancer', None)\n",
      "(302, 'Drosophila Aurora-A is required for centrosome maturation and actin-dependent asymmetric protein localization during mitosis', 'Aurora-A', 'mitosis', None)\n",
      "(303, 'A homologue of Drosophila aurora kinase is oncogenic and amplified in human colorectal cancers', 'aurora kinase', 'colorectal cancers', None)\n",
      "(304, 'K-ras mutations and benefit from cetuximab in advanced colorectal cancer', 'K-ras', 'colorectal cancer', None)\n",
      "(305, 'Aurora kinase expression in colorectal adenocarcinoma: correlations with clinicopathological features, p16 expression, and telomerase activity', 'Aurora kinase', 'colorectal adenocarcinoma', None)\n",
      "(306, 'Aurora kinase expression in colorectal adenocarcinoma: correlations with clinicopathological features, p16 expression, and telomerase activity', 'p16', 'colorectal adenocarcinoma', None)\n",
      "(307, 'Expression of Aurora A (but not Aurora B) is predictive of survival in breast cancer', 'Aurora A', 'breast cancer', None)\n",
      "(308, 'Expression of Aurora A (but not Aurora B) is predictive of survival in breast cancer', 'Aurora B', 'breast cancer', None)\n",
      "(309, 'High copy amplification of the Aurora-A gene is associated with chromosomal instability phenotype in human colorectal cancers', 'Aurora-A', 'colorectal cancers', None)\n",
      "(311, 'Effects of DNAzymes targeting Aurora kinase A on the growth of human prostate cancer', 'Aurora kinase A', 'prostate cancer', None)\n",
      "(312, 'Overexpression of aurora B kinase (AURKB) in primary non-small cell lung carcinoma is frequent, generally driven from one allele, and correlates with the level of genetic instability', 'aurora B', 'carcinoma', None)\n",
      "(313, 'Overexpression of aurora B kinase (AURKB) in primary non-small cell lung carcinoma is frequent, generally driven from one allele, and correlates with the level of genetic instability', 'AURKB', 'carcinoma', None)\n",
      "(314, 'Aurora kinase B is a predictive factor for the aggressive recurrence of hepatocellular carcinoma after curative hepatectomy', 'Aurora kinase B', 'hepatocellular carcinoma', None)\n",
      "(315, 'Inhibition of Aurora-A suppresses epithelial-mesenchymal transition and invasion by downregulating MAPK in nasopharyngeal carcinoma cells', 'Aurora-A', 'carcinoma', None)\n",
      "(316, 'Aurora B expression correlates with aggressive behaviour in glioblastoma multiforme', 'Aurora B', 'glioblastoma multiforme', None)\n",
      "(317, 'Copy number increase of aurora kinase A in colorectal cancers: a correlation with tumor progression', 'aurora kinase A', 'colorectal cancers', None)\n",
      "(318, 'Copy number increase of aurora kinase A in colorectal cancers: a correlation with tumor progression', 'aurora kinase A', 'tumor', None)\n",
      "(319, 'Progression-free survival (PFS) of metastatic colorectal cancer patients receiving first-line and second-line chemotherapy by AURKA-CN and KRAS mutation status', 'AURKA', 'colorectal cancer', None)\n",
      "(320, 'Progression-free survival (PFS) of metastatic colorectal cancer patients receiving first-line and second-line chemotherapy by AURKA-CN and KRAS mutation status', 'KRAS', 'colorectal cancer', None)\n",
      "(321, 'Overall survival of metastatic colorectal cancer patients by AURKA-CN for entire cohort (A), the KRAS wild-type population (B) and the KRAS mutant population (C).', 'AURKA', 'colorectal cancer', None)\n",
      "(322, 'Overall survival of metastatic colorectal cancer patients by AURKA-CN for entire cohort (A), the KRAS wild-type population (B) and the KRAS mutant population (C).', 'KRAS', 'colorectal cancer', None)\n",
      "(323, 'Overall survival of metastatic colorectal cancer patients by AURKA-CN for entire cohort (A), the KRAS wild-type population (B) and the KRAS mutant population (C).', 'KRAS', 'colorectal cancer', None)\n",
      "(324, 'Immunohistochemistry staining for phosphorylated AURKA in metastatic colorectal cancer samples', 'AURKA', 'colorectal cancer', None)\n",
      "(325, 'Interplay between BRCA1 and RHAMM Regulates Epithelial Apicobasal Polarization and May Influence Risk of Breast Cancer', 'BRCA1', 'Breast Cancer', None)\n",
      "(326, 'Interplay between BRCA1 and RHAMM Regulates Epithelial Apicobasal Polarization and May Influence Risk of Breast Cancer', 'RHAMM', 'Breast Cancer', None)\n",
      "(327, 'Genetic analysis identifies the HMMR gene as a modifier of the breast cancer risk associated with BRCA1 gene mutation, while cell biological analysis of the protein product suggests a function in regulating development of the mammary gland.', 'HMMR', 'breast cancer', None)\n",
      "(328, 'Genetic analysis identifies the HMMR gene as a modifier of the breast cancer risk associated with BRCA1 gene mutation, while cell biological analysis of the protein product suggests a function in regulating development of the mammary gland.', 'BRCA1', 'breast cancer', None)\n",
      "(329, 'In BRCA1 mutation carriers, accumulation of stem and progenitor cells in normal breast tissue and increased risk of developing tumors of basal-like type suggest that BRCA1 regulates stem/progenitor cell proliferation and differentiation', 'BRCA1', 'tumors', None)\n",
      "(330, 'In BRCA1 mutation carriers, accumulation of stem and progenitor cells in normal breast tissue and increased risk of developing tumors of basal-like type suggest that BRCA1 regulates stem/progenitor cell proliferation and differentiation', 'BRCA1', 'tumors', None)\n",
      "(331, 'However, the function of BRCA1 in this process and its link to carcinogenesis remain unknown', 'BRCA1', 'carcinogenesis', None)\n",
      "(332, 'Here we depict a molecular mechanism involving BRCA1 and RHAMM that regulates apicobasal polarity and, when perturbed, may increase risk of breast cancer', 'BRCA1', 'breast cancer', None)\n",
      "(333, 'Here we depict a molecular mechanism involving BRCA1 and RHAMM that regulates apicobasal polarity and, when perturbed, may increase risk of breast cancer', 'RHAMM', 'breast cancer', None)\n",
      "(334, 'Starting from complementary genetic analyses across families and populations, we identified common genetic variation at the low-penetrance susceptibility HMMR locus (encoding for RHAMM) that modifies breast cancer risk among BRCA1, but probably not BRCA2, mutation carriers: n = 7,584, weighted hazard ratio (wHR) = 1.09 (95% CI 1.02-1.16), ptrend = 0.017; and n = 3,965, wHR = 1.04 (95% CI 0.94-1.16), ptrend = 0.43; respectively', 'HMMR', 'breast cancer', None)\n",
      "(335, 'Starting from complementary genetic analyses across families and populations, we identified common genetic variation at the low-penetrance susceptibility HMMR locus (encoding for RHAMM) that modifies breast cancer risk among BRCA1, but probably not BRCA2, mutation carriers: n = 7,584, weighted hazard ratio (wHR) = 1.09 (95% CI 1.02-1.16), ptrend = 0.017; and n = 3,965, wHR = 1.04 (95% CI 0.94-1.16), ptrend = 0.43; respectively', 'RHAMM', 'breast cancer', None)\n",
      "(336, 'Starting from complementary genetic analyses across families and populations, we identified common genetic variation at the low-penetrance susceptibility HMMR locus (encoding for RHAMM) that modifies breast cancer risk among BRCA1, but probably not BRCA2, mutation carriers: n = 7,584, weighted hazard ratio (wHR) = 1.09 (95% CI 1.02-1.16), ptrend = 0.017; and n = 3,965, wHR = 1.04 (95% CI 0.94-1.16), ptrend = 0.43; respectively', 'BRCA1', 'breast cancer', None)\n",
      "(337, 'Starting from complementary genetic analyses across families and populations, we identified common genetic variation at the low-penetrance susceptibility HMMR locus (encoding for RHAMM) that modifies breast cancer risk among BRCA1, but probably not BRCA2, mutation carriers: n = 7,584, weighted hazard ratio (wHR) = 1.09 (95% CI 1.02-1.16), ptrend = 0.017; and n = 3,965, wHR = 1.04 (95% CI 0.94-1.16), ptrend = 0.43; respectively', 'BRCA2', 'breast cancer', None)\n",
      "(338, 'Taken together, our data provide fundamental insight into apicobasal polarization through BRCA1 function, which may explain the expanded cell subsets and characteristic tumor type accompanying BRCA1 mutation, while also linking this process to sporadic breast cancer through perturbation of HMMR/RHAMM.', 'BRCA1', 'tumor', None)\n",
      "(339, 'Taken together, our data provide fundamental insight into apicobasal polarization through BRCA1 function, which may explain the expanded cell subsets and characteristic tumor type accompanying BRCA1 mutation, while also linking this process to sporadic breast cancer through perturbation of HMMR/RHAMM.', 'BRCA1', 'breast cancer', None)\n",
      "(340, 'Taken together, our data provide fundamental insight into apicobasal polarization through BRCA1 function, which may explain the expanded cell subsets and characteristic tumor type accompanying BRCA1 mutation, while also linking this process to sporadic breast cancer through perturbation of HMMR/RHAMM.', 'BRCA1', 'tumor', None)\n",
      "(341, 'Taken together, our data provide fundamental insight into apicobasal polarization through BRCA1 function, which may explain the expanded cell subsets and characteristic tumor type accompanying BRCA1 mutation, while also linking this process to sporadic breast cancer through perturbation of HMMR/RHAMM.', 'HMMR', 'tumor', None)\n",
      "(342, 'Taken together, our data provide fundamental insight into apicobasal polarization through BRCA1 function, which may explain the expanded cell subsets and characteristic tumor type accompanying BRCA1 mutation, while also linking this process to sporadic breast cancer through perturbation of HMMR/RHAMM.', 'RHAMM', 'tumor', None)\n",
      "(343, 'Taken together, our data provide fundamental insight into apicobasal polarization through BRCA1 function, which may explain the expanded cell subsets and characteristic tumor type accompanying BRCA1 mutation, while also linking this process to sporadic breast cancer through perturbation of HMMR/RHAMM.', 'BRCA1', 'breast cancer', None)\n",
      "(344, 'Taken together, our data provide fundamental insight into apicobasal polarization through BRCA1 function, which may explain the expanded cell subsets and characteristic tumor type accompanying BRCA1 mutation, while also linking this process to sporadic breast cancer through perturbation of HMMR/RHAMM.', 'HMMR', 'breast cancer', None)\n",
      "(345, 'Taken together, our data provide fundamental insight into apicobasal polarization through BRCA1 function, which may explain the expanded cell subsets and characteristic tumor type accompanying BRCA1 mutation, while also linking this process to sporadic breast cancer through perturbation of HMMR/RHAMM.', 'RHAMM', 'breast cancer', None)\n",
      "(346, 'Mutations in two genes that were initially identified as predisposing carriers to early-onset breast cancer, BRCA1 and BRCA2, cause similar perturbations in cellular responses to DNA damage but predispose carriers to distinct tumor types', 'BRCA1', 'breast cancer', None)\n",
      "(347, 'Mutations in two genes that were initially identified as predisposing carriers to early-onset breast cancer, BRCA1 and BRCA2, cause similar perturbations in cellular responses to DNA damage but predispose carriers to distinct tumor types', 'BRCA2', 'breast cancer', None)\n",
      "(348, 'Mutations in two genes that were initially identified as predisposing carriers to early-onset breast cancer, BRCA1 and BRCA2, cause similar perturbations in cellular responses to DNA damage but predispose carriers to distinct tumor types', 'BRCA1', 'tumor', None)\n",
      "(349, 'Mutations in two genes that were initially identified as predisposing carriers to early-onset breast cancer, BRCA1 and BRCA2, cause similar perturbations in cellular responses to DNA damage but predispose carriers to distinct tumor types', 'BRCA2', 'tumor', None)\n",
      "(350, 'We have used genetic analyses of affected families to uncover additional genetic variation that is linked to the risk of developing cancer for carriers of BRCA1 mutations', 'BRCA1', 'cancer', None)\n",
      "(351, 'Our findings provide new insights into the mechanism through which BRCA1 may promote commitment of initially bipotent mammary cells towards the luminal lineage, and how loss of this function may predispose cells to become breast tumors of a basal-like type.', 'BRCA1', 'breast tumors', None)\n",
      "(352, 'The common pathological features of breast tumors arising in breast cancer 1, early onset (BRCA1) gene mutation carriers, including the basal-like phenotype and ER negativity, led to the proposition that BRCA1 function regulates stem/progenitor cell proliferation and differentiation', 'BRCA1', 'breast tumors', None)\n",
      "(353, 'The common pathological features of breast tumors arising in breast cancer 1, early onset (BRCA1) gene mutation carriers, including the basal-like phenotype and ER negativity, led to the proposition that BRCA1 function regulates stem/progenitor cell proliferation and differentiation', 'BRCA1', 'breast tumors', None)\n",
      "(354, 'The common pathological features of breast tumors arising in breast cancer 1, early onset (BRCA1) gene mutation carriers, including the basal-like phenotype and ER negativity, led to the proposition that BRCA1 function regulates stem/progenitor cell proliferation and differentiation', 'BRCA1', 'breast cancer', None)\n",
      "(355, 'The common pathological features of breast tumors arising in breast cancer 1, early onset (BRCA1) gene mutation carriers, including the basal-like phenotype and ER negativity, led to the proposition that BRCA1 function regulates stem/progenitor cell proliferation and differentiation', 'BRCA1', 'breast cancer', None)\n",
      "(356, 'Expanded luminal progenitor populations have also been detected in breast tissue from BRCA1 mutation carriers and, subsequently, proposed as the target of transformation leading to basal-like tumors', 'BRCA1', 'tumors', None)\n",
      "(357, 'Xrhamm is the ortholog of a candidate low-penetrance breast cancer susceptibility gene product (RHAMM, HMMR gene) whose over-expression in tumors is associated with poor prognosis and early age at diagnosis', 'Xrhamm', 'breast cancer', None)\n",
      "(358, 'Xrhamm is the ortholog of a candidate low-penetrance breast cancer susceptibility gene product (RHAMM, HMMR gene) whose over-expression in tumors is associated with poor prognosis and early age at diagnosis', 'Xrhamm', 'tumors', None)\n",
      "(359, 'Xrhamm is the ortholog of a candidate low-penetrance breast cancer susceptibility gene product (RHAMM, HMMR gene) whose over-expression in tumors is associated with poor prognosis and early age at diagnosis', 'RHAMM', 'breast cancer', None)\n",
      "(360, 'Xrhamm is the ortholog of a candidate low-penetrance breast cancer susceptibility gene product (RHAMM, HMMR gene) whose over-expression in tumors is associated with poor prognosis and early age at diagnosis', 'HMMR', 'breast cancer', None)\n",
      "(361, 'Xrhamm is the ortholog of a candidate low-penetrance breast cancer susceptibility gene product (RHAMM, HMMR gene) whose over-expression in tumors is associated with poor prognosis and early age at diagnosis', 'RHAMM', 'tumors', None)\n",
      "(362, 'Xrhamm is the ortholog of a candidate low-penetrance breast cancer susceptibility gene product (RHAMM, HMMR gene) whose over-expression in tumors is associated with poor prognosis and early age at diagnosis', 'HMMR', 'tumors', None)\n",
      "(363, 'Conversely, loss of BRCA1 function might impair structural cues of terminal differentiation and, consequently, increase risk of breast cancer characterized by the basal-like tumor type', 'BRCA1', 'breast cancer', None)\n",
      "(364, 'Conversely, loss of BRCA1 function might impair structural cues of terminal differentiation and, consequently, increase risk of breast cancer characterized by the basal-like tumor type', 'BRCA1', 'tumor', None)\n",
      "(365, 'Common Genetic Variation in HMMR Modifies Breast Cancer Risk among BRCA1 Mutation Carriers', 'BRCA1', 'HMMR Modifies Breast Cancer', None)\n",
      "(366, 'Although BRCA1 and BRCA2 function coordinately during DNA damage response, genomic, transcriptomic, molecular, and pathological features of breast tumors arising in BRCA1 and BRCA2 mutation carriers suggest that carcinogenesis may occur through perturbation of shared and distinct biological processes', 'BRCA1', 'breast tumors', None)\n",
      "(367, 'Although BRCA1 and BRCA2 function coordinately during DNA damage response, genomic, transcriptomic, molecular, and pathological features of breast tumors arising in BRCA1 and BRCA2 mutation carriers suggest that carcinogenesis may occur through perturbation of shared and distinct biological processes', 'BRCA1', 'carcinogenesis', None)\n",
      "(368, 'Although BRCA1 and BRCA2 function coordinately during DNA damage response, genomic, transcriptomic, molecular, and pathological features of breast tumors arising in BRCA1 and BRCA2 mutation carriers suggest that carcinogenesis may occur through perturbation of shared and distinct biological processes', 'BRCA2', 'breast tumors', None)\n",
      "(369, 'Although BRCA1 and BRCA2 function coordinately during DNA damage response, genomic, transcriptomic, molecular, and pathological features of breast tumors arising in BRCA1 and BRCA2 mutation carriers suggest that carcinogenesis may occur through perturbation of shared and distinct biological processes', 'BRCA2', 'carcinogenesis', None)\n",
      "(370, 'Although BRCA1 and BRCA2 function coordinately during DNA damage response, genomic, transcriptomic, molecular, and pathological features of breast tumors arising in BRCA1 and BRCA2 mutation carriers suggest that carcinogenesis may occur through perturbation of shared and distinct biological processes', 'BRCA1', 'breast tumors', None)\n",
      "(371, 'Although BRCA1 and BRCA2 function coordinately during DNA damage response, genomic, transcriptomic, molecular, and pathological features of breast tumors arising in BRCA1 and BRCA2 mutation carriers suggest that carcinogenesis may occur through perturbation of shared and distinct biological processes', 'BRCA2', 'breast tumors', None)\n",
      "(372, 'Although BRCA1 and BRCA2 function coordinately during DNA damage response, genomic, transcriptomic, molecular, and pathological features of breast tumors arising in BRCA1 and BRCA2 mutation carriers suggest that carcinogenesis may occur through perturbation of shared and distinct biological processes', 'BRCA1', 'carcinogenesis', None)\n",
      "(373, 'Although BRCA1 and BRCA2 function coordinately during DNA damage response, genomic, transcriptomic, molecular, and pathological features of breast tumors arising in BRCA1 and BRCA2 mutation carriers suggest that carcinogenesis may occur through perturbation of shared and distinct biological processes', 'BRCA2', 'carcinogenesis', None)\n",
      "(374, 'Previous analysis of candidate genomic regions using a linkage approach suggested specific modification of breast cancer risk among BRCA1 mutation carriers by common genetic variation at chromosome 5q33-34', 'BRCA1', 'breast cancer', None)\n",
      "(375, 'Common breast cancer-predisposition alleles may differentially modify breast cancer risk among BRCA1 and BRCA2 mutation carriers', 'BRCA1', 'breast cancer', None)\n",
      "(376, 'Common breast cancer-predisposition alleles may differentially modify breast cancer risk among BRCA1 and BRCA2 mutation carriers', 'BRCA2', 'breast cancer', None)\n",
      "(377, 'Common breast cancer-predisposition alleles may differentially modify breast cancer risk among BRCA1 and BRCA2 mutation carriers', 'BRCA1', 'breast cancer', None)\n",
      "(378, 'Common breast cancer-predisposition alleles may differentially modify breast cancer risk among BRCA1 and BRCA2 mutation carriers', 'BRCA2', 'breast cancer', None)\n",
      "(379, 'To complement the linkage approach, we evaluated the effect of common HMMR genetic variation on breast cancer risk in BRCA1 and BRCA2 mutation carriers', 'HMMR', 'breast cancer', None)\n",
      "(380, 'To complement the linkage approach, we evaluated the effect of common HMMR genetic variation on breast cancer risk in BRCA1 and BRCA2 mutation carriers', 'BRCA1', 'breast cancer', None)\n",
      "(381, 'To complement the linkage approach, we evaluated the effect of common HMMR genetic variation on breast cancer risk in BRCA1 and BRCA2 mutation carriers', 'BRCA2', 'breast cancer', None)\n",
      "(382, 'Following a pilot study in Italy and Spain, analysis of carriers (n = 11,609) collected through 24 study groups participating in the Consortium of Investigators of Modifiers of BRCA1/2 (CIMBA) detected significant modification of breast cancer risk by HMMR rs299290 variant among BRCA1, but not BRCA2, mutation carriers: BRCA1 mutation carriers n = 7,584, Cox proportional-hazards regression model, hazard ratio (HR) = 1.08 (95% confidence interval (CI) 1.02-1.13), ptrend = 0.004 (p 2df = 0.014), in the same direction as originally detected in Ashkenazi Jewish populations; BRCA2 mutation carriers n = 3,965, HR = 1.03 (95% CI 0.96-1.10), ptrend = 0.42 (p 2df = 0.67)', 'BRCA1/2', 'breast cancer', None)\n",
      "(383, 'Following a pilot study in Italy and Spain, analysis of carriers (n = 11,609) collected through 24 study groups participating in the Consortium of Investigators of Modifiers of BRCA1/2 (CIMBA) detected significant modification of breast cancer risk by HMMR rs299290 variant among BRCA1, but not BRCA2, mutation carriers: BRCA1 mutation carriers n = 7,584, Cox proportional-hazards regression model, hazard ratio (HR) = 1.08 (95% confidence interval (CI) 1.02-1.13), ptrend = 0.004 (p 2df = 0.014), in the same direction as originally detected in Ashkenazi Jewish populations; BRCA2 mutation carriers n = 3,965, HR = 1.03 (95% CI 0.96-1.10), ptrend = 0.42 (p 2df = 0.67)', 'HMMR', 'breast cancer', None)\n",
      "(384, 'Following a pilot study in Italy and Spain, analysis of carriers (n = 11,609) collected through 24 study groups participating in the Consortium of Investigators of Modifiers of BRCA1/2 (CIMBA) detected significant modification of breast cancer risk by HMMR rs299290 variant among BRCA1, but not BRCA2, mutation carriers: BRCA1 mutation carriers n = 7,584, Cox proportional-hazards regression model, hazard ratio (HR) = 1.08 (95% confidence interval (CI) 1.02-1.13), ptrend = 0.004 (p 2df = 0.014), in the same direction as originally detected in Ashkenazi Jewish populations; BRCA2 mutation carriers n = 3,965, HR = 1.03 (95% CI 0.96-1.10), ptrend = 0.42 (p 2df = 0.67)', 'BRCA1', 'breast cancer', None)\n",
      "(385, 'Following a pilot study in Italy and Spain, analysis of carriers (n = 11,609) collected through 24 study groups participating in the Consortium of Investigators of Modifiers of BRCA1/2 (CIMBA) detected significant modification of breast cancer risk by HMMR rs299290 variant among BRCA1, but not BRCA2, mutation carriers: BRCA1 mutation carriers n = 7,584, Cox proportional-hazards regression model, hazard ratio (HR) = 1.08 (95% confidence interval (CI) 1.02-1.13), ptrend = 0.004 (p 2df = 0.014), in the same direction as originally detected in Ashkenazi Jewish populations; BRCA2 mutation carriers n = 3,965, HR = 1.03 (95% CI 0.96-1.10), ptrend = 0.42 (p 2df = 0.67)', 'BRCA2', 'breast cancer', None)\n",
      "(386, 'Following a pilot study in Italy and Spain, analysis of carriers (n = 11,609) collected through 24 study groups participating in the Consortium of Investigators of Modifiers of BRCA1/2 (CIMBA) detected significant modification of breast cancer risk by HMMR rs299290 variant among BRCA1, but not BRCA2, mutation carriers: BRCA1 mutation carriers n = 7,584, Cox proportional-hazards regression model, hazard ratio (HR) = 1.08 (95% confidence interval (CI) 1.02-1.13), ptrend = 0.004 (p 2df = 0.014), in the same direction as originally detected in Ashkenazi Jewish populations; BRCA2 mutation carriers n = 3,965, HR = 1.03 (95% CI 0.96-1.10), ptrend = 0.42 (p 2df = 0.67)', 'BRCA1', 'breast cancer', None)\n",
      "(415, 'That is, despite the expected differences in the frequency of tumor types between the two sets of carriers, heterogeneity was observed in the distribution of rs299290 genotypes in BRCA1, but not BRCA2, mutation carriers', 'BRCA2', 'tumor', None)\n",
      "(416, 'The growth of nonmalignant human mammary epithelial cells, such as MCF10A and HMT3522 S1, within three-dimensional cultures containing reconstituted basement membrane (rBM) recapitulates aspects of the terminal differentiation of mammary luminal epithelia, including apicobasal polarization, growth arrest, and milk production', 'HMT', 'luminal epithelia', None)\n",
      "(387, 'Following a pilot study in Italy and Spain, analysis of carriers (n = 11,609) collected through 24 study groups participating in the Consortium of Investigators of Modifiers of BRCA1/2 (CIMBA) detected significant modification of breast cancer risk by HMMR rs299290 variant among BRCA1, but not BRCA2, mutation carriers: BRCA1 mutation carriers n = 7,584, Cox proportional-hazards regression model, hazard ratio (HR) = 1.08 (95% confidence interval (CI) 1.02-1.13), ptrend = 0.004 (p 2df = 0.014), in the same direction as originally detected in Ashkenazi Jewish populations; BRCA2 mutation carriers n = 3,965, HR = 1.03 (95% CI 0.96-1.10), ptrend = 0.42 (p 2df = 0.67)', 'Cox', 'breast cancer', None)\n",
      "(388, 'Following a pilot study in Italy and Spain, analysis of carriers (n = 11,609) collected through 24 study groups participating in the Consortium of Investigators of Modifiers of BRCA1/2 (CIMBA) detected significant modification of breast cancer risk by HMMR rs299290 variant among BRCA1, but not BRCA2, mutation carriers: BRCA1 mutation carriers n = 7,584, Cox proportional-hazards regression model, hazard ratio (HR) = 1.08 (95% confidence interval (CI) 1.02-1.13), ptrend = 0.004 (p 2df = 0.014), in the same direction as originally detected in Ashkenazi Jewish populations; BRCA2 mutation carriers n = 3,965, HR = 1.03 (95% CI 0.96-1.10), ptrend = 0.42 (p 2df = 0.67)', 'BRCA2', 'breast cancer', None)\n",
      "(389, 'Effect of HMMR rs299290 variation on breast cancer risk among BRCA1 and BRCA2 mutation carriers.', 'HMMR', 'breast cancer', None)\n",
      "(390, 'Effect of HMMR rs299290 variation on breast cancer risk among BRCA1 and BRCA2 mutation carriers.', 'BRCA1', 'breast cancer', None)\n",
      "(391, 'Effect of HMMR rs299290 variation on breast cancer risk among BRCA1 and BRCA2 mutation carriers.', 'BRCA2', 'breast cancer', None)\n",
      "(392, 'First, since prophylactic oophorectomy reduces the risk of breast cancer in BRCA1 mutation carriers by up to 50%, we included this observation as a time-dependent covariate in the analysis, and a significant association similar to the one shown above was revealed: HR = 1.09 (95% CI 1.03-1.16), ptrend = 4.5x10-4', 'BRCA1', 'breast cancer', None)\n",
      "(393, 'While studies have identified low-penetrance alleles that associate with breast cancer risk in carriers of BRCA1 mutations and carriers of BRCA2 mutations, specificities have also been detected', 'BRCA1', 'breast cancer', None)\n",
      "(394, 'While studies have identified low-penetrance alleles that associate with breast cancer risk in carriers of BRCA1 mutations and carriers of BRCA2 mutations, specificities have also been detected', 'BRCA2', 'breast cancer', None)\n",
      "(395, 'Here, the results of linkage and association studies support a potential, specific genetic interaction between BRCA1 and HMMR (high- and low-penetrance mutations, respectively), which could highlight a BRCA1-RHAMM function altered in familial and sporadic breast carcinogenesis.', 'BRCA1', 'breast carcinogenesis', None)\n",
      "(396, 'Here, the results of linkage and association studies support a potential, specific genetic interaction between BRCA1 and HMMR (high- and low-penetrance mutations, respectively), which could highlight a BRCA1-RHAMM function altered in familial and sporadic breast carcinogenesis.', 'HMMR', 'breast carcinogenesis', None)\n",
      "(397, 'Here, the results of linkage and association studies support a potential, specific genetic interaction between BRCA1 and HMMR (high- and low-penetrance mutations, respectively), which could highlight a BRCA1-RHAMM function altered in familial and sporadic breast carcinogenesis.', 'BRCA1', 'breast carcinogenesis', None)\n",
      "(398, 'Here, the results of linkage and association studies support a potential, specific genetic interaction between BRCA1 and HMMR (high- and low-penetrance mutations, respectively), which could highlight a BRCA1-RHAMM function altered in familial and sporadic breast carcinogenesis.', 'RHAMM', 'breast carcinogenesis', None)\n",
      "(399, 'Breast tumors arising in BRCA1 mutation carriers are typically ER-negative, whereas most tumors in BRCA2 mutation carriers and sporadic cases are ER-positive', 'BRCA1', 'Breast tumors', None)\n",
      "(400, 'Breast tumors arising in BRCA1 mutation carriers are typically ER-negative, whereas most tumors in BRCA2 mutation carriers and sporadic cases are ER-positive', 'ER', 'Breast tumors', None)\n",
      "(401, 'Breast tumors arising in BRCA1 mutation carriers are typically ER-negative, whereas most tumors in BRCA2 mutation carriers and sporadic cases are ER-positive', 'BRCA2', 'Breast tumors', None)\n",
      "(402, 'Breast tumors arising in BRCA1 mutation carriers are typically ER-negative, whereas most tumors in BRCA2 mutation carriers and sporadic cases are ER-positive', 'ER', 'Breast tumors', None)\n",
      "(403, 'Breast tumors arising in BRCA1 mutation carriers are typically ER-negative, whereas most tumors in BRCA2 mutation carriers and sporadic cases are ER-positive', 'BRCA1', 'tumors', None)\n",
      "(404, 'Breast tumors arising in BRCA1 mutation carriers are typically ER-negative, whereas most tumors in BRCA2 mutation carriers and sporadic cases are ER-positive', 'ER', 'tumors', None)\n",
      "(405, 'Breast tumors arising in BRCA1 mutation carriers are typically ER-negative, whereas most tumors in BRCA2 mutation carriers and sporadic cases are ER-positive', 'BRCA2', 'tumors', None)\n",
      "(406, 'Breast tumors arising in BRCA1 mutation carriers are typically ER-negative, whereas most tumors in BRCA2 mutation carriers and sporadic cases are ER-positive', 'ER', 'tumors', None)\n",
      "(407, 'Given the evidence above, we next evaluated whether HMMR variation was associated with ER tumor status in BRCA1 and/or BRCA2 mutation carriers', 'HMMR', 'tumor', None)\n",
      "(408, 'Given the evidence above, we next evaluated whether HMMR variation was associated with ER tumor status in BRCA1 and/or BRCA2 mutation carriers', 'ER', 'tumor', None)\n",
      "(409, 'Given the evidence above, we next evaluated whether HMMR variation was associated with ER tumor status in BRCA1 and/or BRCA2 mutation carriers', 'BRCA1', 'tumor', None)\n",
      "(410, 'Given the evidence above, we next evaluated whether HMMR variation was associated with ER tumor status in BRCA1 and/or BRCA2 mutation carriers', 'BRCA2', 'tumor', None)\n",
      "(411, 'In data provided by several CIMBA groups (Text S1), no ER-positive tumors were observed among rare rs299290 homozygotes in BRCA1 mutation carriers (p interaction = 0.006), whereas this bias was not observed in BRCA2 mutation carriers (p interaction = 0.95) (Table S4)', 'ER', 'tumors', None)\n",
      "(412, 'In data provided by several CIMBA groups (Text S1), no ER-positive tumors were observed among rare rs299290 homozygotes in BRCA1 mutation carriers (p interaction = 0.006), whereas this bias was not observed in BRCA2 mutation carriers (p interaction = 0.95) (Table S4)', 'BRCA1', 'tumors', None)\n",
      "(413, 'In data provided by several CIMBA groups (Text S1), no ER-positive tumors were observed among rare rs299290 homozygotes in BRCA1 mutation carriers (p interaction = 0.006), whereas this bias was not observed in BRCA2 mutation carriers (p interaction = 0.95) (Table S4)', 'BRCA2', 'tumors', None)\n",
      "(414, 'That is, despite the expected differences in the frequency of tumor types between the two sets of carriers, heterogeneity was observed in the distribution of rs299290 genotypes in BRCA1, but not BRCA2, mutation carriers', 'BRCA1', 'tumor', None)\n",
      "(417, 'The growth of nonmalignant human mammary epithelial cells, such as MCF10A and HMT3522 S1, within three-dimensional cultures containing reconstituted basement membrane (rBM) recapitulates aspects of the terminal differentiation of mammary luminal epithelia, including apicobasal polarization, growth arrest, and milk production', 'HMT', 'growth arrest', None)\n",
      "(418, 'Thus, loss of BRCA1 function may impair polarization by altering intracellular cytoskeletal organization, resulting in intermediate filament content consistent with the characteristic basal-like tumor type.', 'BRCA1', 'tumor', None)\n",
      "(419, 'While BRCA1 haploinsufficiency does not preclude the formation of a functional luminal layer, the cytoskeletal structure within luminal epithelia from BRCA1 mutation carriers might be compromised', 'BRCA1', 'BRCA1 haploinsufficiency', None)\n",
      "(420, 'While BRCA1 haploinsufficiency does not preclude the formation of a functional luminal layer, the cytoskeletal structure within luminal epithelia from BRCA1 mutation carriers might be compromised', 'BRCA1', 'luminal epithelia', None)\n",
      "(421, 'Importantly, both proteasome inhibition and BRCA1 depletion increased the abundance of RHAMM (Figure S7), which is also consistent with observed RHAMM over-expression in breast cancer cell lines derived from BRCA1 mutation carriers', 'BRCA1', 'breast cancer', None)\n",
      "(422, 'Importantly, both proteasome inhibition and BRCA1 depletion increased the abundance of RHAMM (Figure S7), which is also consistent with observed RHAMM over-expression in breast cancer cell lines derived from BRCA1 mutation carriers', 'RHAMM', 'breast cancer', None)\n",
      "(423, 'Importantly, both proteasome inhibition and BRCA1 depletion increased the abundance of RHAMM (Figure S7), which is also consistent with observed RHAMM over-expression in breast cancer cell lines derived from BRCA1 mutation carriers', 'RHAMM', 'breast cancer', None)\n",
      "(424, 'Importantly, both proteasome inhibition and BRCA1 depletion increased the abundance of RHAMM (Figure S7), which is also consistent with observed RHAMM over-expression in breast cancer cell lines derived from BRCA1 mutation carriers', 'BRCA1', 'breast cancer', None)\n",
      "(425, 'Consistent with previous reports, total RHAMM decorates all microtubule structures throughout mitosis', 'RHAMM', 'mitosis', None)\n",
      "(426, 'Threonine 703 (T703) is an evolutionarily conserved phosphorylated residue in RHAMM similar to a consensus aurora kinase Ipl1p site', 'RHAMM', 'consensus aurora', None)\n",
      "(427, 'Threonine 703 (T703) is an evolutionarily conserved phosphorylated residue in RHAMM similar to a consensus aurora kinase Ipl1p site', 'Ipl1p', 'consensus aurora', None)\n",
      "(428, 'pT703-RHAMM Expression in BRCA1 Mutant Breast Cancer Cells and Tumors; A Mechanistic Model for Polarization and Increased Risk of Breast Cancer', 'RHAMM', 'Breast Cancer Cells and Tumors', None)\n",
      "(429, 'pT703-RHAMM Expression in BRCA1 Mutant Breast Cancer Cells and Tumors; A Mechanistic Model for Polarization and Increased Risk of Breast Cancer', 'RHAMM', 'Breast Cancer', None)\n",
      "(430, 'pT703-RHAMM Expression in BRCA1 Mutant Breast Cancer Cells and Tumors; A Mechanistic Model for Polarization and Increased Risk of Breast Cancer', 'BRCA1', 'Breast Cancer Cells and Tumors', None)\n",
      "(431, 'pT703-RHAMM Expression in BRCA1 Mutant Breast Cancer Cells and Tumors; A Mechanistic Model for Polarization and Increased Risk of Breast Cancer', 'BRCA1', 'Breast Cancer', None)\n",
      "(432, 'To evaluate the link with carcinogenesis, pT703-RHAMM immunochemistry was performed in BRCA1 mutant breast cancer cells, HCC1937 line, their wild-type reconstituted counterparts, and in primary breast tumors', 'RHAMM', 'carcinogenesis', None)\n",
      "(433, 'To evaluate the link with carcinogenesis, pT703-RHAMM immunochemistry was performed in BRCA1 mutant breast cancer cells, HCC1937 line, their wild-type reconstituted counterparts, and in primary breast tumors', 'BRCA1', 'carcinogenesis', None)\n",
      "(434, 'To evaluate the link with carcinogenesis, pT703-RHAMM immunochemistry was performed in BRCA1 mutant breast cancer cells, HCC1937 line, their wild-type reconstituted counterparts, and in primary breast tumors', 'RHAMM', 'breast cancer', None)\n",
      "(435, 'To evaluate the link with carcinogenesis, pT703-RHAMM immunochemistry was performed in BRCA1 mutant breast cancer cells, HCC1937 line, their wild-type reconstituted counterparts, and in primary breast tumors', 'RHAMM', 'breast tumors', None)\n",
      "(436, 'To evaluate the link with carcinogenesis, pT703-RHAMM immunochemistry was performed in BRCA1 mutant breast cancer cells, HCC1937 line, their wild-type reconstituted counterparts, and in primary breast tumors', 'BRCA1', 'breast cancer', None)\n",
      "(437, 'To evaluate the link with carcinogenesis, pT703-RHAMM immunochemistry was performed in BRCA1 mutant breast cancer cells, HCC1937 line, their wild-type reconstituted counterparts, and in primary breast tumors', 'BRCA1', 'breast tumors', None)\n",
      "(438, 'Subsequently, high expression of pT703-RHAMM was scored in 58% (n = 11) and 50% (n = 4) of BRCA1 mutation carriers and sporadic ER-negative tumors, respectively, but in 36% (n = 5) and 30% (n = 10) of BRCA2 mutation carriers and sporadic ER-positive tumors, respectively (Figure 7B)', 'pT703-RHAMM', 'tumors', None)\n",
      "(439, 'Subsequently, high expression of pT703-RHAMM was scored in 58% (n = 11) and 50% (n = 4) of BRCA1 mutation carriers and sporadic ER-negative tumors, respectively, but in 36% (n = 5) and 30% (n = 10) of BRCA2 mutation carriers and sporadic ER-positive tumors, respectively (Figure 7B)', 'pT703-RHAMM', 'tumors', None)\n",
      "(440, 'Subsequently, high expression of pT703-RHAMM was scored in 58% (n = 11) and 50% (n = 4) of BRCA1 mutation carriers and sporadic ER-negative tumors, respectively, but in 36% (n = 5) and 30% (n = 10) of BRCA2 mutation carriers and sporadic ER-positive tumors, respectively (Figure 7B)', 'BRCA1', 'tumors', None)\n",
      "(441, 'Subsequently, high expression of pT703-RHAMM was scored in 58% (n = 11) and 50% (n = 4) of BRCA1 mutation carriers and sporadic ER-negative tumors, respectively, but in 36% (n = 5) and 30% (n = 10) of BRCA2 mutation carriers and sporadic ER-positive tumors, respectively (Figure 7B)', 'BRCA1', 'tumors', None)\n",
      "(442, 'Subsequently, high expression of pT703-RHAMM was scored in 58% (n = 11) and 50% (n = 4) of BRCA1 mutation carriers and sporadic ER-negative tumors, respectively, but in 36% (n = 5) and 30% (n = 10) of BRCA2 mutation carriers and sporadic ER-positive tumors, respectively (Figure 7B)', 'ER', 'tumors', None)\n",
      "(443, 'Subsequently, high expression of pT703-RHAMM was scored in 58% (n = 11) and 50% (n = 4) of BRCA1 mutation carriers and sporadic ER-negative tumors, respectively, but in 36% (n = 5) and 30% (n = 10) of BRCA2 mutation carriers and sporadic ER-positive tumors, respectively (Figure 7B)', 'ER', 'tumors', None)\n",
      "(444, 'Subsequently, high expression of pT703-RHAMM was scored in 58% (n = 11) and 50% (n = 4) of BRCA1 mutation carriers and sporadic ER-negative tumors, respectively, but in 36% (n = 5) and 30% (n = 10) of BRCA2 mutation carriers and sporadic ER-positive tumors, respectively (Figure 7B)', 'BRCA2', 'tumors', None)\n",
      "(445, 'Subsequently, high expression of pT703-RHAMM was scored in 58% (n = 11) and 50% (n = 4) of BRCA1 mutation carriers and sporadic ER-negative tumors, respectively, but in 36% (n = 5) and 30% (n = 10) of BRCA2 mutation carriers and sporadic ER-positive tumors, respectively (Figure 7B)', 'ER', 'tumors', None)\n",
      "(446, 'Subsequently, high expression of pT703-RHAMM was scored in 58% (n = 11) and 50% (n = 4) of BRCA1 mutation carriers and sporadic ER-negative tumors, respectively, but in 36% (n = 5) and 30% (n = 10) of BRCA2 mutation carriers and sporadic ER-positive tumors, respectively (Figure 7B)', 'BRCA2', 'tumors', None)\n",
      "(447, 'Subsequently, high expression of pT703-RHAMM was scored in 58% (n = 11) and 50% (n = 4) of BRCA1 mutation carriers and sporadic ER-negative tumors, respectively, but in 36% (n = 5) and 30% (n = 10) of BRCA2 mutation carriers and sporadic ER-positive tumors, respectively (Figure 7B)', 'ER', 'tumors', None)\n",
      "(448, 'Although this dataset is limited, the results support the indication of an interplay between BRCA1 and RHAMM, which is altered in breast carcinogenesis.', 'BRCA1', 'breast carcinogenesis', None)\n",
      "(449, 'Although this dataset is limited, the results support the indication of an interplay between BRCA1 and RHAMM, which is altered in breast carcinogenesis.', 'RHAMM', 'breast carcinogenesis', None)\n",
      "(450, 'pT703-RHAMM expression in BRCA1 mutant breast cancer cells and tumors.', 'RHAMM', 'breast cancer', None)\n",
      "(451, 'pT703-RHAMM expression in BRCA1 mutant breast cancer cells and tumors.', 'RHAMM', 'tumors', None)\n",
      "(452, 'pT703-RHAMM expression in BRCA1 mutant breast cancer cells and tumors.', 'BRCA1', 'breast cancer', None)\n",
      "(453, 'pT703-RHAMM expression in BRCA1 mutant breast cancer cells and tumors.', 'BRCA1', 'tumors', None)\n",
      "(454, '(B) Results of pT703-RHAMM staining scores in primary breast tumors with different BRCA1/2 mutation and ER status', 'pT703-RHAMM', 'breast tumors', None)\n",
      "(455, '(B) Results of pT703-RHAMM staining scores in primary breast tumors with different BRCA1/2 mutation and ER status', 'BRCA1', 'breast tumors', None)\n",
      "(456, '(B) Results of pT703-RHAMM staining scores in primary breast tumors with different BRCA1/2 mutation and ER status', 'ER', 'breast tumors', None)\n",
      "(457, 'This model is consistent with reduced expression of AURKA, TPX2, and HMMR, but to a lesser extent BRCA1, with polarization and growth arrest of nonmalignant mammary epithelial cells, as measured by gene expression profiling (Figure S11A)', 'AURKA', 'growth arrest', None)\n",
      "(458, 'This model is consistent with reduced expression of AURKA, TPX2, and HMMR, but to a lesser extent BRCA1, with polarization and growth arrest of nonmalignant mammary epithelial cells, as measured by gene expression profiling (Figure S11A)', 'TPX2', 'growth arrest', None)\n",
      "(459, 'This model is consistent with reduced expression of AURKA, TPX2, and HMMR, but to a lesser extent BRCA1, with polarization and growth arrest of nonmalignant mammary epithelial cells, as measured by gene expression profiling (Figure S11A)', 'HMMR', 'growth arrest', None)\n",
      "(460, 'This model is consistent with reduced expression of AURKA, TPX2, and HMMR, but to a lesser extent BRCA1, with polarization and growth arrest of nonmalignant mammary epithelial cells, as measured by gene expression profiling (Figure S11A)', 'BRCA1', 'growth arrest', None)\n",
      "(461, 'Deviation from this pathway, through loss of BRCA1 function or augmentation of microtubule-associated factors, may impair terminal differentiation of luminal epithelia and promote tumorigenesis', 'BRCA1', 'luminal epithelia', None)\n",
      "(462, 'Consistently, HMMR over-expression might be detectable as early as the transition from normal breast tissue to hyperplasia (Figure S11B)', 'HMMR', 'hyperplasia', None)\n",
      "(463, 'We have investigated gene and protein interactions in a centrosome-cantered module, including BRCA1/BRCA1 and HMMR/RHAMM, across biological systems ranging from breast cancer risk estimates to cellular phenotypes and cytoskeletal structures', 'BRCA1', 'breast cancer', None)\n",
      "(464, 'We have investigated gene and protein interactions in a centrosome-cantered module, including BRCA1/BRCA1 and HMMR/RHAMM, across biological systems ranging from breast cancer risk estimates to cellular phenotypes and cytoskeletal structures', 'BRCA1', 'breast cancer', None)\n",
      "(465, 'We have investigated gene and protein interactions in a centrosome-cantered module, including BRCA1/BRCA1 and HMMR/RHAMM, across biological systems ranging from breast cancer risk estimates to cellular phenotypes and cytoskeletal structures', 'HMMR', 'breast cancer', None)\n",
      "(466, 'We have investigated gene and protein interactions in a centrosome-cantered module, including BRCA1/BRCA1 and HMMR/RHAMM, across biological systems ranging from breast cancer risk estimates to cellular phenotypes and cytoskeletal structures', 'RHAMM', 'breast cancer', None)\n",
      "(467, 'Accordingly, a common candidate breast cancer-predisposition allele in HMMR, originally identified in an Ashkenazi Jewish study, may specifically modify breast cancer risk among BRCA1 mutation carriers', 'HMMR', 'breast cancer', None)\n",
      "(468, 'Accordingly, a common candidate breast cancer-predisposition allele in HMMR, originally identified in an Ashkenazi Jewish study, may specifically modify breast cancer risk among BRCA1 mutation carriers', 'BRCA1', 'breast cancer', None)\n",
      "(469, 'Accordingly, a common candidate breast cancer-predisposition allele in HMMR, originally identified in an Ashkenazi Jewish study, may specifically modify breast cancer risk among BRCA1 mutation carriers', 'HMMR', 'breast cancer', None)\n",
      "(470, 'Accordingly, a common candidate breast cancer-predisposition allele in HMMR, originally identified in an Ashkenazi Jewish study, may specifically modify breast cancer risk among BRCA1 mutation carriers', 'BRCA1', 'breast cancer', None)\n",
      "(471, 'A recent report has suggested that common genetic variation in genes encoding for centrosome pathway components (excluding AURKA and HMMR) may frequently influence risk of breast cancer and, notably, includes variants in TACC3-a proposed HMMR homolog -TUBG1, and TPX2 loci', 'AURKA', 'breast cancer', None)\n",
      "(472, 'A recent report has suggested that common genetic variation in genes encoding for centrosome pathway components (excluding AURKA and HMMR) may frequently influence risk of breast cancer and, notably, includes variants in TACC3-a proposed HMMR homolog -TUBG1, and TPX2 loci', 'HMMR', 'breast cancer', None)\n",
      "(473, 'A recent report has suggested that common genetic variation in genes encoding for centrosome pathway components (excluding AURKA and HMMR) may frequently influence risk of breast cancer and, notably, includes variants in TACC3-a proposed HMMR homolog -TUBG1, and TPX2 loci', 'TACC3', 'breast cancer', None)\n",
      "(474, 'A recent report has suggested that common genetic variation in genes encoding for centrosome pathway components (excluding AURKA and HMMR) may frequently influence risk of breast cancer and, notably, includes variants in TACC3-a proposed HMMR homolog -TUBG1, and TPX2 loci', 'HMMR', 'breast cancer', None)\n",
      "(475, 'A recent report has suggested that common genetic variation in genes encoding for centrosome pathway components (excluding AURKA and HMMR) may frequently influence risk of breast cancer and, notably, includes variants in TACC3-a proposed HMMR homolog -TUBG1, and TPX2 loci', 'TUBG1', 'breast cancer', None)\n",
      "(476, 'A recent report has suggested that common genetic variation in genes encoding for centrosome pathway components (excluding AURKA and HMMR) may frequently influence risk of breast cancer and, notably, includes variants in TACC3-a proposed HMMR homolog -TUBG1, and TPX2 loci', 'TPX2', 'breast cancer', None)\n",
      "(477, 'A unifying mechanism of breast carcinogenesis linked to BRCA1 loss-of-function should provide a comprehensive explanation for the observed accumulation of stem and luminal progenitor cells, and for the characteristic pathological features of the corresponding tumors', 'BRCA1', 'breast carcinogenesis', None)\n",
      "(478, 'A unifying mechanism of breast carcinogenesis linked to BRCA1 loss-of-function should provide a comprehensive explanation for the observed accumulation of stem and luminal progenitor cells, and for the characteristic pathological features of the corresponding tumors', 'BRCA1', 'tumors', None)\n",
      "(479, 'Should BRCA1 function(s) promote this transition in mammary stem/progenitor cells, impaired luminal differentiation in BRCA1 mutation carriers and a propensity to develop basal-like tumors with elevated proliferative capacity would be expected', 'BRCA1', 'tumors', None)\n",
      "(480, 'Should BRCA1 function(s) promote this transition in mammary stem/progenitor cells, impaired luminal differentiation in BRCA1 mutation carriers and a propensity to develop basal-like tumors with elevated proliferative capacity would be expected', 'BRCA1', 'tumors', None)\n",
      "(481, 'Accordingly, BRCA1 depletion increases the clonogenic potential of mammary epithelia, while a BRCA1 S308A mutant alters embryonic stem cell differentiation', 'BRCA1', 'epithelia', None)\n",
      "(482, 'Accordingly, BRCA1 depletion increases the clonogenic potential of mammary epithelia, while a BRCA1 S308A mutant alters embryonic stem cell differentiation', 'BRCA1', 'epithelia', None)\n",
      "(483, 'A key question remains regarding the significance of BRCA1 function to stem/progenitor differentiation and BRCA1 haploinsufficiency', 'BRCA1', 'BRCA1 haploinsufficiency', None)\n",
      "(484, 'Examination of histologically normal breast tissue in BRCA1 mutation carriers revealed cellular foci expressing stem cell markers and lacking cytoskeletal structures characteristic of luminal epithelia', 'BRCA1', 'luminal epithelia', None)\n",
      "(485, 'Thus, BRCA1 dose or mutation type may distinctly affect the tissue architecture and function, leading to differences in the accumulation of stem or progenitor cells, and the resulting tumor type', 'BRCA1', 'tumor', None)\n",
      "(486, 'This work was funded by the Spanish Ministries of Health, and Science ane Innovation (CB07/02/2005; FIS 08/1120, 08/1359, 08/1635, and 09/02483; RTICCC RD06/0020/1060 and RD06/0020/0028; Transversal Action Against Cancer; the Spanish Biomedical Research Centre Networks for Epidemiology and Public Health, and Rare Diseases; and the \"Ramon y Cajal\" Young Investigator Program), the Spanish National Society of Medical Oncology (2010), the Spanish Association Against Cancer (AECC 2010), the AGAUR Catalan Government Agency (2009SGR1489 and 2009SGR293; and the Beatriu Pinos Postdoctoral Program), the Ramon Areces Foundation (XV), the \"Roses Contra el Cancer\" Foundation, the Michael Cuccione Foundation for Childhood Cancer Research, Cancer Research-UK (C490/A10119, C1287/A8874, C1287/A10118, C5047/A8385, and C8197/A10123), the National Institute for Health Research (UK), the Association for International Cancer Research (AICR-07-0454), the Ligue National Contre le Cancer (France), the Association \"Le cancer du sein, parlons-en!\", the Dutch Cancer Society (NKI 1998-1854, 2004-3088, and 2007-3756), the Fondazione Italiana per la Ricerca sul Cancro (\"Hereditary Tumors\"), the Associazione Italiana per la Ricerca sul Cancro (4017), the Italian Ministero della Salute (RFPS-2006-3-340203 and \"Progetto Tumori Femminili\"), the Italian Ministero dell\\'Universita\\' e Ricerca (RBLAO3-BETH), the Fondazione IRCCS Istituto Nazionale Tumori (INT \"5x1000\"), the Fondazione Cassa di Risparmio di Pisa (Istituto Toscano Tumori), the National Breast Cancer Foundation (Australia), the Australian National Health and Medical Research Council (145684, 288704, and 454508), the Queensland Cancer Fund, the Cancer Councils of New South Wales, Victoria, Tasmania, and South Australia, the Cancer Foundation of Western Australia, the German Cancer Aid (107054), the Center for Molecular Medicine Cologne (TV93), the National Cancer Institute (USA; CA128978 and CA122340), National Institutes of Health (RFA-CA-06-503, BCFR U01 CA69398, CA69417, CA69446, CA69467, CA69631, and CA69638), the Research Triangle Institute Informatics Support Center (RFP N02PC45022-46), the Specialized Program of Research Excellence (SPORE P50 CA83638 and CA113916), the Department of Defense Breast Cancer Research Program (05/0612), the Eileen Stein Jacoby Fund, the Breast Cancer Research Foundation, the Marianne and Robert MacDonald Foundation, the Komen Foundation, the Helsinki University Central Hospital Research Fund, the Academy of Finland (110663), the Finnish Cancer Society, the Sigrid Juselius Foundation, and the EU FP7 (223175, HEALTH-F2-2009-223175)', 'FIS', 'Contra el Cancer', None)\n",
      "(487, 'This work was funded by the Spanish Ministries of Health, and Science ane Innovation (CB07/02/2005; FIS 08/1120, 08/1359, 08/1635, and 09/02483; RTICCC RD06/0020/1060 and RD06/0020/0028; Transversal Action Against Cancer; the Spanish Biomedical Research Centre Networks for Epidemiology and Public Health, and Rare Diseases; and the \"Ramon y Cajal\" Young Investigator Program), the Spanish National Society of Medical Oncology (2010), the Spanish Association Against Cancer (AECC 2010), the AGAUR Catalan Government Agency (2009SGR1489 and 2009SGR293; and the Beatriu Pinos Postdoctoral Program), the Ramon Areces Foundation (XV), the \"Roses Contra el Cancer\" Foundation, the Michael Cuccione Foundation for Childhood Cancer Research, Cancer Research-UK (C490/A10119, C1287/A8874, C1287/A10118, C5047/A8385, and C8197/A10123), the National Institute for Health Research (UK), the Association for International Cancer Research (AICR-07-0454), the Ligue National Contre le Cancer (France), the Association \"Le cancer du sein, parlons-en!\", the Dutch Cancer Society (NKI 1998-1854, 2004-3088, and 2007-3756), the Fondazione Italiana per la Ricerca sul Cancro (\"Hereditary Tumors\"), the Associazione Italiana per la Ricerca sul Cancro (4017), the Italian Ministero della Salute (RFPS-2006-3-340203 and \"Progetto Tumori Femminili\"), the Italian Ministero dell\\'Universita\\' e Ricerca (RBLAO3-BETH), the Fondazione IRCCS Istituto Nazionale Tumori (INT \"5x1000\"), the Fondazione Cassa di Risparmio di Pisa (Istituto Toscano Tumori), the National Breast Cancer Foundation (Australia), the Australian National Health and Medical Research Council (145684, 288704, and 454508), the Queensland Cancer Fund, the Cancer Councils of New South Wales, Victoria, Tasmania, and South Australia, the Cancer Foundation of Western Australia, the German Cancer Aid (107054), the Center for Molecular Medicine Cologne (TV93), the National Cancer Institute (USA; CA128978 and CA122340), National Institutes of Health (RFA-CA-06-503, BCFR U01 CA69398, CA69417, CA69446, CA69467, CA69631, and CA69638), the Research Triangle Institute Informatics Support Center (RFP N02PC45022-46), the Specialized Program of Research Excellence (SPORE P50 CA83638 and CA113916), the Department of Defense Breast Cancer Research Program (05/0612), the Eileen Stein Jacoby Fund, the Breast Cancer Research Foundation, the Marianne and Robert MacDonald Foundation, the Komen Foundation, the Helsinki University Central Hospital Research Fund, the Academy of Finland (110663), the Finnish Cancer Society, the Sigrid Juselius Foundation, and the EU FP7 (223175, HEALTH-F2-2009-223175)', 'FIS', 'Childhood Cancer', None)\n",
      "(488, 'This work was funded by the Spanish Ministries of Health, and Science ane Innovation (CB07/02/2005; FIS 08/1120, 08/1359, 08/1635, and 09/02483; RTICCC RD06/0020/1060 and RD06/0020/0028; Transversal Action Against Cancer; the Spanish Biomedical Research Centre Networks for Epidemiology and Public Health, and Rare Diseases; and the \"Ramon y Cajal\" Young Investigator Program), the Spanish National Society of Medical Oncology (2010), the Spanish Association Against Cancer (AECC 2010), the AGAUR Catalan Government Agency (2009SGR1489 and 2009SGR293; and the Beatriu Pinos Postdoctoral Program), the Ramon Areces Foundation (XV), the \"Roses Contra el Cancer\" Foundation, the Michael Cuccione Foundation for Childhood Cancer Research, Cancer Research-UK (C490/A10119, C1287/A8874, C1287/A10118, C5047/A8385, and C8197/A10123), the National Institute for Health Research (UK), the Association for International Cancer Research (AICR-07-0454), the Ligue National Contre le Cancer (France), the Association \"Le cancer du sein, parlons-en!\", the Dutch Cancer Society (NKI 1998-1854, 2004-3088, and 2007-3756), the Fondazione Italiana per la Ricerca sul Cancro (\"Hereditary Tumors\"), the Associazione Italiana per la Ricerca sul Cancro (4017), the Italian Ministero della Salute (RFPS-2006-3-340203 and \"Progetto Tumori Femminili\"), the Italian Ministero dell\\'Universita\\' e Ricerca (RBLAO3-BETH), the Fondazione IRCCS Istituto Nazionale Tumori (INT \"5x1000\"), the Fondazione Cassa di Risparmio di Pisa (Istituto Toscano Tumori), the National Breast Cancer Foundation (Australia), the Australian National Health and Medical Research Council (145684, 288704, and 454508), the Queensland Cancer Fund, the Cancer Councils of New South Wales, Victoria, Tasmania, and South Australia, the Cancer Foundation of Western Australia, the German Cancer Aid (107054), the Center for Molecular Medicine Cologne (TV93), the National Cancer Institute (USA; CA128978 and CA122340), National Institutes of Health (RFA-CA-06-503, BCFR U01 CA69398, CA69417, CA69446, CA69467, CA69631, and CA69638), the Research Triangle Institute Informatics Support Center (RFP N02PC45022-46), the Specialized Program of Research Excellence (SPORE P50 CA83638 and CA113916), the Department of Defense Breast Cancer Research Program (05/0612), the Eileen Stein Jacoby Fund, the Breast Cancer Research Foundation, the Marianne and Robert MacDonald Foundation, the Komen Foundation, the Helsinki University Central Hospital Research Fund, the Academy of Finland (110663), the Finnish Cancer Society, the Sigrid Juselius Foundation, and the EU FP7 (223175, HEALTH-F2-2009-223175)', 'FIS', 'cancer', None)\n",
      "(499, 'The pattern of expression of the microtubule-binding protein RHAMM/IHABP in mammary carcinoma suggests a role in the invasive behaviour of tumour cells.', 'RHAMM', 'tumour', None)\n",
      "(500, 'The pattern of expression of the microtubule-binding protein RHAMM/IHABP in mammary carcinoma suggests a role in the invasive behaviour of tumour cells.', 'IHABP', 'carcinoma', None)\n",
      "(489, 'This work was funded by the Spanish Ministries of Health, and Science ane Innovation (CB07/02/2005; FIS 08/1120, 08/1359, 08/1635, and 09/02483; RTICCC RD06/0020/1060 and RD06/0020/0028; Transversal Action Against Cancer; the Spanish Biomedical Research Centre Networks for Epidemiology and Public Health, and Rare Diseases; and the \"Ramon y Cajal\" Young Investigator Program), the Spanish National Society of Medical Oncology (2010), the Spanish Association Against Cancer (AECC 2010), the AGAUR Catalan Government Agency (2009SGR1489 and 2009SGR293; and the Beatriu Pinos Postdoctoral Program), the Ramon Areces Foundation (XV), the \"Roses Contra el Cancer\" Foundation, the Michael Cuccione Foundation for Childhood Cancer Research, Cancer Research-UK (C490/A10119, C1287/A8874, C1287/A10118, C5047/A8385, and C8197/A10123), the National Institute for Health Research (UK), the Association for International Cancer Research (AICR-07-0454), the Ligue National Contre le Cancer (France), the Association \"Le cancer du sein, parlons-en!\", the Dutch Cancer Society (NKI 1998-1854, 2004-3088, and 2007-3756), the Fondazione Italiana per la Ricerca sul Cancro (\"Hereditary Tumors\"), the Associazione Italiana per la Ricerca sul Cancro (4017), the Italian Ministero della Salute (RFPS-2006-3-340203 and \"Progetto Tumori Femminili\"), the Italian Ministero dell\\'Universita\\' e Ricerca (RBLAO3-BETH), the Fondazione IRCCS Istituto Nazionale Tumori (INT \"5x1000\"), the Fondazione Cassa di Risparmio di Pisa (Istituto Toscano Tumori), the National Breast Cancer Foundation (Australia), the Australian National Health and Medical Research Council (145684, 288704, and 454508), the Queensland Cancer Fund, the Cancer Councils of New South Wales, Victoria, Tasmania, and South Australia, the Cancer Foundation of Western Australia, the German Cancer Aid (107054), the Center for Molecular Medicine Cologne (TV93), the National Cancer Institute (USA; CA128978 and CA122340), National Institutes of Health (RFA-CA-06-503, BCFR U01 CA69398, CA69417, CA69446, CA69467, CA69631, and CA69638), the Research Triangle Institute Informatics Support Center (RFP N02PC45022-46), the Specialized Program of Research Excellence (SPORE P50 CA83638 and CA113916), the Department of Defense Breast Cancer Research Program (05/0612), the Eileen Stein Jacoby Fund, the Breast Cancer Research Foundation, the Marianne and Robert MacDonald Foundation, the Komen Foundation, the Helsinki University Central Hospital Research Fund, the Academy of Finland (110663), the Finnish Cancer Society, the Sigrid Juselius Foundation, and the EU FP7 (223175, HEALTH-F2-2009-223175)', 'FIS', 'Hereditary Tumors', None)\n",
      "(490, 'This work was funded by the Spanish Ministries of Health, and Science ane Innovation (CB07/02/2005; FIS 08/1120, 08/1359, 08/1635, and 09/02483; RTICCC RD06/0020/1060 and RD06/0020/0028; Transversal Action Against Cancer; the Spanish Biomedical Research Centre Networks for Epidemiology and Public Health, and Rare Diseases; and the \"Ramon y Cajal\" Young Investigator Program), the Spanish National Society of Medical Oncology (2010), the Spanish Association Against Cancer (AECC 2010), the AGAUR Catalan Government Agency (2009SGR1489 and 2009SGR293; and the Beatriu Pinos Postdoctoral Program), the Ramon Areces Foundation (XV), the \"Roses Contra el Cancer\" Foundation, the Michael Cuccione Foundation for Childhood Cancer Research, Cancer Research-UK (C490/A10119, C1287/A8874, C1287/A10118, C5047/A8385, and C8197/A10123), the National Institute for Health Research (UK), the Association for International Cancer Research (AICR-07-0454), the Ligue National Contre le Cancer (France), the Association \"Le cancer du sein, parlons-en!\", the Dutch Cancer Society (NKI 1998-1854, 2004-3088, and 2007-3756), the Fondazione Italiana per la Ricerca sul Cancro (\"Hereditary Tumors\"), the Associazione Italiana per la Ricerca sul Cancro (4017), the Italian Ministero della Salute (RFPS-2006-3-340203 and \"Progetto Tumori Femminili\"), the Italian Ministero dell\\'Universita\\' e Ricerca (RBLAO3-BETH), the Fondazione IRCCS Istituto Nazionale Tumori (INT \"5x1000\"), the Fondazione Cassa di Risparmio di Pisa (Istituto Toscano Tumori), the National Breast Cancer Foundation (Australia), the Australian National Health and Medical Research Council (145684, 288704, and 454508), the Queensland Cancer Fund, the Cancer Councils of New South Wales, Victoria, Tasmania, and South Australia, the Cancer Foundation of Western Australia, the German Cancer Aid (107054), the Center for Molecular Medicine Cologne (TV93), the National Cancer Institute (USA; CA128978 and CA122340), National Institutes of Health (RFA-CA-06-503, BCFR U01 CA69398, CA69417, CA69446, CA69467, CA69631, and CA69638), the Research Triangle Institute Informatics Support Center (RFP N02PC45022-46), the Specialized Program of Research Excellence (SPORE P50 CA83638 and CA113916), the Department of Defense Breast Cancer Research Program (05/0612), the Eileen Stein Jacoby Fund, the Breast Cancer Research Foundation, the Marianne and Robert MacDonald Foundation, the Komen Foundation, the Helsinki University Central Hospital Research Fund, the Academy of Finland (110663), the Finnish Cancer Society, the Sigrid Juselius Foundation, and the EU FP7 (223175, HEALTH-F2-2009-223175)', 'FIS', 'Breast Cancer', None)\n",
      "(491, 'This work was funded by the Spanish Ministries of Health, and Science ane Innovation (CB07/02/2005; FIS 08/1120, 08/1359, 08/1635, and 09/02483; RTICCC RD06/0020/1060 and RD06/0020/0028; Transversal Action Against Cancer; the Spanish Biomedical Research Centre Networks for Epidemiology and Public Health, and Rare Diseases; and the \"Ramon y Cajal\" Young Investigator Program), the Spanish National Society of Medical Oncology (2010), the Spanish Association Against Cancer (AECC 2010), the AGAUR Catalan Government Agency (2009SGR1489 and 2009SGR293; and the Beatriu Pinos Postdoctoral Program), the Ramon Areces Foundation (XV), the \"Roses Contra el Cancer\" Foundation, the Michael Cuccione Foundation for Childhood Cancer Research, Cancer Research-UK (C490/A10119, C1287/A8874, C1287/A10118, C5047/A8385, and C8197/A10123), the National Institute for Health Research (UK), the Association for International Cancer Research (AICR-07-0454), the Ligue National Contre le Cancer (France), the Association \"Le cancer du sein, parlons-en!\", the Dutch Cancer Society (NKI 1998-1854, 2004-3088, and 2007-3756), the Fondazione Italiana per la Ricerca sul Cancro (\"Hereditary Tumors\"), the Associazione Italiana per la Ricerca sul Cancro (4017), the Italian Ministero della Salute (RFPS-2006-3-340203 and \"Progetto Tumori Femminili\"), the Italian Ministero dell\\'Universita\\' e Ricerca (RBLAO3-BETH), the Fondazione IRCCS Istituto Nazionale Tumori (INT \"5x1000\"), the Fondazione Cassa di Risparmio di Pisa (Istituto Toscano Tumori), the National Breast Cancer Foundation (Australia), the Australian National Health and Medical Research Council (145684, 288704, and 454508), the Queensland Cancer Fund, the Cancer Councils of New South Wales, Victoria, Tasmania, and South Australia, the Cancer Foundation of Western Australia, the German Cancer Aid (107054), the Center for Molecular Medicine Cologne (TV93), the National Cancer Institute (USA; CA128978 and CA122340), National Institutes of Health (RFA-CA-06-503, BCFR U01 CA69398, CA69417, CA69446, CA69467, CA69631, and CA69638), the Research Triangle Institute Informatics Support Center (RFP N02PC45022-46), the Specialized Program of Research Excellence (SPORE P50 CA83638 and CA113916), the Department of Defense Breast Cancer Research Program (05/0612), the Eileen Stein Jacoby Fund, the Breast Cancer Research Foundation, the Marianne and Robert MacDonald Foundation, the Komen Foundation, the Helsinki University Central Hospital Research Fund, the Academy of Finland (110663), the Finnish Cancer Society, the Sigrid Juselius Foundation, and the EU FP7 (223175, HEALTH-F2-2009-223175)', 'FIS', 'Breast Cancer', None)\n",
      "(492, 'This work was funded by the Spanish Ministries of Health, and Science ane Innovation (CB07/02/2005; FIS 08/1120, 08/1359, 08/1635, and 09/02483; RTICCC RD06/0020/1060 and RD06/0020/0028; Transversal Action Against Cancer; the Spanish Biomedical Research Centre Networks for Epidemiology and Public Health, and Rare Diseases; and the \"Ramon y Cajal\" Young Investigator Program), the Spanish National Society of Medical Oncology (2010), the Spanish Association Against Cancer (AECC 2010), the AGAUR Catalan Government Agency (2009SGR1489 and 2009SGR293; and the Beatriu Pinos Postdoctoral Program), the Ramon Areces Foundation (XV), the \"Roses Contra el Cancer\" Foundation, the Michael Cuccione Foundation for Childhood Cancer Research, Cancer Research-UK (C490/A10119, C1287/A8874, C1287/A10118, C5047/A8385, and C8197/A10123), the National Institute for Health Research (UK), the Association for International Cancer Research (AICR-07-0454), the Ligue National Contre le Cancer (France), the Association \"Le cancer du sein, parlons-en!\", the Dutch Cancer Society (NKI 1998-1854, 2004-3088, and 2007-3756), the Fondazione Italiana per la Ricerca sul Cancro (\"Hereditary Tumors\"), the Associazione Italiana per la Ricerca sul Cancro (4017), the Italian Ministero della Salute (RFPS-2006-3-340203 and \"Progetto Tumori Femminili\"), the Italian Ministero dell\\'Universita\\' e Ricerca (RBLAO3-BETH), the Fondazione IRCCS Istituto Nazionale Tumori (INT \"5x1000\"), the Fondazione Cassa di Risparmio di Pisa (Istituto Toscano Tumori), the National Breast Cancer Foundation (Australia), the Australian National Health and Medical Research Council (145684, 288704, and 454508), the Queensland Cancer Fund, the Cancer Councils of New South Wales, Victoria, Tasmania, and South Australia, the Cancer Foundation of Western Australia, the German Cancer Aid (107054), the Center for Molecular Medicine Cologne (TV93), the National Cancer Institute (USA; CA128978 and CA122340), National Institutes of Health (RFA-CA-06-503, BCFR U01 CA69398, CA69417, CA69446, CA69467, CA69631, and CA69638), the Research Triangle Institute Informatics Support Center (RFP N02PC45022-46), the Specialized Program of Research Excellence (SPORE P50 CA83638 and CA113916), the Department of Defense Breast Cancer Research Program (05/0612), the Eileen Stein Jacoby Fund, the Breast Cancer Research Foundation, the Marianne and Robert MacDonald Foundation, the Komen Foundation, the Helsinki University Central Hospital Research Fund, the Academy of Finland (110663), the Finnish Cancer Society, the Sigrid Juselius Foundation, and the EU FP7 (223175, HEALTH-F2-2009-223175)', 'FIS', 'Breast Cancer', None)\n",
      "(493, 'CD24 staining of mouse mammary gland cells defines luminal epithelial, myoepithelial/basal and non-epithelial cells.', 'CD24', 'myoepithelial', None)\n",
      "(494, 'Basal-like breast cancer and the BRCA1 phenotype.', 'BRCA1', 'breast cancer', None)\n",
      "(495, 'The basal phenotype of BRCA1-related breast cancer: past, present and future.', 'BRCA1', 'breast cancer', None)\n",
      "(496, 'Aberrant luminal progenitors as the candidate target population for basal tumor development in BRCA1 mutation carriers.', 'BRCA1', 'basal tumor', None)\n",
      "(497, ' BRCA1 basal-like breast cancers originate from luminal epithelial progenitors and not from basal stem cells.', 'BRCA1', 'breast cancers', None)\n",
      "(498, 'The pattern of expression of the microtubule-binding protein RHAMM/IHABP in mammary carcinoma suggests a role in the invasive behaviour of tumour cells.', 'RHAMM', 'carcinoma', None)\n",
      "(501, 'The pattern of expression of the microtubule-binding protein RHAMM/IHABP in mammary carcinoma suggests a role in the invasive behaviour of tumour cells.', 'IHABP', 'tumour', None)\n",
      "(502, 'Receptor for hyaluronan-mediated motility correlates with centrosome abnormalities in multiple myeloma and maintains mitotic integrity.', 'Receptor for hyaluronan-mediated motility', 'centrosome abnormalities in multiple myeloma', None)\n",
      "(503, 'A locus on 19p13 modifies risk of breast cancer in BRCA1 mutation carriers and is associated with hormone receptor-negative breast cancer in the general population.', 'BRCA1', 'breast cancer', None)\n",
      "(504, 'A locus on 19p13 modifies risk of breast cancer in BRCA1 mutation carriers and is associated with hormone receptor-negative breast cancer in the general population.', 'BRCA1', 'breast cancer', None)\n",
      "(505, 'Common variants in LSP1, 2q35 and 8q24 and breast cancer risk for BRCA1 and BRCA2 mutation carriers.', 'LSP1', 'breast cancer', None)\n",
      "(506, 'Common variants in LSP1, 2q35 and 8q24 and breast cancer risk for BRCA1 and BRCA2 mutation carriers.', 'BRCA1', 'breast cancer', None)\n",
      "(507, 'Common variants in LSP1, 2q35 and 8q24 and breast cancer risk for BRCA1 and BRCA2 mutation carriers.', 'BRCA2', 'breast cancer', None)\n",
      "(508, 'Common breast cancer-predisposition alleles are associated with breast cancer risk in BRCA1 and BRCA2 mutation carriers.', 'BRCA1', 'breast cancer', None)\n",
      "(509, 'Common breast cancer-predisposition alleles are associated with breast cancer risk in BRCA1 and BRCA2 mutation carriers.', 'BRCA2', 'breast cancer', None)\n",
      "(510, 'Common breast cancer-predisposition alleles are associated with breast cancer risk in BRCA1 and BRCA2 mutation carriers.', 'BRCA1', 'breast cancer', None)\n",
      "(511, 'Common breast cancer-predisposition alleles are associated with breast cancer risk in BRCA1 and BRCA2 mutation carriers.', 'BRCA2', 'breast cancer', None)\n",
      "(512, ' RAD51 135G C modifies breast cancer risk among BRCA2 mutation carriers: results from a combined analysis of 19 studies.', 'RAD51', 'breast cancer', None)\n",
      "(513, ' RAD51 135G C modifies breast cancer risk among BRCA2 mutation carriers: results from a combined analysis of 19 studies.', 'BRCA2', 'breast cancer', None)\n",
      "(514, 'RHAMM expression and isoform balance predict aggressive disease and poor survival in multiple myeloma.', 'RHAMM', 'aggressive disease', None)\n",
      "(515, 'RHAMM expression and isoform balance predict aggressive disease and poor survival in multiple myeloma.', 'RHAMM', 'multiple myeloma', None)\n",
      "(516, 'Vimentin and laminin expression is associated with basal-like phenotype in both sporadic and BRCA1-associated breast carcinomas.', 'Vimentin', 'breast carcinomas', None)\n",
      "(517, 'Vimentin and laminin expression is associated with basal-like phenotype in both sporadic and BRCA1-associated breast carcinomas.', 'BRCA1', 'breast carcinomas', None)\n",
      "(518, 'Identification of a new APC/C recognition domain, the A box, which is required for the Cdh1-dependent destruction of the kinase Aurora-A during mitotic exit.', 'Cdh1', 'APC', None)\n",
      "(519, 'Identification of a new APC/C recognition domain, the A box, which is required for the Cdh1-dependent destruction of the kinase Aurora-A during mitotic exit.', 'Aurora-A', 'APC', None)\n",
      "(520, 'Aurora-A kinase regulates breast cancer-associated gene 1 inhibition of centrosome-dependent microtubule nucleation.', 'Aurora-A', 'breast cancer', None)\n",
      "(521, ' Hyaluronan-mediated motility receptor gene single nucleotide polymorphisms and risk of breast cancer.', 'Hyaluronan-mediated motility receptor', 'breast cancer', None)\n",
      "(522, 'Inconsistent association between the STK15 F31I genetic polymorphism and breast cancer risk.', 'STK15', 'breast cancer', None)\n"
     ]
    }
   ],
   "source": [
    "#store the dataframe into psql table\n",
    "import psycopg2\n",
    "import pandas as pd\n",
    "from sqlalchemy import create_engine\n",
    "\n",
    "conn_string = 'postgresql://postgres:sinkiru@localhost:5432/gdmbiobert'\n",
    "\n",
    "db = create_engine(conn_string)\n",
    "conn = db.connect()\n",
    "\n",
    "final_re_output.to_sql('gd', con=conn, if_exists='replace', index=False)\n",
    "conn = psycopg2.connect(conn_string)\n",
    "conn.autocommit = True\n",
    "cursor = conn.cursor()\n",
    "\n",
    "sql1 = '''select * from gd;'''\n",
    "cursor.execute(sql1)\n",
    "for i in cursor.fetchall():\n",
    "\tprint(i)\n",
    "\n",
    "# conn.commit()\n",
    "conn.close()\n"
   ]
  }
 ],
 "metadata": {
  "colab": {
   "provenance": []
  },
  "gpuClass": "standard",
  "kernelspec": {
   "display_name": "Python 3.6.13 ('env1')",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.13"
  },
  "vscode": {
   "interpreter": {
    "hash": "6626a9a69e96ea8665bf0f933e0462e9e2599c09c430a57c21911de9c8d78cbc"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
